%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Doctoral Thesis 
% LaTeX Template
% Version 2.5 (27/8/17)
%
% This template was downloaded from:
% http://www.LaTeXTemplates.com
%
% Version 2.x major modifications by:
% Vel (vel@latextemplates.com)
%
% This template is based on a template by:
% Steve Gunn (http://users.ecs.soton.ac.uk/srg/softwaretools/document/templates/)
% Sunil Patel (http://www.sunilpatel.co.uk/thesis-template/)
%
% Template license:
% CC BY-NC-SA 3.0 (http://creativecommons.org/licenses/by-nc-sa/3.0/)
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%----------------------------------------------------------------------------------------
%	PACKAGES AND OTHER DOCUMENT CONFIGURATIONS
%----------------------------------------------------------------------------------------
%\PassOptionsToPackage{english,greek}{babel}
\documentclass[
11pt, % The default document font size, options: 10pt, 11pt, 12pt
%oneside, % Two side (alternating margins) for binding by default, uncomment to switch to one side
english, % ngerman for German
singlespacing, % Single line spacing, alternatives: onehalfspacing or doublespacing
%draft, % Uncomment to enable draft mode (no pictures, no links, overfull hboxes indicated)
%nolistspacing, % If the document is onehalfspacing or doublespacing, uncomment this to set spacing in lists to single
liststotoc, % Uncomment to add the list of figures/tables/etc to the table of contents
toctotoc, % Uncomment to add the main table of contents to the table of contents
%parskip, % Uncomment to add space between paragraphs
%nohyperref, % Uncomment to not load the hyperref package
headsepline, % Uncomment to get a line under the header
%chapterinoneline, % Uncomment to place the chapter title next to the number on one line
%consistentlayout, % Uncomment to change the layout of the declaration, abstract and acknowledgements pages to match the default layout
]{MastersDoctoralThesis} % The class file specifying the document structure

\usepackage[utf8]{inputenc} % Required for inputting international characters
\usepackage[T1]{fontenc} % Output font encoding for international characters
%\usepackage{textgreek}
%\usepackage[main=english,greek]{babel}

\usepackage{mathpazo} % Use the Palatino font by default

\usepackage[
backend=biber,
style=authoryear,
sorting=nyt]{biblatex} % Use the bibtex backend with the authoryear citation style (which resembles APA)

\addbibresource{references.bib} % The filename of the bibliography
% try to mention my works
\DeclareSourcemap{
  \maps[datatype=bibtex]{
    \map{
      \step[fieldsource=author,
            match=Paragkamian,
            final]
      \step[fieldset=keywords, fieldvalue=own]
    }
  }
}


\usepackage[autostyle=true]{csquotes} % Required to generate language-dependent quotes in the bibliography

% Here you can load other packages or provide your own definitions
% ----------------------------------------------------------------
% Finally, hyperref is used for PDF files.
\usepackage[pdfusetitle, colorlinks, citecolor, plainpages=false]{hyperref}
\usepackage{xcolor}
%----------------------------------------------------------------------------------------
% Colors
%----------------------------------------------------------------------------------------
\hypersetup{
linkcolor=teal
,citecolor=Green
}

% tables
\usepackage{graphicx}
% To have multirows in tables
\usepackage{multirow}


% To have rotated tables
\usepackage{rotating} 
\usepackage{float}

\usepackage{changepage}
% To have check boxes
\usepackage{pifont}
\usepackage{amsmath}
\usepackage{multirow}
% For lemas
\usepackage{amsthm}
\usepackage{amssymb}

% To have todo notes
%\usepackage{todonotes}

% To have part A, B etc in a figure
\usepackage{subcaption}
\usepackage{caption}

% to have multiline equations
\usepackage{mathtools}

%----------------------------------------------------------------------------------------
%	MARGIN SETTINGS
%----------------------------------------------------------------------------------------

\geometry{
	paper=a4paper, % Change to letterpaper for US letter
	inner=2.5cm, % Inner margin
	outer=3.8cm, % Outer margin
	bindingoffset=.5cm, % Binding offset
	top=1.5cm, % Top margin
	bottom=1.5cm, % Bottom margin
	%showframe, % Uncomment to show how the type block is set on the page
}

%----------------------------------------------------------------------------------------
%	THESIS INFORMATION
%----------------------------------------------------------------------------------------

\thesistitle{Deciphering the relation of the microbiome interactome with ecosystem function and biogeochemical processes} % Your thesis title, this is used in the title and abstract, print it elsewhere with \ttitle
\supervisor{Prof. Panagiotis F. \textsc{Sarris}} % Your supervisor's name, this is used in the title page, print it elsewhere with \supname
\examiner{} % Your examiner's name, this is not currently used anywhere in the template, print it elsewhere with \examname
\degree{Doctor of Philosophy} % Your degree name, this is used in the title page and abstract, print it elsewhere with \degreename
\author{Savvas \textsc{Paragkamian}} % Your name, this is used in the title page and abstract, print it elsewhere with \authorname
\addresses{} % Your address, this is not currently used anywhere in the template, print it elsewhere with \addressname

\subject{Microbial Ecology} % Your subject area, this is not currently used anywhere in the template, print it elsewhere with \subjectname
\keywords{ecosystem functioning, macroecology, microbial networks, conservation} % Keywords for your thesis, this is not currently used anywhere in the template, print it elsewhere with \keywordnames
\university{\href{https://www.uoc.gr}{University of Crete}} % Your university's name and URL, this is used in the title page and abstract, print it elsewhere with \univname
\department{\href{https://www.biology.uoc.gr}{University of Crete, Department of Biology}} % Your department's name and URL, this is used in the title page and abstract, print it elsewhere with \deptname
\group{\href{https://imbbc.hcmr.gr}{Institute of Marine Biology, Biotechnology and Aquaculture (IMBBC) - HCMR}} % Your research group's name and URL, this is used in the title page, print it elsewhere with \groupname
\faculty{\href{https://www.biology.uoc.gr/en/content/faculty-members}{Faculty, Department of Biology}} % Your faculty's name and URL, this is used in the title page and abstract, print it elsewhere with \facname

\AtBeginDocument{
\hypersetup{pdftitle=\ttitle} % Set the PDF's title to your title
\hypersetup{pdfauthor=\authorname} % Set the PDF's author to your name
\hypersetup{pdfkeywords=\keywordnames} % Set the PDF's keywords to your keywords
}

\begin{document}

\frontmatter % Use roman page numbering style (i, ii, iii, iv...) for the pre-content pages

\pagestyle{plain} % Default to the plain heading style until the thesis style is called for the body content
\hypersetup{linkcolor=teal}
%----------------------------------------------------------------------------------------
%	TITLE PAGE
%----------------------------------------------------------------------------------------

\begin{titlepage}
\begin{center}
\begin{minipage}{4cm}
\begin{flushleft}
    \raggedleft
\includegraphics{figures/uoc-logo-1.jpg} % University/department logo - uncomment to place it
\end{flushleft}
\end{minipage}
\begin{minipage}{6cm}
\begin{flushright}
%\vspace{1cm}
\LARGE \univname
%\vspace{1.0cm} % University name
\end{flushright}
\end{minipage}

\vspace{1cm}
\textsc{\Large Doctoral Thesis}\\[0.5cm] % Thesis type

\HRule \\[0.4cm] % Horizontal line
{\huge \bfseries \ttitle\par}\vspace{0.4cm} % Thesis title
\HRule \\[1.0cm] % Horizontal line
 
\begin{minipage}[t]{0.4\textwidth}
\begin{flushleft} \large
\emph{Author:}\\
\href{http://www.johnsmith.com}{\authorname} % Author name - remove the \href bracket to remove the link
\end{flushleft}
\end{minipage}
\begin{minipage}[t]{0.5\textwidth}
\begin{flushright} \large
\emph{Doctoral Advisory Committee:} \\
\href{https://www.imbb.forth.gr/imbb-people/en/sarris-members/item/2695-dr-panagiotis-f-sarris}{Prof. Panagiotis F. \textsc{Sarris}}\\ % Supervisor name - remove the \href bracket to remove the link  
\href{http://lab42open.hcmr.gr/people/evangelospafilis/}{Dr Evangelos \textsc{Pafilis}}\\ % Supervisor name - remove the \href bracket to remove the link  
\href{https://users.auth.gr/~iantonio/MEMBERSAntoniou.html}{Prof. Ioannis \textsc{Antoniou}}\\ % Supervisor name - remove the \href bracket to remove the link  
\emph{Examination Committee:} \\
\href{https://dinalika.weebly.com}{Prof. Dina \textsc{Lika}}\\ % Supervisor name - remove the \href bracket to remove the link  
\href{https://www.biology.uoc.gr/index.php/en/personnel/faculty-members?view=article&id=246:ladoukakis-emmanuel&catid=27:dep-en-gb}{Prof. Emmanuel \textsc{Ladoukakis}}\\ % Supervisor name - remove the \href bracket to remove the link  
\href{http://pop-gen.eu/wordpress/people}{Prof. Pavlos \textsc{Pavlidis}}\\ % Supervisor name - remove the \href bracket to remove the link  
\href{https://www.medschool.umaryland.edu/profiles/holm-johanna/}{Prof. Johanna \textsc{Holm}}\\ % Supervisor name - remove the \href bracket to remove the link  
\end{flushright}
\end{minipage}\\[1cm]
 

\large \textit{A thesis submitted in fulfillment of the requirements\\ for the degree of \degreename}\\[0.3cm] % University requirement text
\textit{in the}\\[0.4cm]
\deptname\\[0.4cm] 

\textit{and the}\\[0.4cm]
\groupname\\[0.5cm] % Research group name and department name
 
%%
\vspace{1cm}
{\large \today}\\[4cm] % Date
 
\end{center}
\end{titlepage}

%----------------------------------------------------------------------------------------
%	DECLARATION PAGE
%----------------------------------------------------------------------------------------

\begin{declaration}
\addchaptertocentry{\authorshipname} % Add the declaration to the table of contents
\noindent I, \authorname, declare that this thesis titled, \enquote{\ttitle} and the work presented in it are my own. The work of three chapters is 
published in peer reviewed journals with impact factor and I am the first author with shared first authorship, see \hyperref[app:publications]{Publications}. I confirm that:

\begin{itemize} 
    \item This work was done wholly while in candidature for a research degree at this University.
    \item No part of this thesis has previously been submitted for a degree or any other qualification at this University or any other institution.
    \item Where I have consulted the published work of others, this is always clearly attributed.
    \item Where I have quoted from the work of others, the source is always given. With the exception of such quotations, this thesis is entirely my own work.
    \item I have acknowledged all main sources of help.
    \item Where the thesis is based on work done by myself jointly with others, I have made clear exactly what was done by others and what I have contributed myself. The same is clearly indicated in each \hyperref[app:publications]{Publication} with the Contributor Roles Taxonomy (CRediT) system.
    \item This thesis has scored 20\% similarity score using Turnitin account of University of Crete on 26th of June 2024 using the settings "exclude bibliography, exclude quoted material".
\end{itemize}
% 
\noindent Signed:\\
\rule[0.5em]{25em}{0.5pt} % This prints a line for the signature
 
\noindent Date:\\
\rule[0.5em]{25em}{0.5pt} % This prints a line to write the date
\end{declaration}
%
%\cleardoublepage

%----------------------------------------------------------------------------------------
%	QUOTATION PAGE
%----------------------------------------------------------------------------------------

%\vspace*{0.2\textheight}

%\noindent\enquote{\itshape Thanks to my solid academic training, today I can write hundreds of words on virtually any topic without possessing a shred of information, which is how I got a good job in journalism.}\bigbreak

%\hfill Dave Barry

%----------------------------------------------------------------------------------------
%	ABSTRACT PAGE
%----------------------------------------------------------------------------------------

\begin{abstract}
\addchaptertocentry{\abstractname} % Add the abstract to the table of contents

To comprehend ecosystem functioning, it's imperative to discern
the processes occurring in various environments (where) and the organisms
responsible for them (who). Ecosystems form complex associations of taxa,
abiotic parameters and physical characteristics of the materials. For example,
the soil ecosystem holds 10 billion cells in each gram. The texture and the 
chemistry of the soil affects and is affected by the existing taxa across the
tree of life, plants, arthropods, bacteria, fungi etc. Regarding microbes and
metagenomics there is a wealth of data in open omics databases and in the 
literature. Yet, all this information is inhomogeneous and spread in separate 
resources. PREGO, a comprehensive knowledge base, amalgamates
text mining and data integration techniques to extract these what-where-who
associations from scattered scientific literature and omics repositories. It
identifies microorganisms, biological processes, and environmental types,
mapping them to ontology terms. Through text and metagenomics data analysis,
PREGO extracts associations, assigning confidence levels via a scoring scheme.
With 364,508 microbial taxa, 1090 environmental types, 15,091 biological processes,
and 7,971 molecular functions, PREGO aims to aid researchers in experiment
design and interpretation. Additionally, it facilitates exploration of
environment-process-microbe associations.

Furthermore, to understand the current status of ecosystems is important to 
know their past conditions. Historical biodiversity documents are crucial for
long-term data cycles but pose challenges in data curation due to their historical
context. The data rescue process involves document digitization, transcription, information extraction
using text mining tools, and publication to standardized formats. Information
Extraction (IE) tools, evolved over the years, recognize entities in text, aiding in curation.
To this end, this work expanded on IE from a marine historical biodiversity perspective,
orchestrating tools to provide a unified methodology. The classification of
tools enables curators to choose based on their needs. A new tool, DECO, is
introduced, aiming to enhance the data rescue process in biodiversity research.

Studying ecosystems past and current conditions has made apparent the need for 
conservation actions because there are under multiple threats.
An example of ecosystem collapse is the arthropod decline, a globally documented
trend that remains insufficiently handled by the society. In
Crete, a biodiversity hotspot, research on its arthropod fauna dates back centuries.
Here we investigate the conservation status of the Cretan Arthropods using
Preliminary Automated Conservation Assessments (PACA) and the overlap of Cretan
Arthropod distributions with the Natura 2000 protected areas. Moreover we
investigate their endemicity hotspots and propose candidate Key Biodiversity Areas.
In order to perform these analyses, we assembled occurrences of the endemic
Arthropods in Crete located in the collections of the Natural History Museum of
Crete together with literature data. Conservation status assessments using PACA
reveal potential threats to 75\% of endemic arthropods.
Endemic taxa hotspots and candidate Key Biodiversity Areas are mainly in
mountainous regions, often overlapping with Natura 2000 protected areas.
However, human activities impact these areas, and some taxa lack sufficient
protection.

Apart from plants and fauna, microbes significantly influence soil ecosystem
functioning, yet synthesizing microbial biodiversity data remains challenging.
Island biogeography presents a unique opportunity, to advance the understanding
of microbiome diversity in the soil environment.
Crete, as a continental island, has a distinct natural and evolutionary history
with extreme contrasts in vegetation cover, climatic conditions and geology. 
It has been studied for centuries and because of its location in southeastern
Mediterranean is important for climate change studies.
To this end, the first island-wide soil microbiome study conducted on the
island of Crete in 2016, assessed the microbiome diversity drivers and determined that
Crete microbial diversity is driven by pH and soil moisture along elevation
gradient. Together, a team of researchers and citizen scientists collected 435 soil samples
from 72 sites across four distinct ecozones in a single day. The Island Sampling Day Crete 2016
project, further integrates microbial data with soil, spatial and remote sensing data,
to decipher ecosystem function drivers on the island. High-altitude areas harbor diverse
microorganisms, mirroring patterns seen in other faunal groups like arthropods.
Spatial data integration identifies warning signals in pristine and grazing
ecosystems, aiding in identifying biodiversity drivers and evaluating ecosystem threats.

In summary, studying ecosystems ecology with contemporary tools and data poses unique 
opportunities. Cumulative work across the centuries is becoming integrated to the 
digital and holistic representation of ecosystems. Multiple steps are still required 
to this realisation but there is a ground opportunity to unlock novel insights 
through the integration. Conservation and basic research are occurring simultaneously 
because the window of action in the face of ecosystem services collapse is narrow.

%\textgreek{
%Για να κατανοήσουμε τη λειτουργία των οικοσυστημάτων, είναι ανάγκη να διακρίνουμε
%τις διεργασίες που συμβαίνουν σε διάφορα περιβάλλοντα και τους οργανισμούς
%που τις επιτελούν. Τα οικοσυστήματα αποτελούνται από πολύπλοκες αλληλεπιδράσεις τάξων,
%αβιοτικών παραμέτρων και τα φυσικά χαρακτηριστικά των υλικών. Για παράδειγμα,
%το οικοσύστημα του εδάφους περιέχει 10 δισεκατομμύρια κύτταρα σε κάθε γραμμάριο. Η υφή και η
%η χημεία του εδάφους επηρεάζει και επηρεάζεται από τα υπάρχοντα τάξα από όλο το
%δέντρο της ζωής, φυτά, αρθρόποδα, βακτήρια, μύκητες κλπ. Όσον αφορά τα μικρόβια
%υπάρχει πληθώρα δεδομένων σε ανοιχτές βάσεις δεδομένων και στη
%βιβλιογραφία. Ωστόσο, όλες αυτές οι πληροφορίες είναι ανομοιογενείς και αποθηκευμένες σε 
%βάσεις δεδομένων ως σιλό. Το PREGO, αποτελεί μια συγκεντρωτική βάση γνώσης. Μέσω
%τεχνικών εξόρυξης κειμένου και ενσωμάτωσης δεδομένων συγχωνεύει 
%συσχετίσεις οργανισμών, περιβάλλοντος και λειτουργιών από
%από την επιστημονική βιβλιογραφία και βάσεις δεδομένων. 
%Οι μικροοργανισμοί, βιολογικές διεργασίες και περιβαλλοντικοί τύποι αντιστοιχούνται
%με όρους οντολογίας και κάθε συσχέτιση κατηγοριοποιείται με ένα σύστημα αξιολόγησης 
%βαθμών εμπιστοσύνης.
%Με 364.508 μικροβιακά τάξα, 1090 περιβαλλοντικούς τύπους, 15.091 βιολογικές διεργασίες,
%και 7.971 μοριακές λειτουργίες, το PREGO στοχεύει να βοηθήσει τους ερευνητές στο 
%σχεδιασμό και ερμηνεία πειραμάτων αλλά και την διατύπωση νέων υποθέσων εργασίας. Επιπλέον, διευκολύνει την εξερεύνηση
%συσχετίσεις περιβάλλοντος-διαδικασιών-μικροβίων.
%
%Είναι σημαντικό να κατανοήθεί η τρέχουσα κατάσταση των οικοσυστημάτων. Σε αυτό είναι
%απαραίτητη η ιστορική γνώση που υπάρχει από προηγούμενες μελέτες.
%Τα ιστορικά έγγραφα βιοποικιλότητας είναι ζωτικής σημασίας για
%μακροπρόθεσμες αναλύσεις. Όμως η διάσωση και η επιμέλεια δεδομένων ιστορικών δεδομένων είναι δύσκολη λόγω του ιστορικού τους
%πλαισίου. Η διαδικασία διάσωσης δεδομένων περιλαμβάνει ψηφιοποίηση εγγράφων, μεταγραφή, εξαγωγή πληροφοριών
%χρησιμοποιώντας εργαλεία εξόρυξης κειμένου και τη μετέπειτα δημοσίευση σε βάσεις δεδομένων.
%Τα εργαλεία εξόρυξης γνώσης, που έχουν εξελιχθεί ραγδαία τα τελευταία χρόνια,
%αναγνωρίζουν οντότητες στο κείμενο, βοηθώντας στην επιμέλεια.
%Για το σκοπό αυτό, αυτή η εργασία εστίασε στην διάσωση γνώσης θαλάσσιας ιστορικής βιοποικιλότητας και την
%αξιολόγηση των εργαλείων για τον σχεδιασμό μιας ενοποιημένης μεθοδολογίας επιμέλειας. 
%Πηγαίνοντας ένα βήμα παραπέρα, ένα νέο εργαλείο, το DECO, αναπτύχθηκε
%με στόχο την ενίσχυση της διαδικασίας διάσωσης δεδομένων στην έρευνα για τη βιοποικιλότητα.
%
%Η μελέτη των οικοσυστημάτων του παρελθόντος και των σημερινών συνθηκών έχει κάνει εμφανή την ανάγκη για
%δράσεις διατήρησης επειδή βρίσκονται υπό πολλαπλές απειλές.
%Ένα παράδειγμα είναι η κατάρρευση των αρθροπόδων, μια παγκόσμια τεκμηριωμένη
%τάση που έχει αντιμετωπιστεί ανεπαρκώς από την κοινωνία.
%Η Κρήτη είναι ένα hotspot βιοποικιλότητας και η έρευνα για την πανίδα των αρθροπόδων της χρονολογείται από τον 19ο αιώνα.
%Στην παρούσα εργασία αξιολογήθηκε η κατάσταση της διατήρησης των ενδημικών αρθρόποδων της Κρήτης χρησιμοποιώντας
%μια αυτοματοποιημένη αξιολόγηση (PACA) και η επικάλυψη των κατανομών των αρθρόποδων
%στις προστατευόμενες περιοχές Natura 2000. Επιπλέον 
%να διερευνήθηκαν τα hotspot ενδημικότητάς τους και προτάθηκαν υποψήφιες Βασικές Περιοχές Βιοποικιλότητας.
%Για αυτές τις αναλύσεις, συγκεντρώθηκαν παρουσίες του ενδημικών αρθροπόδων της Κρήτης
%που βρίσκονται στις συλλογές του Μουσείου Φυσικής Ιστορίας της
%Κρήτη μαζί με στοιχεία από την εκτενή βιβλιογραφία. Η αξιολόγηση της κατάστασης διατήρησης με την μέθοδο PACA
%κατέταξε το 75\% των ενδημικών αρθροπόδων ως πιθανά απειλούμενα.
%Τα ενδημικά hotspots και οι υποψήφιες βασικές περιοχές βιοποικιλότητας βρίσκονται κυρίως σε
%ορεινές περιοχές, που συχνά επικαλύπτονται με προστατευόμενες περιοχές Natura 2000.
%Ωστόσο, οι ανθρώπινες δραστηριότητες επηρεάζουν αυτές τις περιοχές και ορισμένα αρθρόποδα δεν 
%προστατεύονται επαρκώς.
%
%Εκτός από την χλωρίδα και την πανίδα, τα μικρόβια επηρεάζουν σημαντικά το οικοσύστημα του εδάφους.
%Όμως η αποσαφήνιση των λειτουργιών της μικροβιακής βιοποικιλότητας παραμένει πρόκληση.
%Η νησιωτική βιογεωγραφία προσφέρει μια μοναδική ευκαιρία, για να προωθηθεί η κατανόηση
%της ποικιλότητας των μικροβιωμάτων στο εδαφικό περιβάλλον.
%Η Κρήτη, ως ηπειρωτικό νησί, έχει μια ξεχωριστή φυσική και εξελικτική ιστορία
%με ακραίες αντιθέσεις στη βλάστηση, τις κλιματικές συνθήκες και τη γεωλογία.
%Έχει μελετηθεί για αιώνες και λόγω της θέσης του στα νοτιοανατολικά
%της Μεσογείου είναι σημαντική για τις μελέτες για την κλιματική αλλαγή.
%Για το σκοπό αυτό, η πρώτη ολοκληρωμένη μελέτη μικροβιώματος εδάφους σε νησί διεξήχθη στο
%νησί της Κρήτης το 2016. Αξιολογήθηκαν οι δείκτες ποικιλότητας του μικροβιώματος και ως αποτέλεσμα
%φαίνεται να είναι ότι η μικροβιακή ποικιλότητα της Κρήτης επηρεάζεται από το pH και
%την υγρασία του εδάφους, σε σχέση και με το υψόμετρο. Αυτή η έρευνα συντονίστηκε από μια ομάδα ερευνητών η οποία συνέλεξε 435 δείγματα εδάφους
%από 72 τοποθεσίες σε τέσσερις διακριτές οικολογικές ζώνες της Κρήτης σε μια μέρα. The Island Sampling Day Κρήτη 2016
%ενσωματώνει επίσης, μικροβιακά δεδομένα με δεδομένα εδάφους, χωρικά και τηλεπισκόπησης.
%Βασικός στόχος είναι η ψηφιακή αναπαράσταση του εδάφους της Κρήτης για να αποκρυπτογραφήθούν
%οι λειτουργίες των ποικίλων οικοσυστημάτων του νησιού. Οι περιοχές με μεγάλο υψόμετρο φιλοξενούν μεγάλη βιοποικιλότητα
%μικροοργανισμών, αντικατοπτρίζοντας μοτίβα που παρατηρούνται σε άλλες ομάδες πανίδας όπως τα αρθρόποδα.
%Η ενοποίηση χωρικών δεδομένων ανέδειξε πιθανά σήματα αρνητικής ανθρωπογενούς επίδρασης σε παρθένα οικοσυστήματα αλλά και 
%βοσκοτόπια βοηθώντας στον εντοπισμό των παραγόντων της βιοποικιλότητας και στην αξιολόγηση των απειλών.
%
%Συνοψίζοντας, η μελέτη της οικολογίας των οικοσυστημάτων με σύγχρονα εργαλεία και δεδομένα
%έχει σημαντικές προοπτικές. Η συσσώρευση δεδομένων καταστεί εφικτή την
%ψηφιακή και ολιστική αναπαράσταση των οικοσυστημάτων. Απαιτούνται ακόμη πολλά βήματα
%για την υλοποίηση αυτή όμως υπάρχουν ευκαιρίες για να νέα ερωτήματα και νέες ιδέες
%μέσω της ενσωμάτωσης. Η προστασία, η διατήρηση και η βασική έρευνα πλεόν χρειάζεται να 
%γίνονται ταυτόχρονα επειδή τα περιθώρια δράσης ενόψει της κατάρρευσης των
%οικοσυστημάτων είναι στενά.
%}

\end{abstract}

\textbf{Keywords}

\keywordnames

%----------------------------------------------------------------------------------------
%	ACKNOWLEDGEMENTS
%----------------------------------------------------------------------------------------

\begin{acknowledgements}
\addchaptertocentry{\acknowledgementname} % Add the acknowledgements to the table of contents

First, I am truly thankful to Dr Kotoulas for introducing me to his HCMR colleagues and
made the connection to begin my PhD.
His true interest in deep scientific and philosophical questions is uplifting. He is constantly
joining people together around the globe while valuing most the creative working environment,
the new generations of scientists. Working with openness and without borders is an
important lesson he taught me that made possible this PhD.

I want to thank Dr Pafilis for having me in the PREGO project team, the Island Sampling
Day and teaching me how to make the most out
of previous work and implementing big projects in small steps, one step at a time. You are the only one that 
thoroughly read my work and came with questions. This level of interest was a great lesson for me. Also thank you 
for these amazing hackathons, a remarkable way to collaborate and setting up projects.
Many thanks go to Prof Panos Sarris for sharing his academic values and excellence and helping me in making
decisions. His deep knowledge in molecular microbiology and thirst for debate inspire for
a collaborating and creative environment. 
Prof Ioannis Antoniou thank you for being a concise and sincere mentor. Your amazing ability to focus
on the important questions and hold on to information that's relevant, removing the clutter, 
is remarkable. Really looking forward to being able to collaborating in ecological 
projects through your thermodynamics, network and dynamics expertise!
Thank you all as a committee, a great assembly, that guided me and simultaneously 
allowed me to develop my interests and creativity as a scientist.

\begin{figure}[hbt!] 
    \centering\includegraphics[width=\columnwidth]{passions.png}
    \caption{Four and a half years of beautiful friendships and expeditions}
    \label{fig:passions}
\end{figure}

I thank the super EMODnet team of HCMR, Dr Gerobasileiou, Dr Arvanitidis, Ms Mavraki and
Ms Georgia Sarafidou. What a great project and nice collaboration about Historical 
biodiversity literature.

Special thanks go to the Arthropods Lab of the Natural History Museum, to
Dr Apostolos Trichas and Ms Giannis Bolanakis for the long hours working and discussing together.
Thank you for contacting and sharing your 
precious data and exchanging movies, music and great scientific texts! Working with you is a joy!

My gratitude and respect go to all the people of Island Sampling Day project and especially
Lynn Schriml and Johanna Holm. I have learned and continue to learn so much from 
you. Working together in a hobby project and giving our care means a lot to me and your
generosity is so humbling.

Haris Zafeiropoulos, my friend, I look forward to many collaborations in the future. Seeing you code and work
is inspiring! Most of all, feeling your positive spirit in people has made me more
open that I thought I could be, life is more fun this way!

Special thanks go to the Heinz Troll photography crew! Thank you Heinz and Stef for teaching me 
how to travel the world, how to be a professional and also for having me in your photoshoots and 
funding me. You are the unofficial sponsor of this PhD!

During this PhD I realised many things, the most important being trying to be around the people I love no matter the job.
Giota, Maria, Kaloust thank you so much for the unconditional love and acceptance of people as they are.
My friends Dimitri, Kosta, Stefane thank you for the amazing gatherings in the mountains and discussions while 
hanging in the void. 

To CAT, my Christina, you are the music, the dance, the laughter, the truth, thank you showing me 
how to enrich life and love. Looking forward for our explorations together and sharing our passions.

\end{acknowledgements}

%----------------------------------------------------------------------------------------
%	PREFACE
%----------------------------------------------------------------------------------------

\addcontentsline{toc}{chapter}{Preface}
\chapter*{Preface}

Being an bioinformatician means spending a most of my time in the office. This is a 
stark contrast in what I do in other aspects of my life. Canyoning, Climbing, Caving 
outdoor activities in heights using ropes is truly what brings me peace. Yet the 
amazing questions that come from ecology and the origin of life require theoretical 
knowledge, mathematics and programming. One important realisation
during my PhD was that I felt sheer joy when combining these two extremes. This was
of course during the microbial sampling of Crete and of the deepest cave of Greece, Gourgouthakas
and in capturing raptor birds in the islands of the Aegean sea (Figure \ref{fig:passions}). Bringing data from 
extreme environments and then, after the incredible bench work of colleagues, analyse 
these data made my heart sing.

Another aspect of being a bioinformatician is working on different projects with 
multiple people. This realisation was a revelation to me and this PhD is the outcome 
of deep collaboration with four different groups of people and many new friendships. 
The chapters of this PhD cover many disciplines, similar to most bioinformatic PhDs,
yet they all converge to integrating multiple types of data to investigate the multifaceted
and complex nature and functioning of ecosystems. The main focus is the soil ecosystems and 
their microbiome, with the island of Crete being the field of application.

Academic writing and publishing is important, albeit difficult, and I have learned a
lot during the PhD. Three chapters are published in journals as
mentioned in each chapter. In these papers I share first authorship, meaning that 
both first authors were indispensable for the work. For the complete list of my published work see \hyperref[cv:my_refs]{here}.
The first and last chapters will be 2 back to back submissions. I am proud for this level of collaboration
with so diverse teams because it helped me develop teamwork thinking and trust. These skills are important 
for interdisciplinary work, like this PhD, and in modern science in general.

The chapters are :

\begin{enumerate}
    \item Island Sampling Day Crete project in Chapter \ref{cha:isd-crete-soil}
    \item PREGO project in Chapter \ref{cha:prego}
    \item Gathering soil data to establish a Cretan earth system in Chapter \ref{cha:crete-idea}
    \item DECO workflow in Chapter \ref{cha:deco}
    \item Endemic arthropods of Crete island in Chapter \ref{cha:arthropods}
    \item The data integration and the soil microbiome diversity in \ref{cha:crete-soil}
\end{enumerate}

The Chapter \ref{cha:isd-crete-soil} is about the Island Sampling Day (ISD), a large scale study for the microbial diversity of the
topsoil of Crete. It the first large scale microbiome sampling in Crete and I analysed
all this wealth of data to describe the hidden biodiversity. ISD is a 16s rRNA amplicon study designed to put standards into action. 
ISD is a citizen science standardisation effort, voluntarily designed and
operated by the participants of the 18th Genome Standards Consortium
Workshop and the Hellenic Centre for Marine Research that took place in June 2016.
This project adds a new chapter to the study of biodiversity in Crete,
the soil microbial diversity. But these data and metadata alone are 
insufficient to understand ecosystem function and even the drivers 
of microbial community structures. How to enrich these valuable data 
with existing available information? 

Regarding the microbes there a wealth of information hidden in text and metagenomic resources.
To address this issue, I co-developed PREGO, a global data and text mining metagenomic resourse, designed to assist microbiologists
to enrich their knowledge about microbes, the processes they perform and the environments 
they appear. We developed PREGO's knowledge base with both a web interface and an Application Programming Interface.
PREGO is a meta platform, which is based upon open data and standards of metagenomics. 
It brings the wealth of taxonomic, environmental and biological functions of 
microbial data and literature all in one place. Multiple platforms are harvested and the data 
and metadata are harmonised using text mining. In addition, academic texts contain 
valuable information that is missing from databases and text mining solutions unearth 
this information. Hence, the PREGO approach allowed to bring the heterogeneous
data together in a single knowledge base. 

How to bring all this information from literature, biodiversity and maps in one place? Focusing on Crete
provided a unique opportunity to use it as a model for the digital representation of ecosystems. 
As an island, it is confined in space, and also has been studied for thousands of years.
Being able to collect all the these data in one place allows to perform analyses on the soil ecosystems of Crete.
Hence, the chapter \ref{cha:crete-idea} is about the total environment of Crete.
In this chapter I address the questions of 

\begin{itemize}
    \item How many documents are containing valuable information?
    \item What kind of samplings have been carried out?
    \item Which spatial data about Crete are available? 
\end{itemize}

Bringing all this information together poses unique challenges of different types of data, standards and code. 
But with the establishment of this digital description of the open knowledge about Crete allows the generation of novel questions and hypotheses.
Additionally, a literature and biodiversity data summary is 
provided to set the basis for future integrative studies and for the documentation of the current status of 
research regarding the island.

Yet, another biodiversity pillar of the Cretan ecosystems is arthropods which 
interact with microbiomes. Based on a novel and dataset
that contained historical data and specimens of the Natural History Museum of Crete
we assessed the conservation status of the endemic arthropods of Crete. Doing a assessment of 
conservation of the endemic arthropods was an obvious priority. Because to have the ability to 
study them, first they have to be conserved from their documented collapse. Hence, this work contained 
work from the 19th century along with the 4 decades long samplings of the NHMC. 
The overlap of Cretan Arthropod distributions with the Natura 2000 protected areas
were investigated along with their endemicity hotspots and proposed candidate Key Biodiversity Areas.
This work is serves as an example to further highlight the importance of historical data along with contemporary.

Upon realising the struggles and the benefits of rescuing historical biodiversity data, I
developed a workflow to accelerate the process. Because, apart from the contemporary data and metagenomics, important biodiversity data is 
included in historical biodiversity documents. This information from 
previous centuries reports, books and expeditions notes is an important glimpse into
the past. These data due to their legacy format and other complexities, for example 
multiple languages, different taxa names and approximate coordinates, require 
enormous effort to rescue in databases. Curators are indispensable in this process 
yet there are multiple tools that assist them. Using text mining and other tools 
for Object Character Recognition we demonstrated a workflow to rescue such documents.
This workflow is named DECO and is a software container that can handle multiple 
documents and also combines multiple tools for the steps of the data rescue process.

The soil microbiome is core of the soil ecosystems, and using the aforementioned data I build the Crete Data Cube.
In the final chapter, \ref{cha:crete-soil} of this PhD, I integrate the soil biodiversity, it's functional potential and 
the Crete Data Cube to decipher the soil ecosystems of concern, vulnerable soils and one health related taxa. The results 
of this analysis provide a new perspective of some well known ecosystems of Crete.
But also sets an example of the importance of integration of prior knowledge and data to address current challenges.

There is so much production of papers and journals that is difficult to decide where to publish. 
So except from the common flagship journals, I discovered the work that shaped my 
PhD in themed issues of Philosophical Transactions of the Royal Society B.
A journal that really stands out and it was established on 1887! This was an
important realisation during the PhD; there are journals launched only 10 years
ago with incredible high impact factor but there are journals in ecology
with rich history and groundbreaking articles.

Most notably:

\begin{itemize}
    \item Conceptual challenges in microbial community ecology \href{https://doi.org/10.1098/rstb.2019.0241}{Volume 375, Issue 1798}
    \item Ecological complexity and the biosphere: the next 30 years \href{https://doi.org/10.1098/rstb.2021.0376}{Volume 377, Issue 1857}
    \item Integrative research perspectives on marine conservation \href{https://doi.org/10.1098/rstb.2019.0444}{Volume 375, Issue 1814}
\end{itemize}

Having these issues as guides covered all the different fields that I am interested 
in working on. Microbial ecology is a great field, here I focused on soil mostly, that
combines all aspects of numerical ecology. There are many things that scientists 
haven't deciphered yet, like how the above ground biodiversity influences the below
ground biodiversity. Ecological complexity is what inspired me to continue to a 
master's degree. Still, I am looking forward to apply and learn the amazing 
theoretical mathematical models. And lastly is the integration of multiple methods 
and data to describe an ecosystem. This is what I worked mostly on my PhD and 
harvesting different types of data, i.e sequences, spatial data, occurrence data, 
text etc, has it's unique challenges. Each type of data requires different tools and has 
different objects to handle them. Furthermore, types of data come from different 
fields with different scientific communities and standards. Incorporating and even 
discovering these data is time consuming and mind bending. Yet, utilising diverse 
published data to analyse an ecosystem can lead to novel insights. 

The projects that are presented in the following chapters are about 
bringing together contrasts. Starting with biodiversity, we investigated taxa that 
have been studied for centuries, e.g coleoptera, as well as new undiscovered ones, the soil microbiome of 
Crete. These works span from basic research to conservation analyses. 
Spatial analysis brought together the micro scale of microbes with the macro
scale of the island of Crete.
Microbes also unlocked some processes and functions they facilitate in their environments, thereby assisting in
categorising them and finding possible one health concerns.
Text mining techniques were used to incorporate historical texts and contemporary literature.
Lastly, in the Island Sampling Day project we covered
the whole spectrum of the analysis from large scale field sampling to informatics.

All this work was made possible because of the adoption of open data and standards
and open source software. Following these values, all the data and software produced
in this work are documented and deposited in platforms with open source licences. 
Working also with the Genome Standards Consortium was a great experience to witness
great scientists devoting a lot of time to design and promote standards across the 
community. This openness is both inspiring and humbling and I hope to contribute and follow 
their lead.

\addcontentsline{toc}{chapter}{Funding}
\chapter*{Funding}

The first two years of this PhD were about the PREGO project, discussed in Chapter \ref{cha:prego}. My work in this 
project was supported by the Hellenic Foundation for Research and
Innovation (HFRI) and the General Secretariat for Research and Innovation (GSRI),
under Grant No. 241, PREGO project.

The last two years were supported by the 3rd HFRI. Scholarships for PHD
Candidates (no. 5726) leading to the Chapters \ref{cha:isd-crete-soil}, \ref{cha:arthropods} and \ref{cha:crete-soil}.

   \begin{figure}[h]
      \centering
      \includegraphics[width=0.30\columnwidth]{figures/ELIDEK_Logo_EN_and_GSRI_logo_EN.png}
      \caption[HFRI and GSRI funding]{
          This PhD was supported by HFRI - GSRI grands. 
      }
      \label{fig:hfri_logo}
   \end{figure}


%----------------------------------------------------------------------------------------
%	LIST OF CONTENTS/FIGURES/TABLES PAGES
%----------------------------------------------------------------------------------------

\tableofcontents % Prints the main table of contents

\listoffigures % Prints the list of figures

\listoftables % Prints the list of tables

%----------------------------------------------------------------------------------------
%	ABBREVIATIONS
%----------------------------------------------------------------------------------------

\begin{abbreviations}{ll} % Include a list of abbreviations (a table of two columns)

\textbf{API} & \textbf{A}pplication \textbf{P}rogramming \textbf{I}nterface\\
\textbf{ASV} & \textbf{A}mplicon \textbf{S}equencing \textbf{V}ariants\\
\textbf{BHL} & \textbf{B}iodiversity \textbf{H}eritage \textbf{L}ibrary\\
\textbf{BOM} & \textbf{B}iodiversity \textbf{O}bservations \textbf{M}iner\\
\textbf{BODC} & \textbf{B}ritish \textbf{O}ceanographic \textbf{D}ata \textbf{C}entre\\
\textbf{CLI} & \textbf{C}ommand \textbf{L}ine \textbf{I}nterface\\
\textbf{CTI} & \textbf{C}ommunity \textbf{T}emperature \textbf{I}ndex\\
\textbf{DECO} & Bio\textbf{D}iv\textbf{E}rsity Data \textbf{C}uration w\textbf{O}rkflow\\
\textbf{DPI} & \textbf{D}ots \textbf{P}er \textbf{I}nch\\
\textbf{DwC-A} & \textbf{D}arwin \textbf{C}ore \textbf{A}rchive\\
\textbf{EM} & \textbf{E}ntity \textbf{M}apping\\
\textbf{EMODnet} & \textbf{E}uropean \textbf{M}arine \textbf{O}bservation and \textbf{D}ata \textbf{Net}work\\
\textbf{ENVO} & \textbf{Env}ironment \textbf{O}ntology\\
\textbf{EOL} & \textbf{E}ncyclopedia \textbf{O}f \textbf{L}ife\\
\textbf{FAIR} & \textbf{F}indable \textbf{A}ccessible \textbf{I}nteroperable \textbf{R}eusable\\
\textbf{GBIF} & \textbf{G}lobal \textbf{B}iodiversity \textbf{I}nformation \textbf{F}acility\\
\textbf{GLOBI} & \textbf{GLO}bal \textbf{B}iotic \textbf{I}nteractions\\
\textbf{GNA} & \textbf{G}lobal \textbf{N}ames \textbf{A}rchitecture\\
\textbf{GNRD} & \textbf{G}lobal \textbf{N}ames \textbf{R}ecognition and \textbf{D}iscovery\\
\textbf{GUI} & \textbf{G}raphical \textbf{U}ser \textbf{I}nterface\\
\textbf{HTML} & \textbf{H}yper\textbf{T}ext \textbf{M}arkup \textbf{L}anguage\\
\textbf{HTTP} & \textbf{H}yper\textbf{T}ext \textbf{T}ransfer \textbf{P}rotocol\\
\textbf{ICZN} & \textbf{I}nternational \textbf{C}ommission on \textbf{Z}oological \textbf{N}omenclature\\
\textbf{ID} & \textbf{I}dentifier\\
\textbf{IE} & \textbf{I}nformation \textbf{E}xtraction\\
\textbf{IPT} & \textbf{I}ntegrated \textbf{P}ublishing \textbf{T}oolkit\\
\textbf{IR} & \textbf{I}nformation \textbf{R}etrieval\\
\textbf{JSON} & \textbf{J}ava\textbf{S}cript \textbf{O}bject \textbf{N}otation\\
\textbf{LSID} & \textbf{L}ife \textbf{S}cience \textbf{I}dentifier\\
\textbf{MedOBIS} & \textbf{Med}iterranean node of the \textbf{O}cean \textbf{B}iodiversity \textbf{I}nformation \textbf{S}ystem\\
\textbf{NCBI} & \textbf{N}ational \textbf{C}enter for \textbf{B}iotechnology \textbf{I}nformation\\
\textbf{NER} & \textbf{N}amed \textbf{E}ntity \textbf{R}ecognition\\
\textbf{NERC} & \textbf{N}atural \textbf{E}nvironment \textbf{R}esearch \textbf{C}ouncil\\
\textbf{NLP} & \textbf{N}atural \textbf{L}anguage \textbf{P}rocess\\
\textbf{OBIS} & \textbf{O}cean \textbf{B}iodiversity \textbf{I}nformation \textbf{S}ystem\\
\textbf{OCR} & \textbf{O}ptical \textbf{C}haracter \textbf{R}ecognition\\
\textbf{OS} & \textbf{O}perating \textbf{S}ystem\\
\textbf{OTU} & \textbf{O}perational \textbf{T}axonomic \textbf{U}nit\\
\textbf{PDF} & \textbf{P}ortable \textbf{D}ocument \textbf{F}ormat\\
\textbf{PNG} & \textbf{P}ortable \textbf{N}etwork \textbf{G}raphics\\
\textbf{REST} & \textbf{RE}presentational \textbf{S}tate \textbf{T}ransfer\\
\textbf{SI} & \textbf{I}nternational \textbf{S}ystem of \textbf{U}nits\\
\textbf{URL} & \textbf{U}niform \textbf{R}esource \textbf{L}ocator\\
\textbf{WoRMS} & \textbf{Wo}rld \textbf{R}egister of \textbf{M}arine \textbf{S}pecies\\
\end{abbreviations}

%----------------------------------------------------------------------------------------
%	PHYSICAL CONSTANTS/OTHER DEFINITIONS
%----------------------------------------------------------------------------------------

%\begin{constants}{lr@{${}={}$}l} % The list of physical constants is a three column table
%
%% The \SI{}{} command is provided by the siunitx package, see its documentation for instructions on how to use it
%
%Speed of Light & $c_{0}$ & \SI{2.99792458e8}{\meter\per\second} (exact)\\
%%Constant Name & $Symbol$ & $Constant Value$ with units\\
%
%\end{constants}
%
%%----------------------------------------------------------------------------------------
%	SYMBOLS
%----------------------------------------------------------------------------------------

%\begin{symbols}{lll} % Include a list of Symbols (a three column table)
%
%$a$ & distance & \si{\meter} \\
%$P$ & power & \si{\watt} (\si{\joule\per\second}) \\
%%Symbol & Name & Unit \\
%
%\addlinespace % Gap to separate the Roman symbols from the Greek
%
%$\omega$ & angular frequency & \si{\radian} \\
%
%\end{symbols}

%----------------------------------------------------------------------------------------
%	DEDICATION
%----------------------------------------------------------------------------------------

%\dedicatory{For/Dedicated to/To my\ldots} 

%----------------------------------------------------------------------------------------
%	THESIS CONTENT - CHAPTERS
%----------------------------------------------------------------------------------------

\mainmatter % Begin numeric (1,2,3...) page numbering

\pagestyle{thesis} % Return the page headers back to the "thesis" style

% Include the chapters of the thesis as separate files from the Chapters folder
% Uncomment the lines as you write the chapters

\include{Chapters/chapter-intro}
\include{Chapters/chapter-isd-crete}
\include{Chapters/chapter-prego}
\include{Chapters/chapter-crete-idea}
\include{Chapters/chapter-deco}
\include{Chapters/chapter-arthropods}
\include{Chapters/chapter-crete-microbiome}
\include{Chapters/chapter-conclusions}

%----------------------------------------------------------------------------------------
%	THESIS CONTENT - APPENDICES
%----------------------------------------------------------------------------------------
\appendix % Cue to tell LaTeX that the following "chapters" are Appendices

% Include the appendices of the thesis as separate files from the Appendices folder
% Uncomment the lines as you write the Appendices

\include{Appendices/AppendixA}
\include{Appendices/AppendixB}
%\include{Appendices/AppendixC}

%----------------------------------------------------------------------------------------
%	BIBLIOGRAPHY
%----------------------------------------------------------------------------------------

% Add ref 
\printbibliography[heading=bibintoc]

%----------------------------------------------------------------------------------------

\end{document}  
