% --------------------------------------------------
% 
% This chapter is for Island Samplind Day, Crete
% 
% --------------------------------------------------


\chapter{Island Sampling Day, the case of Crete}
\label{cha:isd-crete-soil}

%\textbf{Citation:} \\ 

\section{Introduction}\label{intro_isd}

Microbes influence global ecosystem functions \parencite{falkowski2008microbial}
and are ubiquitous \parencite{delgado2016microbial}. The terrestrial biomass of bacteria is
70 Gt C, second to plants with a sixfold difference \parencite{bar2018biomass}. In subsurface systems, bacteria and
archaea are the most abundant, having 90\% of the biomass. These findings of microbial
ecology are based on amplicon 16s rRNA studies that have flourished since the
2010s.
Amplicon rRNA sequencing provides a collection of sequence reads per sample
based on a marker gene. For bacteria the 16s rRNA gene is amplified with PCR 
using specific sets of primers.
This amplification step occurs after the DNA extraction and it's resulting DNA products
are are sequenced.
The endeavor to understand the microbial world faces multiple challenges
across the scientific workflow, from sampling to ecological analyses \parencite{Lee2012}.

Errors propagate starting with the sampling. There are contaminations from the
people in the field and in the lab for the DNA extraction \parencite{EISENHOFER2019105}. 
Errors also arise from PCR amplification and from sequencing \parencite{Schloss2011, Schimer2015}.
Approximation and computation errors are introduced from algorithms that cluster, measure similarities between
sequences. Semantic errors occur because of reductionist approaches and/or oversimplification
of the microbial communities. Furthermore, microbiome matrix data are compositional, 
meaning that the abundances of the reads are not the real representations of the natural populations \parencite{Gloor2017}.
These limitations are important to be acknowledged in order to avoid overarching conclusions,
especially in complex systems such as soil.

The ecological interpretation of the sequenced reads to taxonomic information
requires their preprocessing to remove errors propagated across steps, e.g PCR and sequencing.
To do so there are two approaches currently in use, 
the clustering method and the denoising method. With clustering, reads are 
grouped together and a best representing sequence is produced for each 
cluster, i.e. the Operational Taxonomic Unit (OTU). 
Clustering needs an arbitrary threshold, usually 97\%, and a centroid selection
that is dependant on the order of input \parencite{zafeiropoulos2020pema}. 
This approach makes the OTUs 
from different runs, i.e executions of the algorithm and/or different studies,
incomparable and irreproducible.
Denoising methods produce Amplicon Sequence Variants (ASVs),
which are considered real biological sequences. 
ASVs can be used for comparison and require less computational power. 
Yet, because of the vast number of microvariant sequences from ASV algorithms,
the subsequent analysis is more complex \parencite{Glassman2018}.
Currently, many studies propose the use of Amplicon Sequence Variants
instead of OTUs \parencite{Callahan2017}.
The influence of the different methods to subsequent ecological analyses has
little impact \parencite{Glassman2018}.

Accumulated research on taxonomic analyses of bacteria and archaea has
distinguished them as major drivers of soil ecosystems functionality \parencite{Delgado-Baquerizo-atlas}.
In soil ecosystems, the pillars of human activities and ecosystem services, 
there is a positive correlation between soil microbial diversity and their multifunctionality \parencite{Delgado-Baquerizo2020}.
The community structure defies their macroscopic functionality \parencite{Bahram2018}.
Soils are complex environments that are characterised by chemical, physical and biological properties
that form interdependent associations with feedback loops by their environment. 
Global soil microbiome studies have been employed to decipher soil microbiome
compositions \parencite{thompson2017a-communal, Delgado-Baquerizo-atlas, Labouyrie2023},
functions \parencite{Bahram2018} and biogeography \parencite{Martiny2006, guerra2020Blind}.
The profound diversity of soils is clearly stated in these works, yet there are
significant blind spots \parencite{guerra2020Blind}.
When considering the samples per area density, there global studies are rather sparse.
National wide studies like the one by \textcite{Karimi2020} show the
complexity of soil bacteria and the need to design 
denser samplings and use isolated systems \parencite{Dini-Andreote2021}.

Isolated systems are important to avoid co-founding effects and to reduce complexity.
Islands are considered nature's labs and are used in biogeography extensively due to their smaller scale and isolation \parencite{Whittaker2017}. 
Island microbiomes from soil \parencite{Li2020} and mycorrhizal fungi \parencite{Delavaux2021} studies
show the benefits of using islands as models.
This was the case for the Island Sampling Day (ISD) Crete project
of the Genomic Standards Consortium \parencite{Field2011}
during the 18th workshop in June 2016 in Crete island, Greece. The goal of this sampling was to "put standards into action"
in a soil microbiome survey with a dense sampling, 0.017 samples per km\textsuperscript{2}.
Hence, ISD Crete is a large scale study in the confined space of the island of Crete which 
is considered a miniature continent \parencite{Vogiatzakis2008_crete}.

The Island Sampling Day draws inspiration from the Ocean Sampling Day \parencite{kopf_2015} that is a
consortium that facilitates yearly samplings in stations from all around the world.
This has great applications in monitoring and comprehending the complexity of the ocean microbiome and itâ€™s drivers.
The Island Sampling Day has been organised in two islands, Mo'orea in the French Polynesia and Crete in Greece.
In the ISD Crete project, 26 people grouped in ten teams went across the island
in a single day to sample as much diversity as possible from different ecosystems of Crete. 
Samples were collected in a single day in order to limit the variations of environmental
factors that may influence the abundance and diversity of microbes such as season,
temperature and humidity.
For each sampling site, teams selected two specific sub-sites that were
at least 3 meters from the edge of the road and 0.6 m from the base of
the identified plant. Flora located at the sampling site was
identified and photographed.
At each sub site, soil was sampled (3 replicates, collected one 3cm apart). 

The University of Maryland Soil Lab (Dr. Stephanie Yarwood) received 288 soil cores
144 for DNA extraction and 144 for physico-chemical measurements. The shipment was on dry ice,
under the USDA permit and a Material Transfer Agreement with HCMR.
The water content, Total Organic Carbon and Nitrogen weights were measured using LECO CN628 analysis platform.
DNA extraction of 144 soil samples used the protocols of the \href{https://www.protocols.io/view/emp-16s-illumina-amplicon-protocol-cpisvkee}{Earth Microbiome Project} (EMP)
with the the MoBio DNA extraction kit. Three of the samples, collected in sand, had no detectable DNA
following extractions.
The PCR-based protocol used is designed for the V3-V4 region of the 16S rRNA gene.
The primers used were 338F, Forward: 5'-ACTCCTACGGGAGGCAGCAG-3' and  806R, Reverse: 5'-GGACTACHVGGGTWTCTAAT-3'.
The sequencing of 16s rRNA amplicons were carried out on an Illumina HiSeq2500 platform
at the Institute for Genome Sciences, Genomics Resource Center (Maryland Genomics) at the University of Maryland School of Medicine.

The Genomic Standards Consortium uploaded the metadata and the sequences to ENA
database under the project \href{https://www.ebi.ac.uk/ena/browser/view/PRJEB21776}{PRJEB21776} immediately upon realise. 
This is an act of unrestricted use of genomic data which exemplifies the importance of 
having public data even prior publications \parencite{kopf_2015}.

In this study, the ISD data and metadata were downloaded from ENA database analysed to decipher drivers of 
biodiversity and community structure of the top soil microbiome of Crete. Diversity, 
ordination and correlations of metadata were carried out. Following the examples of 
transparency of the GSC, all code is reproducible and documented as mentioned in FAIR principles \parencite{wilkinson2016the-fair}.
This is the first analysis 
of this scale of the Cretan soil microbiome, henceforth adding a new chapter of 
biodiversity in Crete, the microbiome.

\section{Materials and Methods}\label{isd_methods}

The work is purely computational; scripts of the analysis cover the following tasks:
\begin{itemize}
    \item Search ENA for samples in the island of Crete
    \item Get ISD metadata and sequences
    \item HPC jobs and parameter files
    \item Filtering, clustering/denoising and taxonomic assignment
    \item Biodiversity and ecological analysis 
    \item Visualisation
\end{itemize}

The bioinformatic workflow can be summarized to : GET, INFER and ANALYZE as shown in Figure \ref{fig:isd_workflow_taxonomy}.

\begin{figure}[h]
      \centering
      \includegraphics[width=\textwidth,keepaspectratio]{figures/isd_crete_flowchart.png}
      \caption[Reproducible workflow of ISD analysis]{Three steps of the ISD bioinformatics workflow, reproducible by design}
      \label{fig:isd_workflow_taxonomy}
   \end{figure}
   
\subsection{Data retrieval}\label{isd_get}

This study uses the ISD Crete data that have been deposited
in the European Nucleotide Archive (ENA) at EMBL-EBI under accession number PRJEB21776.
The raw sequences (fastq files) and metadata (xml files) were downloaded with custom scripts using the ENA API \parencite{Yuan2023}.
The sample identifiers of the ENA project were retrieved using the \textit{curl} command from Bash. Using these 
sample ids with \textit{wget} command the fastq sequences were saved on the HPC facility. Regarding the metadata, 
a python script was developed to retrieve a xml file for each sample. All xml files
were transformed to tab delimited files with a second custom python script.

\begin{figure}[htp!] 
    \centering\includegraphics[width=\columnwidth]{isd_map_fig1-small}
    \caption{Crete ISD sampling. A. The routes of the single day event sampling. B. Beta diversity differences are represented with color.}
    \label{fig:isd_crete_sampling}
\end{figure}

\subsection{Taxonomic inference}\label{tax_inference}
Amplicon Sequence Variants were inferred using DADA2 \parencite{Callahan2016} for 
filtering, denoising and chimeric reads removal. Normalisation of the reads
across samples was implemented with the SRS R package \parencite{Beule2020}. Samples
with less than 10000 reads were removed. ASVs were assigned to taxonomy using 
DADA2 with Silva 138 taxonomy \parencite{quast_silva_2013}.
PEMA was used for OTU inference based on VSEARCH \parencite{zafeiropoulos2020pema} with Silva 132.

\subsection{Biodiversity analyses}\label{biodiversity}

For the numerical ecology analyses,e.g diversity indices, NMDS, PERMANOVA, were calculated
with the vegan R package \parencite{oksanen2024vegan}.
For PCoA ordination we used the ape R packege \parencite{Paradis2004} and UMAP python library \parencite{mcinnes2018umap-software}.
U-CIE R package was used for coloring 3 dimensional data \parencite{Koutrouli2022} to 
visualise the $\beta$ diversity differences of samples on the map.
Visualisation was implemented with ggplot2 \parencite{wickham_ggplot2_2016} and pheatmap \parencite{Kolde2019}.

\subsection{Environment and Code}

Computations were performed on HPC infrastructure of HCMR \parencite{zafeiropoulos_0s_2021}.
The operating system is Debian 4.19. Conda environments were utilised
with Python 3.11.4, R version 4.3.2 \parencite{rcoreteam}.
Additionally GNU bash 5.0.3 and GNU Awk 4.2.1 were used for the streamlined workflow 
and for the reads statistics, respectively.

The documentation and scripts developed for this study are available in
\href{https://github.com/GenomicsStandardsConsortium/ISD}{ISD Crete soil microbiome github repository}.
This repository contains all the necessary scripts for the data retrieval,
filtering and ASV inference, taxonomy assignment, data integration of spatial data, 
functional annotation and the subsequent analyses and visualisation.
Code is structured to be reproducible and interoperable as described in Figure \ref{fig:isd_workflow_taxonomy}.

\section{Results}\label{isd_results}

\subsection{Data description}
Data and its' associated metadata were retrieved from ENA under study
accession number PRJEB21776.
Even though ISD Crete project has 144 sampling locations, 140 samples were
downloaded from ENA because 3 samples didn't yield any DNA and 1 wasn't uploaded to ENA due 
to some errors.
The metadata retrieved from the project were rich as described in the project. 
Each sample contained information about the location such as the source material identifiers as an internal
id, the id of the route, the place name, latitude and longitude, elevation, vegetation zone,
environment (biome), environment (feature), environment (material) among others.
Sample metadata were also about the methodology, i.e the amount or size of sample collected,
pcr primers, sequencing method, target gene, sample collection device or method,
sample volume or weight for DNA extraction and the DNA concentration.
Metadata about soil physical and chemical characteristics were also included, 
such as total nitrogen, total nitrogen method, total organic C method,
total organic carbon, water content and water content method. The pH measurements
were not in the metadata because of an error during upload and thus are not 
included in this analysis.

The fastq files were downloaded to the HPC and they contained 60 million reads (approximately 15 GB in storage),
averaging 250-500 K reads/sample, all derived from the Illumina HiSeq2500 sequencing platform.
The read quality filtering and merging steps led to an almost 50\% reducing in sequence size, Table \ref{tab:metrics}. 
On average, each sample contained about 355,326 reads, though this varied from
2,437 to 525,144.

\begin{table}[h!]
\centering
\caption{Summary of sequencing and quality filtering.}
\begin{tabular}{@{}ll@{}}
\textbf{Metric} & \textbf{Value}     \\
Samples         & 140                \\
Reads           & $60.6 \times 10^6$ \\
Ns              & $76.000$           \\
Filtered        & $50.1 \times 10^6$ \\
denoisedF       & $45.9 \times 10^6$ \\
denoisedR       & $48.5 \times 10^6$ \\
merged          & $35.0 \times 10^6$ \\
nonchim         & $33.3 \times 10^6$ \\
\end{tabular}
\label{tab:metrics}
\end{table}

Reads shorter than 445 bp or longer than 515 bp were fewer than
15,000, see Figure \ref{fig:isd_srs-curve_samples}.
Reads with unknown bases, Ns, were removed.
From the 140 samples retrieved from ENA, two samples were 
neglected because they had fewer than 10000 reads.
Thus, 138 samples in total were included in the subsequent analysis.
The reads of the remaining samples were normalised with SRS package at 10000 reads \parencite{Beule2020}
   
   \begin{figure}[hbt!]
      \centering
      \includegraphics[width=\textwidth,keepaspectratio]{figures/isd_crete_srs_curve.jpeg}
      \caption[SRS curve]{SRS curve of the samples. }
      \label{fig:isd_srs-curve_samples}
   \end{figure}

\subsection{Inference and taxonomy}\label{inference_taxonomy}

In summary, DADA2 resulted in 216,360 ASVs (2,704 unique
taxa; 1123 ASVs at species level and 1059 at genus level) and
PEMA (VSEARCH) in 13,285 OTUs; 2319 at family level and 1920 at genus level, Table \ref{table:asv_taxonomy}.
There is a two orders of magnitude difference in the output of different methods. 
About 98\% of ASVs occur in 2 samples or less as shown in Figure \ref{fig:isd_fig2_taxonomy}A.
Hence, finding the prevalent ASVs is no trivial in soils of Crete.

\begin{table}[]
    \caption{Taxonomic depth of ASVs and the unique number of taxa of each level.}%
\begin{tabular}{@{}lllll@{}}
    \textbf{classification depth} & \textbf{Total ASV}    & \textbf{Total (ASV) taxa} & \textbf{Total OTUs} & \textbf{Total (OTU) taxa}\\
Kingdom              & 1974         & 2                & 284        & 2               \\
Phylum               & 4034         & 33               & 121        & 15              \\
Order                & 38517        & 193              & 1224       & 135             \\
Class                & 24157        & 83               & 978        & 62              \\
Family               & 71355        & 287              & 2319       & 218             \\
Genus                & 90137        & 1166             & 1920       & 582             \\
Species              & 9120         & 1338             & 44         & 43              \\
Total                & $\sim$239000 & 3102             & 6890       & 1057            
\end{tabular}
\label{table:asv_taxonomy}
\end{table}

There are 25 specialist taxa (present in samples < 10 and mean relative
abundance > 0.003) and 146 generalist taxa (samples > 120), Figure \ref{fig:isd_fig2_taxonomy}.
The representative Phyla, i.e the Phyla with higher than 5\% presence in all samples, are:
Actinobacteriota, Proteobacteria, Chloroflexi, Acidobacteriota,
Bacterodota and Planctomycetota, Figure \ref{fig:isd_top_phyla_samples}.
Actinobacteriota and Proteobacteria in particular are present in all samples with 
more than 50\% joined relative abundance, 36\% and 19.5\% respectively.
Acidobacteriota aren't amongst the top phyla in 11 samples, 9 of them are beach
ecosystems.
Additional notable mentions of phyla are the Desulfobacterota which have specialist
role in the ERR3697703 sample from Richtis gorge. Gemmatimonadota appear mostly
in low carbon, dry samples and related with grazing, Figure \ref{fig:isd_top_phyla_samples}.
The genus Deinococcus,
from the Deinococcota phylum, appears in 3 samples from sandy beaches with high relative abundance.
Apart from bacteria, there are two phyla of archaea in the ISD data, Thermoplasmatota and Euryarchaeota.
These occur in samples from Kolokytha islet, Agiofaraggo gorge and Schinaria beach, all in the 
beach environment. 

\begin{figure}[hbt!] 
    \centering\includegraphics[width=\columnwidth]{isd_fig2_taxonomy}
    \caption{Taxonomic prevalence and representative phyla of the soil bacteria of Crete. 
    A. Distribution of ASVs samples proportion. B. Distribution of taxa samples
proportion and categorisation in generalists and specialists. C. Phyla samples proportion.
D. Phyla relative abundance box plots, each dot represents one sample that the phylum occurs.}
    \label{fig:isd_fig2_taxonomy}
\end{figure}


\begin{figure}[hbt!]
      \centering
      \includegraphics[width=\textwidth,keepaspectratio]{figures/isd_asv_taxonomy_ratios_top_phyla_samples.png}
      \caption[Top phyla of each samples]{The top phyla for all samples with their relative abundance}
      \label{fig:isd_top_phyla_samples}
\end{figure}
   

\subsection{Drivers and Communities}\label{isd_communities}

The samples that have the highest total nitrogen values of soil are
ERR3697708 (Richtis gorge) and ERR3697732 (Lasithi plateau) with 12.3 and 7.9, respectively.
Samples ERR3697703 and ERR3697702 (Richtis gorge) have the highest water content values with 141 and 102, respectively.
Regarding total organic carbon, the samples ERR3697655 (Petres bridge, Rethymno)
and ERR3697675 (Asterousia mountains) have the highest values of 238 and 177, respectively.

Regarding sample diversity, the Shannon index was used.
The samples with the highest diversity using OTUs are ERR3697693 (Elos village, Chania) and
ERR3697675 (Asterousia mountains) with 4.57 and 4.56, respectively.
The sample, ERR3697759 (Agiofaraggo gorge) has the highest number of OTUs, with 1075,
and the sample ERR3697765 (Agiofaraggo gorge) has the second highest number of OTUs, 1051.
The most unique taxa occur in sample ERR3697703 with 871 and in sample ERR3697702
with 869 taxa, both samples from Richtis gorge.

The ASV richness of the samples is correlated (0.25, Pearson correlation) with total organic carbon (p=0.003).
Taxa richness is negatively correlated (-0.30, Pearson correlation) with elevation (p=0.0003). 
Taxa richness is positively correlated (0.26, Pearson correlation) with water content (p=0.002).
$\alpha$-diversity (Shannon Index) is not correlated significantly with any physical and chemical features.
In addition, shannon diversity is negatively correlated with elevation (-0.21, p=0.013).
Total nitrogen and water content also have a positive correlation with elevation, Figure \ref{fig:isd_elevation_metadata}. 

\begin{figure}[hbt!]
      \centering
      \includegraphics[width=\textwidth,keepaspectratio]{figures/isd_abiotic_metadata_elevation_bin_boxplot.png}
      \caption[Elevation and metadata distributions]{The distributions of the available metadata across the elevation of samples}
      \label{fig:isd_elevation_metadata}
\end{figure}

Microbial $\beta$ diversity, measured with the Bray-Curtis metric, is associated in different ways with physical and chemical
measurements. The difference of microbial communities as measured with Bray-Curtis dissimilarity, can be distinguished with 
the UMAP and PCoA ordination methods. UMAP major axis 1 was largely driven by
elevation (F=23, p < 0.001), Figure \ref{fig:isd_elevation_umap1}. 
UMAP 2, was significantly associated with soil moisture (F=90, p < 0.001).
PCoA 2,  also positively correlated with both organic carbon and nitrogen.

\begin{figure}[hbt!]
      \centering
      \includegraphics[width=0.7\textwidth,keepaspectratio]{figures/isd_asv_ordination_UMAP1_elevation_bin_boxplot.png}
      \caption[Elevation and UMAP1]{UMAP major axis and elevation bins}
      \label{fig:isd_elevation_umap1}
\end{figure}

In addition, the dissimilarity of samples was clustered using the complete linkage method, Figure \ref{fig:isd_samples_dendro}. 
A chi square test was performed to investigate whether the cluster membership of the samples
is statistically associated with metadata. Elevation bin, p=0.07646, didn't show any 
significance. Vegetation zone (p=0.0004998) on the other hand was significant.
The sample ERR3697714 (Gournes beach) is a group on its' own, and additional 5 groups can be distinguished. 
This clustering also makes a cluster of the samples of Richtis gorge, the blue group, with an additional sample
of Lasithi plateau. The purple group also has samples from north beaches of the island together.
The brown cluster has samples from Gournes beach and Skinaria beach.

\begin{figure}[!hbt]
    \centering
    % First image (top)
    \begin{subfigure}[b]{0.8\textwidth}
        \centering
        \includegraphics[width=\textwidth,keepaspectratio]{figures/isd_asv_clustering_bray_hclust_samples.png}
        \caption{Sample clustering}
        \label{fig:dendro}
    \end{subfigure}
    \vspace{1cm} % Add vertical space between the images
    % Second image (bottom)
    \begin{subfigure}[b]{0.8\textwidth}
        \centering
        \includegraphics[width=\textwidth]{figures/isd_map_crete_dend_cluster.png}
        \caption{Crete samples with clusters}
        \label{fig:crete_dendro}
    \end{subfigure}
    \caption[Samples dendrogram]{Dendrogram of clustering of samples based on the Bray-Curtis dissimilarity}
    \label{fig:isd_samples_dendro}
\end{figure}

Using the three major axes of PCoA to feed the UCIE tool \parencite{Koutrouli2022} returns the differences 
of microbial communities as color representations. In Figure \ref{fig:isd_crete_sampling}B, each 
sample represents a color and it showcases the diversity of samples even when they are 
in the same transect. Sample dissimilarity of samples from the same site was also investigated \ref{fig:isd_site_locations}.
There are samples that are apart 1 meter that can have up to 40\% difference in
microbial community. Most of the diversity is between samples that are distant apart,
yet it is important to note that soils of Crete can have high diversity even when they 
are adjacent. 

\begin{figure}[hbt!] 
    \centering\includegraphics[width=0.5\columnwidth]{figures/isd_community_site_locations_dif.png}
\caption{Sample dissimilarity between samples in the same site (2 subsites) and the rest.}
    \label{fig:isd_site_locations}
\end{figure}


\section{Discussion}\label{isd_discussion}

The Island Sampling Day Crete provided a great opportunity to 
exercise reproducibility, as intended by the Genomic Standards Consortium.
All data and metadata were available in one platform
under one project identifier. This is rare in microbiome projects. Yet retrieving
all data had it's challenges mostly with metadata. ENA provided xml files when 
using the API which can be cumbersome to the inexperienced users. 
Using one platform to host all data showed some limitations though. The pH values
were not uploaded due to a limitation of the platform because of some samples 
having missing values (the major drivers didn't change when adding these pH values manually, from personal communication).
In addition, control samples couldn't be uploaded,
therefore limiting the decontamination of samples during the bioinformatics workflow.
All these limitations are currently under development and/or are already fixed. 
Even with these limitation, the ISD project remains one of the most transparent
and reproducible soil microbiome projects of this scale.

The design of the ISD samplings has some unique features to study soil microbiome 
in a large scale. First, Crete is an established studied model for biodiversity and 
other studies \parencite{Vogiatzakis2008_crete}. Even though an island,
Crete is mentioned as a miniature continent for it's diverse landscapes in a 
confined space \parencite{Vogiatzakis_land_2017}. Second, the idea of sampling in a single day 
also removes an important variable of microbiomes, seasonality \parencite{Chase2021}.
Third, the notion of transects of sampling to investigate how the microbiome changes
along a route, a gorge and/or a shoreline is very important and still unknown.
These factors constitute ISD Crete a great foundation dataset to study the microbial complexities
of the Cretan soils.

Soil bacterial biodiversity is very complex, even sites a few meters apart can differ
significantly in their community composition as demonstrated in Figure \ref{fig:isd_site_locations}.
This is a fact that is sometimes left unmentioned in worldwide studies \parencite{Bahram2018}.
The island paradigm can assist to remove clutter and use denser samplings.
Apart from the profound diversity in soils, there is also high speciation and uniqueness. 
As shown in Figure \ref{fig:isd_fig2_taxonomy}, most ASVs occur in 1 or 2 samples.
This is different when compering with the ocean \parencite{Sunagawa2015}. When using the deepest taxonomic
level of ASVs is possible to identify the specialists and generalists \parencite{Barberan2012}. 
In addition, focusing on phyla presence in samples, there is a pattern looking like a phase transition, from 
rare phyla to phyla that dominate all samples, Figure \ref{fig:isd_fig2_taxonomy} C.
In Figure \ref{fig:isd_fig2_taxonomy} D,
and \ref{fig:isd_top_phyla_samples}, the distinct top phyla are shown across all samples.
In the soils of Crete, the global patterns also applies with Actinobacteriota,
Proteobacteria, Acidobacteriota and Chloroflexi dominating the samples \parencite{Delgado-Baquerizo-atlas}.

Apart from top phyla, studies like ISD showcase the hidden diversity of soils 
that traditional microbiology cannot isolate \parencite{gilbert2014earth}. 
An example is the extremophile taxa from the Deinococcota phylum and archaea that
inhabit some of the well known beaches of Crete.
Gemmatimonadota phylum also displays an interesting distribution across Crete, 
in dry and high elevations. An observation of these samples is that they 
appear in places where goat grazing is taking place. Crete's landscape 
has been greatly modified by grazing \parencite{JouffroyBapicot2016}.
Maybe the Gemmatimonadota abundant presence can be an indication in some cases.
Yet more investigations are needed both to comprehend the functions of this understudied phylum as well as the 
microbial trails of grazing.
As mentioned in section \ref{inference_taxonomy},
Desulfobacterota have a prominent role in one of the samples, Richtis gorge which
comprises an distinct transect category of samples, Figure \ref{fig:isd_samples_dendro}.
The constant flow of water, the high biomass of plants and/or intense touristic activity may
contribute to the gorge's distinct microbiome. 

Elevation transects and gradients of biodiversity are known since Humboldt's work \parencite{Rahbek2019} 
yet these patterns remain elusive regarding the soil microbiome \parencite{Looby2020, Siles2023}.
Mostly because it's difficult to isolate other co-founding effects \parencite{Nottingham2018}.
In this chapter, elevation showed important distinction of taxa and of diversity, 
with increasing elevation the diversity was reduced. Endemic plants \parencite{trigas2013elevational}
and arthropods \parencite{sfenthourakis2001hotspots} have also been studied in Crete due to its' sharp gradients.
Yet more work is needed to explore the Asterousia mountain, Agiofaraggo and Richtis gorge transects in isolation to avoid indirect influences of 
other variables.

Deciphering and validating the results presented here requires future work.
There are missing links of the drivers and community compositions. These 
gaps can be filled by sampling more data and/or by data integration methods.
The former create new knowledge and are important to continue but they 
require a lot of resources. The latter uses already available knowledge 
to enrich the microbial information (e.g traits) and samples metadata
(e.g. climatic, land use, other taxa occurring in the same area).

"A holistic perspective on soil architecture is needed as a key to soil functions" \parencite{philippot2024the-interplay}, is 
an important statement for future soil projects.
The pillar of data integration is the unrestricted open data across disciplines and 
the open source software. 


