% --------------------------------------------------
% 
% This chapter is for PREGO
% 
% --------------------------------------------------


\chapter{Harmonising the literature and metagenomic resources to infer microbial, environmental and functional relations}
\label{cha:prego}


\section{Introduction}
\label{sec:prego-intro}

The rapid advancements in genomic technology have significantly impacted the
all fields of biology, creating a need for the establishment
of genomic standards to ensure the reliability and reproducibility of research
findings \parencite{Field2011}. As genomic datasets grow exponentially, the importance of these standards
in genomic research cannot be overstated. Standards are essential for enabling
the comparison of results across different studies, facilitating data sharing and
collaboration, and ultimately driving the progression of genomic medicine \parencite{vangay2021microbiome}.
However, the potential of genomic data is often limited by the lack of accompanying
metadata.

Data devoid of metadata are essentially a black box, with no contextual
information to interpret the results or to assess their relevance \parencite{michener_nongeospatial_1997}.
This issue is
prevalent in genomics, where the complexity and diversity of the data
necessitate comprehensive metadata for accurate analysis \parencite{vangay2021microbiome}.
Without proper metadata, it becomes challenging to assess data quality,
reproducibility, and to ensure that research conclusions are robust.
The emergence of metagenomic resources has added a new dimension to genomic
research, with a myriad of databases providing a wealth of data on microbial
communities from various environments.
Examples of public resources are MGnify \parencite{mitchell2020mgnify},
JGI/IMG \parencite{chen2021img} and MG-RAST \parencite{wilke2015restful}). Each one 
has it's one pipeline to analyse user submitted environmental sequences. MGnify
directly imports data from ENA sequence database.
These resources are invaluable for
understanding the genetic diversity and functional potential of microbial
ecosystems. Adding to the complexity, the sheer volume of data and the varying formats in which
they are presented can make comparing and combining these resources a daunting task. 

Metagenomic mining approaches are required to leverage vast databases and analyze
microorganisms from diverse environments \parencite{delmont2011metagenomic}.
Mining allows scientists to extract and interpret genetic material 
from multiple resources thus providing a comprehensive view of the microbiome.
Metadata associated with each sample in these databases, such as source information and collection conditions,
contribute significantly to the contextual understanding and statistical interpretation of the data.
Enable researchers to make meaningful comparisons across different studies.
Furthermore, the application of metagenomic mining to these rich databases can help
uncover novel genes, metabolic pathways, and potential therapeutic targets \parencite{ma2023a-genomic},
thus driving advancements in health, agriculture, and environmental sustainability.
To accomplish this, named entity recognition techniques have become essential for
efficiently harmonising relevant information from metadata.

Named entity recognition is a term identification process. It is based upon
vocabularies and ontologies that provided multiple versions of terms and 
hierarchy relationships. Environment Ontology (ENVO) \parencite{buttigieg2016environment} 
is an important ontology that has classified environments and metagenomic 
resources suggest it's usage in the metadata in fields like biome and 
environmental feature. Molecular functions and biological processes are two separate 
ontologies, part of Gene Ontology \parencite{ashburner2000gene, gene2021gene} that
are widely used in bioinformatics and are open source. 
Regarding functions and pathways, the KEGG database \parencite{kanehisa2000kegg} is the most used resource
in microbial ecology, yet it requires a licence to implement it in a new resource.
Taxa names are also extracted from free text based on different taxonomies. In microbiology,
NCBI taxonomy \parencite{schoch2020ncbi} is convenient to use and implement because every taxon 
is represented by sequences. Nevertheless, LPSN (List of Prokaryotic names with Standing in Nomenclature) \parencite{parte2020list}
is a curated nomenclature that has high accuracy. Based on some of the aforementioned 
systems it is possible to extract entity names from metadata fields and free text in general 
in order to homogenise metadata and data fields.

Apart from metagenomic data, scientific literature contains invaluable information
about microbes, their functions and habitats. This information is usually hidden 
or captured by experts on the fields. 
Text mining methods have been long assisted in extracting knowledge from 
the scientific literature \parencite{jensen2006Literature}. 
By leveraging text mining, researchers can more readily identify and use the rich information that
is often hidden within the text, thereby accelerating discoveries in this
rapidly evolving field \parencite{badal2019Challenges}. Text mining is a multiple step
process \parencite{10.5555/1199003}. It begins with establishing a corpus with 
documents specifically formatted for the tools used. PubMed \parencite{roberts2001pubmed}
holds more that 40 million abstracts through MEDLINE. These data are accessible 
and open access, therefore useful for text mining applications. A downside is the 
domain of publications because it doesn't cover most of ecological and environmental 
journals. Then these documents are analysed with Named Entity Recognition and a 
score of co-occurrences between entities is produced. This lies in the principle 
that the highest the co-mention of terms the higher the probability there is 
a biological meaning of this association \parencite{jensen2006Literature}.
Text mining approaches have been very successful in unearthing novel 
associations of proteins and deceases \parencite{pletscher2015diseases}.

PREGO, the resource developed and presented in \textcite{microorganisms10020293}
as part of this PhD (see \ref{fig:prego_paper}), is a hypothesis generation platform designed to integrate the 
available knowledge about microbes and their processes and environments. 
It is a global platform, first of it's kind that combines metagenomic data with literature
using an established text mining methodology. 
In this chapter, my contributions in the development of PREGO are presented along with the analysis of global 
soil microbiome knowledge.

\section{Methods}
\label{sec:prego-methods}

   \subsection{Pipeline and structure}
   \label{subsec:prego-pipeline}

PREGO's pipeline can be summarised in 6 steps as presented in Figure \ref{fig:prego-pipeline}.
In steps 1 and 2, the web resources are harvested from 3 different types of data,
literature, environmental samples and genome annotations. 
In step 3, the Named Entity Recognition system of the Jensen Lab Tagger is employed \parencite{jensen2016one}
to identify taxa, environments and processes. In addition, in genome annotations
the processes entities are mapped to Gene Ontology terms with custom scripts. 
During the step 4, the co-occurrences of terms is calculated and scored,
each type of resource has it's own scoring scheme. 
This creates a global association network with taxa, processes and environments as 
nodes, co-occurrences as edges and score values as edge weights. 
Lastly, this knowledge graph is uploaded and provided in a web interface 
and through Application Programming Interface.

   \begin{figure}[hbt!]
      \centering
      \includegraphics[width=0.85\columnwidth]{figures/prego_analysis.png}
      \caption[PREGO analysis methodology]{
         PREGO methodology: retrieval of 3 types of open access data resources, literature, environmental DNA samples, and genomic annotations. 
         Named Entity Recognition and co-occurrences bring together associations from organisms, environments and processes. 
         These associations has a score and are provided is a Web interface and an API. Figure adopted by \parencite{microorganisms10020293}.
      }
      \label{fig:prego-pipeline}
   \end{figure}

The knowledge graph in Figure \ref{fig:prego-pipeline} step 5, contains
nodes categorised in three entity types: \textit{Process}, \textit{Environment}, and \textit{Organism}. 
Organisms, i.e taxa, are the microbial NCBI Taxonomy Ids (Bacteria, Archaea, and unicellular eukaryotes).
All environments are mapped to terms of Environmental Ontology. 
Each process corresponds to either a Biological process (GObp) or a Molecular function (GOmf) identifier of Gene Ontology. 

PREGO's knowledge base is structured into three channels, depended on the category of the data type.
\textit{Literature} channel is about the associations extracted by abstracts and full open access text of scientific articles.
The \textit{Annotated Genomes and Isolates} channel contains genome annotations and their harmonised metadata.
The \textit{Environmental Samples} channel integrates metagenomic analyses from amplicon and shotgun sequencing studies. 
Hence, in this channel, the taxonomic and functional profiles and their metadata are harmonised and the associations are extracted.


   \subsection{Dictionary of entities}
   \label{subsec:prego-resources}


PREGO's entities are 4 types and each term corresponds to specific identifier. 
NCBI taxonomy identifiers for taxa, Environmental Ontology for environments and
Gene Ontology for Biological Processes (GObp) and Molecular Functions (GOmfs).
Gene Ontology was selected because it has a Creative Commons Attribution 4.0 License
and because it has been mapped to many other resource identifiers.
These terms, their names and synonyms comprise the PREGO dictionary. In addition,
some terms have been removed because of their uniqueness. The dictionary is 
available for download in the Jensen Lab resources. 
NCBI contains all taxonomy, therefore it was filtered for bacteria, archaea and
for the unicellular eukaryotes. Due to unicellular complex taxonomy a manually curated
list was populated with these taxa.

Some resources use KEGG orthology terms or Uniprot50 ids to indicate functions. 
Hence, a mapping KEGG orthology to GOmf and Uniprot50 to GOmf was carried out via UniProtKB mapping files (see \texttt{idmapping.dat} and \texttt{idmapping\_selected.tab} files). 

   \subsection{Literature}
   \label{subsec:prego-tm}

   PREGO implements a text mining methodology to extract associations of the aforementioned entities from literature. 
   The origin of text mining is a corpus that comprises scientific abstracts and full text articles from MEDLINE® and PubMed® and PubMed Central® Open Access Subset (PMC OA Subset) \parencite{sayers2021database}, respectively. 
   The building and periodic update of the corpus is possible through the NCBI File Transfer Protocol (FTP) services. 
   PREGO also has a dedicated text-mining dictionary (see Availability of Supporting Source Codes section) that contains the entities ids, names, synonyms, and neglected words (stop words). 
   PREGO dictionary incorporates the ORGANISMS \parencite{pafilis2013species} and ENVIRONMENTS \parencite{pafilis2015environments} evaluated dictionaries as well as the experimental dictionaries of Gene Ontology Biological Process and Molecular Function.
   Text mining is subsequently performed on the corpus using the dictionary through the EXTRACT tagger \parencite{pafilis2016extract, jensen2016one}. 
   The tagger recognizes the entities of the dictionary in each abstract and full text article and assigns their co-mentions with a score. 
   The score is sensitive to the text structural level of co-mention; higher to lower scoring when co-mention appears in the same sentence, then, in the same paragraph, and lastly in the same article. 
   All these are integrated and normalized to a single score for each association, as implemented in STRING 9.1 \parencite{franceschini2012string} (see Appendix ~\ref{app:C} for more details). 
   In addition, the tagger extracts each mention in every article to provide the origin of each association it extracts.

The tagger output files, the pubmed tsv files and dictionary files are used by
the 6 perl scripts, Figure \ref{fig:prego-textmining-structure}. These scripts create the database files. More specifically, 
the create documents script reads all the tsv files of pubmed and stores all abstracts
in a single file while simultaneously selecting the latest abstract of the duplicated ones. 
The total duration is 211 minutes. The maximum amount of memory used is 6gb and the tagger
is running using 8 threads.


   \begin{figure}[hbt!]
      \centering
      \includegraphics[width=0.85\columnwidth]{figures/prego_textmining-structure.png}
      \caption[PREGO analysis methodology]{
         Structure of the Literature channel with multiple scripts working together.
         The initial data are the Dictionary and Pubmed. Each one has preprocessing steps. 
         The Text mining steps are the last steps to create the database pairs file 
         with all associations and scores.
      }
      \label{fig:prego-textmining-structure}
   \end{figure}

   \subsection{Environmental Samples}
   \label{subsec:prego-envsamples}

   MGnify \parencite{mitchell2020mgnify} and MG-RAST \parencite{wilke2015restful} repositories provide a great number of public metagenomic records. 
   In the PREGO framework, both amplicon and shotgun metagenomic analyses are retrieved periodically along with their corresponding metadata. 
   Data retrieval from these resources is possible from their Application Programming Interfaces (APIs). Marker gene analyses are retrieved and by measuring
   the co-occurrence of taxa present in the various environmental types (e.g., biomes, materials, features, etc.) organisms—environments associations are extracted. 
   These associations emerge when a taxon is reported together with a certain environmental type, being mentioned in the metadata of a sample (metadata based co-occurrence). 
   Similarly, analyses of metagenomic samples along with their corresponding metadata and annotations are also retrieved and organisms—environments, organisms—processes and processes—environments are extracted. 
   The processes—environments associations are possible through co-occurrence of the functional annotations of metagenomes with the environmental metadata of the samples.
   
   In all cases, the EXTRACT tagger is used on the microorganism names and the corresponding metadata of each sample to identify their identifiers (NCBI ids, ENVO terms, GOmf, GObp). 
   All associations in this channel are scored based on the number of samples the entity of interest co-occurs with specific sample metadata (e.g., environmental type) or annotations (functional annotations or taxonomic annotations). 
   The same scoring scheme was implemented across the channel resources (see Appendix~\ref{app:C} for more details), which ranks these associations with a value in the (0,5] space.
    
   In cases in which the retrieved data and metadata are in text form, they are standardized to the aforementioned identifiers and taxonomies using Named Entity Recognition (NER) tools, namely the EXTRACT tagger \parencite{pafilis2016extract, jensen2016one}. 
   In cases where data contain KEGG Orthology terms and/or Uniref identifiers, they are mapped to the respective GOmf using the mapping files available from the UniProt). 
   Associations are extracted after the mapping and standardization of the entities from each resource (Figure~\ref{fig:prego-pipeline}, step 3).
   The association extraction pipeline is distinct for each channel and resource because of differences in the data type origin (see \texttt{prego\_gathering\_data} in the Availability of Supporting Source Codes section). 
   By the means of navigation, the large number of associations returned to the user require a type of sorting; 
   ideally, one that ranks the most trustworthy associations at the top. 
   For those reasons, each channel of PREGO has a dedicated scoring scheme bounded within the (0,5] space for consistency. 
   In Appendix, the scoring scheme of each channel is elaborated.




   \subsection{Annotated Genomes and Isolates}
   \label{subsec:prego-isolates}

   Annotated genomes and isolates comprise the most trustworthy data in PREGO's knowledge base because they refer to a single species/strain and also have manually curated metadata. 
   Among other data types, JGI-IMG \parencite{chen2021img, mukherjee2021genomes} includes millions of genes from isolated genomes (isolates), SAGs and MAGs. Such annotations, along with their corresponding metadata, were collected using web-parsing technologies. Their metadata, describing their related environment/ecosystem, were tagged using the EXTRACT tagger to infer organisms—environments associations. The annotated KEGG terms were mapped to GOmf terms (see Appendix A). The GOmf terms were then used to extract organisms—processes associations.
   
   The Struo pipeline \parencite{de2020struo} and its outcome when using the Genome Taxonomy DataBase (GTDB) (v.03-RS86) \parencite{parks2020complete} was exploited to enrich organisms—processes associations. 
   A set of 21,276 representative genomes, accompanied by UniRef50 annotations, was retrieved using the provided FTP server. The annotations were then mapped to GOmf terms. 
   Related GTDB genomes were mapped to their corresponding NCBI taxa. 
   All associations extracted from these resources were assigned arbitrarily a confidence level of four out of five. 
   This score choice reflects the high-quality of these data and metadata.
   
   In addition, BioProject data were integrated to PREGO using the NCBI FTP/e-utils services \parencite{sayers2021database}. 
   The BioProject ids that were integrated are the ones that have been assigned a PubMed abstract, a unicellular taxon, and Genome sequencing as data type. Then, using the text mining pipeline, associations were extracted connecting the assigned taxon with the rest of the entities that appear in the abstracts. This method resulted in associations that were assigned a confidence level of three (out of five) because of the combined method of curated data with text mining.


\subsection{Scoring}
\label{scoring}

Scoring in PREGO is used to answer the questions:
\begin{itemize}
   \item Which associations are more thrustworthy?
   \item Which associations are more relevant to the user's query?
\end{itemize}

Relevant, informative, and probable associations are presented to the user through the three channels that were discussed previously. 
Each channel has its own scoring scheme for the associations it contains and all of them are fit in the interval $(0,5]$ to maintain consistency. 
The values of the score are visually shown as stars. 
The Genome Annotation and Isolates channel has fixed values of scores depending on the resource because Genome Annotation is straightforward, and the microbe id is known a priori. 
On the other hand, Environmental Samples channel data are based on samples, which contain metagenomes and OTU tables. 
Thus, it has two levels of organization, microbes with metadata, and sample identifiers. Each association of two entities is scored based on the number of samples they co-occur. 
A Literature channel scoring scheme is based on the co-mention of a pair of entities in each document, paragraph, and sentence. The differences in the nature of data require different scoring schemes in these channels.
The contingency table (Table~\ref{table:pregoA1}) of two random variables, $X$ and $Y$ are the starting point for the calculation of scores. The term $X = 1$ might be a specific NCBI id and $Y = 1$ a ENVO term. 
The $c_{1,1}$ is the number of instances that two terms of $X = 1$ and $Y = 1$ are co-occurring, i.e., the joint frequency. 
The marginals are the $c_{1,.}$ and $c{.,1}$ for $x$ and $y$, respectively, which are the backgrounds for each entity type. 
Different handling of these frequencies leads to different measures. 
There is not a perfect scoring scheme, just the one that works best on a particular instance. 
Consequently, scoring attributes require testing different measures and their parameters.

\begin{table}[ht]
   \centering
   \begin{tabular}{c|llll}
    & \multicolumn{4}{l}{Y = y} \\ \cline{2-5} 
   \multirow{4}{*}{X = x} &  & Yes & No & Total \\ \cline{3-5} 
    & \multicolumn{1}{l|}{Yes} & $c_{x,y}$ & $c_{x,0}$ & $c_{x,.}$ \\
    & \multicolumn{1}{l|}{No} & $c_{0,y}$ & $c_{0,0}$ & $c_{0,.}$ \\
    & \multicolumn{1}{l|}{Total} & $c_{.,y}$ & $c_{.,0}$ & $c_{.,.}$
   \end{tabular}
   \caption[PREGO contingency table between two terms]{Contingency table of co-occurrences between entities $X = x$ and $Y = y$. 
   This is the basic structure for all scoring schemes. $c_{x,y}$ is the count of the co-occurrence of these entities. $c_{x,.}$ is the count of the $x$ with all the entities of $Y$ type (e.g., Molecular function). Conversely, $c_{.,y}$ is the count of $y$ with all the entities of $X$ type (e.g., taxonomy}
   \label{table:pregoA1}
\end{table}


\subsubsection{Literature}

Scoring in the Literature channel is implemented as in STRING 9.1 \parencite{franceschini2012string} and COMPARTMENTS \parencite{binder2014compartments}, where the text mining method uses a three-step scoring scheme. 
First, for each co-mention/co-occurrence between entities (e.g., Methanosarcina mazei with Sulfur carrier activity), a weighted count is calculated because of the complexity of the text.  


\begin{equation}
   c_{x,y} = \sum_{k=1}^{n}{w_d \delta_{dk}(x,y) +w_p \delta_{p,k}(x,y) + w_s \delta_{sk}(x,y)}
   \label{eq:prego-score-1}
\end{equation}



Different weights are used for each part of the document ($k$) for which both entities have been co-mentioned, $w_d = 1$ for the weight for the whole document level, $w_p = 2$ for the weight of the paragraph level, and $w_s = 0.2$ for the same sentence weight. 
Additionally, the delta functions are one (Equation~\ref{eq:prego-score-1}) in cases the co-mention exists, zero otherwise. Thus, the weighted count becomes higher as the entities are mentioned in the same paragraph and even higher when in the same sentence.
Subsequently, the co-occurrence score is calculated as follows:

\begin{equation}
   score_{x,y} = c_{x,y}^a (\frac{c_{x,y} c_{.,.}}{c_{x, .}c_{.,y}})^{1-a}
   \label{eq:prego-score-2}
\end{equation}
   


where $a = 0.6$ is a weighting factor, and the $c_{x,.}$, $c_{.,.}$, 
$c_{.,y}$ are the weighted counts as shown in Table~\ref{table:pregoA1} estimated using the same Equation~\ref{eq:prego-score-2}. 
This value of the weighting factor has been chosen because it has been optimized and benchmarked in various 
applications of text mining~\parencite{franceschini2012string, binder2014compartments, pletscher2015diseases}. 
The value of Equation~\ref{eq:prego-score-2} is sensitive to the increasing size of the number of documents (MEDLINE PubMed—PMC OA).
Therefore, to obtain a more robust measure, the value of the score is transformed to $z$-score. 
This transformation is elaborated in detail in the COMPARTMENTS resource \parencite{binder2014compartments}. 
Finally, the confidence score is the $z$-score divided by two. Cases in which the scores exceed the (0,4] interval are capped to a maximum of 4 to reflect the uncertainty of the text mining pipeline.

\subsubsection{Environmental Samples}

Data from environmental samples are OTU tables and metagenomes. 
Thus, for each entity $x$, the number of samples is calculated as the background 
and a number of samples of the associated entity (metadata background) $c.,y$ (see Table~\ref{table:pregoA1}). 
Each association between entities $x$,$y$ has a number of samples, $c_{x,y}$ that they co-occur. 
Note that each resource is independent and the scoring scheme is applied to its entities. 
This means that the same association can appear in multiple resources with different scores. 
The score is calculated with the following formula:

\begin{equation}
   score_{x,y} = 2.0*{\frac{\sqrt{c_{x,y}}}{c_{.,y}^{0.1}}}
\end{equation}


This score is asymmetric because the denominator is the marginal of the associated entity. 
Thus, the score decreases as the marginal of $y$ is increasing, i.e., the number of samples that $y$ is found. 
On the other hand, it promotes associations in which the number of samples of 
the association are similar to the marginal of $y$. 
The exponents on the numerator and denominator equal to $0.5$ and 
to $0.1$, respectively, in order to reduce the rapid increase of score.
Lastly, the value of the score is capped in the range $(0,4]$.


\subsection{Behind the scenes}
\label{deamons}

PREGO's pipeline and hosting is running on a 64 GB RAM DELL R540, 20 core, Debian server.
The server has 2 terabyte SSD storage for high responsiveness. 
All code is versioned with the git versioning tool to keep history and 
resolve bugs. The development cycles of PREGO are based on DevOps practices that
combine development and operations and enable the efficient deployment of software services.
The Continuous Development and Continuous Integration (CD/CI) method proved to be
important aspect during the developed of PREGO in terms of multiple testing,
resolving bugs and effective collaborative coding. 

The pipeline of PREGO is streamlined and integrated to be executed regularly,
spanning from once a month to six-month cycles, ensuring that changes of resources are
continuously integrated into PREGO. These cycles are called daemons in PREGO and 
each channel has a dedicated script that executes all the code while 
maintaining backups, Figure \ref{fig:devops}.


\begin{figure}[hbt!]
   \centering
   \includegraphics[width=95mm]{figures/figure_A1_PREGO_daemons.png}
   \caption[PREGO DevOps]{Scripts called daemons are executing the PREGO methodology in cycles depending on the updates of the databases used.}
   \label{fig:devops}
\end{figure}


   \subsection{Code} 
Code produced for PREGO are available under BSD 2-Clause “Simplified” License.
Scripts, where third party libraries have been used, are subject to their individual licenses.
   
   \begin{itemize}
      \item prego\_gathering\_data 
      \href{https://github.com/lab42open-team/prego_gathering_data}{github.com/lab42open-team/prego\_gathering\_data}
      \item prego\_daemons \href{https://github.com/lab42open-team/prego_daemons}{github.com/lab42open-team/prego\_daemons}
      \item prego\_mappings \href{https://github.com/lab42open-team/prego_mappings}{github.com/lab42open-team/prego\_mappings} 
      \item prego\_statistics \href{https://github.com/lab42open-team/prego_statistics}{github.com/lab42open-team/prego\_statistics}
   \end{itemize}

Additional software and curated lists along with their individual license are:
   \begin{itemize}
      \item tagger:	\href{https://github.com/larsjuhljensen/tagger}{https://github.com/larsjuhljensen/tagger}, BSD 2-Clause "Simplified" License
      \item mamba: \href{https://github.com/larsjuhljensen/mamba}{https://github.com/larsjuhljensen/mamba}, BSD 2-Clause "Simplified" License 
      \item tagger dictionary:  \href{https://download.jensenlab.org/}{https://download.jensenlab.org/} and there in: \\
      \href{https://download.jensenlab.org/prego_dictionary.tar.gz}{https://download.jensenlab.org/prego\_dictionary.tar.gz}, CC-BY 4.0 license
   \end{itemize}

\section{Results}
\label{sec:prego-results}

   \subsection{PREGO Contents}
   \label{subsec:prego-contents}

The data sources of PREGO can be categorized into six types: abstracts, full articles,
isolates, annotated genomes, markergene samples, and metagenomic samples, Figure \ref{fig:prego-summary}.
In terms of metadata availability, JGI IMG, Struo, BioProject, and MG-RAST provide
metadata for their respective data types.
BioProject provides metadata for annotated genomes with abstracts, which could be useful for researchers seeking comprehensive information.
The number of items varies significantly across sources, with MEDLINE and PubMed
having the largest collection of items (33 million) and PubMed Central OA Subset
having a significant collection of full articles (2.7 million).
JGI IMG, Struo, BioProject, and MG-RAST have a smaller but still substantial number of isolates, annotated genomes, and markergene samples.
In terms of licenses, the majority of the sources have open licenses,
such as CC0, CC BY-SA 4.0, CC-BY, and NLM Copyright, which allows for the free
use and sharing of the data. This is an essential aspect of data sharing and
collaboration in the scientific community.

   \begin{figure}[hbt!]
      \centering
      \includegraphics[width=\columnwidth]{figures/PREGO_summary_channels.png}
      \caption[PREGO contents summary]{
         Resources, entities and associations summary in each channel of information of PREGO.}
      \label{fig:prego-summary}
   \end{figure}

Regarding the entity types, PREGO's knowledge base contains 364,000 taxa
(NCBI Taxonomy has 620,000 Bacteria, Archaea, and microbial eukaryotes). 
About 258,000 taxa are at the species level. 
All environment Ontology terms are found at about 1000 terms. Regarding Gene ontology, 15,000 biological process terms and 7.900 molecular function term are present.
PREGO's knowledge base of entities and associations are a multipartite network
with entities as nodes and co-occurrences as links with score as weight.


\subsection{Bulk download}
\label{bulk-download}

All the knowledge base of PREGO is available for download programmatically.
In Table~\ref{table:prego-appD-1} the hyperlinks are provided, one per channel, along with md5sum files for sanity checks.
Data are compressed to tar.gz format so expect an order of magnitude higher file size when decompressed.

   \begin{table}[ht]
      

      \begin{tabular}{lll}
      \toprule
      Channel Link & md5sum & Size (in GB) \\ \midrule

      \href{https://prego.hcmr.gr/download/literature.tar.gz}{Literature} & \href{https://prego.hcmr.gr/download/literature.tar.gz.md5}{literature.tar.gz.md5} & 5.4 \\

      \href{https://prego.hcmr.gr/download/environmental\_samples.tar.gz}{Environmental samples} & 
      \href{https://prego.hcmr.gr/download/environmental\_samples.tar.gz.md5}{environmental\_samples.tar.gz.md5}
      & 0.69 \\

      \href{https://prego.hcmr.gr/download/annotated\_genomes\_isolates.tar.gz}{Annotated genomes} &
      \href{https://prego.hcmr.gr/download/annotated\_genomes\_isolates.tar.gz.md5}{annotated\_genomes\_isolates.tar.gz.md5} & 0.26 \\ \bottomrule
      \end{tabular}
      \caption[PREGO Bulk download.]{Bulk download links and md5sum files.}
      \label{table:prego-appD-1}
   \end{table}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% soil %%%%%%%%%%%%%%%%%%%%%%%%%%%%
   \subsection{PREGO about soil}
   \label{subsec:prego-soil}


%   \begin{table}[ht]
%      \begin{tabular}{@{}cccrccc@{}}
%      \toprule
%      \textbf{Channel} & \textbf{Source} & \multicolumn{2}{c}{\textbf{Taxonomy}} & \textbf{\begin{tabular}[c]{@{}c@{}}Environ- \\ ments\end{tabular}} & \textbf{\begin{tabular}[c]{@{}c@{}}Biological \\ Processes\end{tabular}} & \textbf{\begin{tabular}[c]{@{}c@{}}Molecular \\ Functions\end{tabular}} \\ \midrule
%      \multirow{3}{*}{Literature} & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MEDLINE \\ PubMed - \\ PMC OA\end{tabular}} & Strains & 8,929 & \multirow{3}{*}{1,077} & \multirow{3}{*}{15,079} & \multirow{3}{*}{7,318} \\
%      &  & Species & 240,377 &  &  &  \\
%      &  & Total & 342,506 &  &  &  \\
%      \multirow{9}{*}{\begin{tabular}[c]{@{}c@{}}Environ- \\ mental \\ samples\end{tabular}} & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MG-RAST \\ amplicon\end{tabular}} & Strains & 1,392 & \multirow{3}{*}{162} & \multirow{3}{*}{-} & \multirow{3}{*}{-} \\
%      &  & Species & 4,324 &  &  &  \\
%      &  & Total & 5,859 &  &  &  \\
%      & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MG-RAST \\ metagenome\end{tabular}} & Strains & 2,522 & \multirow{3}{*}{258} & \multirow{3}{*}{-} & \multirow{3}{*}{3,839} \\
%      &  & Species & 4,406 &  &  &  \\
%      &  & Total & 7,157 &  &  &  \\
%      & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MGnify \\ amplicon\end{tabular}} & Strains & 2 & \multirow{3}{*}{216} & \multirow{3}{*}{11} & \multicolumn{1}{l}{\multirow{3}{*}{-}} \\
%      &  & Species & 1,471 &  &  & \multicolumn{1}{l}{} \\
%      &  & Total & 2,955 &  &  & \multicolumn{1}{l}{} \\
%      \multirow{9}{*}{\begin{tabular}[c]{@{}c@{}}Annotated\\Genomes \&\\Isolates\end{tabular}} & \multirow{3}{*}{JGI IMGisolates} & Strains & 2,398 & \multirow{3}{*}{241} & \multirow{3}{*}{-} & \multirow{3}{*}{3,670} \\
%      &  & Species & 11,203 &  &  &  \\
%      &  & Total & 13,849 &  &  &  \\
%      & \multirow{3}{*}{STRUO} & Strains & 6 & \multirow{3}{*}{-} & \multirow{3}{*}{-} & \multirow{3}{*}{2,789} \\
%      &  & Species & 19,289 &  &  &  \\
%      &  & Total & 19,325 &  &  &  \\
%      & \multirow{3}{*}{BioProject} & Strains & 5,754 & \multirow{3}{*}{309} & \multirow{3}{*}{626} & \multirow{3}{*}{-} \\
%      &  & Species & 3,373 &  &  &  \\
%      &  & Total & 9,393 &  &  &  \\
%      \multirow{3}{*}{Total} & \multirow{3}{*}{All} & Strains & 12,840 & \multirow{3}{*}{1,090} & \multirow{3}{*}{15,091} & \multirow{3}{*}{7,971} \\
%      &  & \multicolumn{1}{l}{} & \multicolumn{1}{l}{} &  &  &  \\
%      &  & \multicolumn{1}{l}{} & \multicolumn{1}{l}{} &  &  &  \\ \bottomrule
%      \end{tabular}
%
%      \caption[The entities of PREGO after the NER and mapping of every source]{
%         The entities of PREGO after the NER and mapping of every source. 
%         Counts of distinct entities of Taxa, Environments (ENVO terms), Biological Processes (Gene Ontology Biological process) and Molecular Function (Gene Ontology Molecular Function).
%      }
%      \label{table:prego2}
%
%   \end{table}



\section{Discussion}
\label{sec:prego-discussion}

   \subsection{PREGO Contents}
   \label{subsec:prego-contents-disc}

   On its current version and according to the NCBI Taxonomy that it is based on, PREGO manages to cover a great range of microbial taxa, as most (if not all phyla) are present in the knowledge base (Figure~\ref{fig:prego-entities}). 
   The different number of organisms' entities per phylum highlights the diverse number of the members of the various phyla. On the contrary, the similar number of molecular functions in all cases indicates the robustness of the main metabolic processes required for life. 
   With respect to biological processes, their number per phylum varies to some extent, especially for the case of Bacteria and Archaea. 
   That could be observed as, in many cases, phyla that have been recently described using molecular techniques have not been studied extensively yet, e.g., Candidatus Delongbacteria. 
   As expected, the number of environmental types that have been associated with members of each phylum varies, as a phylum may be universally present, while others could be strongly niche-specific (e.g., Hydrothermarchaeota).

   Because of its three different channels, PREGO manages to extract associations both in the species and higher taxonomic levels. The Isolates channel supports explicit associations at the species level (Table~\ref{table:prego3} and Figure S3). 
   Interestingly, the number of such genomes seems to have reached a plateau for now, as PREGO-like platforms include the same order of magnitude. 
   The \textit{Literature} channel, on the other hand, promotes the extraction of associations at higher taxonomic levels (Table~\ref{table:prego3} and Figure S1). 
   This also applies to environment—organisms associations derived from the Environmental Samples channel (Table~\ref{table:prego3} and Figure S2). Associations regarding biological processes, though, are strongly enhanced by the Literature channel and the massive increase of literature.


%   % ASSOCIATIOS TABLE - table:prego3
%   \begin{sidewaystable}
%      \begin{tabular}{@{}cccccrcr@{}}
%      \toprule
%      \textbf{Channel} & \textbf{Source} & \textbf{\begin{tabular}[c]{@{}c@{}}Environments \\ - \\ Processes\end{tabular}} & \textbf{\begin{tabular}[c]{@{}c@{}}Environments \\ - \\ Functions\end{tabular}} & \textbf{Taxonomy} & \multicolumn{1}{c}{\textbf{\begin{tabular}[c]{@{}c@{}}Taxa \\ -\\  Environments\end{tabular}}} & \textbf{\begin{tabular}[c]{@{}c@{}}Taxa \\ - \\ Processes\end{tabular}} & \multicolumn{1}{c}{\textbf{\begin{tabular}[c]{@{}c@{}}Taxa \\ -\\ Function\end{tabular}}} \\ \midrule
%      \multirow{3}{*}{Literature} & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MEDLINE \\ PubMed - \\ PMC OA\end{tabular}} & \multirow{3}{*}{883,997} & \multirow{3}{*}{422,579} & Strains & 69,968 & \multicolumn{1}{r}{590,630} & 384,079 \\
%      &  &  &  & Species & 778,877 & \multicolumn{1}{r}{3,501,635} & 1,961,920 \\
%      &  &  &  & Total & 1,669,608 & \multicolumn{1}{r}{7,969,310} & 4,613,827 \\
%      \multirow{9}{*}{\begin{tabular}[c]{@{}c@{}}Environmental \\ samples\end{tabular}} & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MG-RAST \\ amplicon\end{tabular}} & \multirow{3}{*}{-} & \multirow{3}{*}{-} & Strains & 13,645 & \multirow{3}{*}{-} & \multicolumn{1}{c}{\multirow{3}{*}{-}} \\
%      &  &  &  & Species & 39,007 &  & \multicolumn{1}{c}{} \\
%      &  &  &  & Total & 53,439 &  & \multicolumn{1}{c}{} \\
%      & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MG-RAST \\ metagenome\end{tabular}} & \multirow{3}{*}{-} & \multirow{3}{*}{620,846} & Strains & 262,106 & \multirow{3}{*}{-} & 8,626,328 \\
%      &  &  &  & Species & 103,913 &  & 10,715,548 \\
%      &  &  &  & Total & 372,301 &  & 19,950,096 \\
%      & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MGnify \\ amplicon\end{tabular}} & \multirow{3}{*}{-} & \multirow{3}{*}{-} & Strains & 18 & - & \multicolumn{1}{l}{} \\
%      &  &  &  & Species & 30,122 & \multicolumn{1}{r}{351} & \multicolumn{1}{c}{-} \\
%      &  &  &  & Total & 111,976 & \multicolumn{1}{r}{2,097} & \multicolumn{1}{l}{} \\
%      \multirow{9}{*}{\begin{tabular}[c]{@{}c@{}}Annotated Genomes \\ and Isolates\end{tabular}} & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}JGI IMG\\ isolates\end{tabular}} & \multirow{3}{*}{-} & \multirow{3}{*}{-} & Strains & 8,229 & \multirow{3}{*}{-} & 3,461,693 \\
%      &  &  &  & Species & 42,141 &  & 13,216,559 \\
%      &  &  &  & Total & 50,888 &  & 16,821,850 \\
%      & \multirow{3}{*}{STRUO} & \multirow{3}{*}{-} & \multirow{3}{*}{-} & Strains & \multicolumn{1}{c}{\multirow{3}{*}{-}} & \multirow{3}{*}{-} & 1,803 \\
%      &  &  &  & Species & \multicolumn{1}{c}{} &  & 4,070,195 \\
%      &  &  &  & Total & \multicolumn{1}{c}{} &  & 4,079,312 \\
%      & \multirow{3}{*}{BioProject} & \multirow{3}{*}{-} & \multirow{3}{*}{-} & Strains & 3,263 & \multicolumn{1}{r}{7,473} & \multicolumn{1}{l}{} \\
%      &  &  &  & Species & 4,187 & \multicolumn{1}{r}{4,294} & \multicolumn{1}{l}{} \\
%      &  &  &  & Total & 7,641 & \multicolumn{1}{r}{12,169} & \multicolumn{1}{l}{} \\
%      \multirow{3}{*}{Total} & \multirow{3}{*}{All} & \multirow{3}{*}{883,997} & \multirow{3}{*}{1,043,425} & Strains & 357,229 & \multicolumn{1}{r}{598,103} & 12,473,903 \\
%      &  &  &  & Species & 998,247 & \multicolumn{1}{r}{3,506,280} & 29,964,222 \\
%      &  &  &  & Total & 2,265,853 & \multicolumn{1}{r}{7,983,576} & 45,465,085 \\ \cmidrule(l){5-8} 
%      \end{tabular}
%      \caption[Associations among the PREGO entities]{
%         The associations between entities of PREGO after co-occurrence analysis: The supported entity types of associations are Environments—Biological Processes, Environments—Molecular Functions, Taxa—Environments, Taxa—Biological Processes, Taxa—Molecular Functions.
%      }
%      \label{table:prego3}
%   \end{sidewaystable}

   Additionally, the text mining methodology of the Literature channel has retrieved most of the entities present in PREGO knowledge base. 
   A significant contribution to the taxa with associations is due to the PMC OA processing by the text mining pipeline of the Literature channel. 
   This is in-line with reports in other applications of text mining when using full text articles \parencite{westergaard2018comprehensive}. 
   However, the resulting associations are suggestive because of the text mining nature, and therefore subject for further review by the users.

   \subsection{Platform comparison}
   \label{subsec:prego-similar-platforms}


Based on the table \ref{table:prego4}, BacDive seems to be the most versatile
platform among the four, offering a wide range of features. It has a high level
of manual curation, with environment-taxa associations, process/function-taxa
associations, and phenotypic data available. Additionally, BacDive provides
spatial coordinates, an application programming interface but limited bulk download of data.
It has a high score for manual curation, indicating that it is well-maintained and regularly updated with accurate information.
As for the presence of various types of data, BacDive appears to be the platform
with the most comprehensive dataset, offering original and integrated data from
various sources. Its high scores for environment-taxa associations,
process/function-taxa associations, and phenotypic data suggest that it contains a broad range of microbial data.

PREGO appears to have some notable benefits. Specifically, it has a high score
for literature integration, which suggests that it is able to integrate microbial
data from various sources and publications. Additionally, PREGO is capable of
processing and storing data on environment-process/function associations, which
could be useful for researchers studying microbial processes and functions.
Another benefit of PREGO is its ability to facilitate bulk downloads of data,
making it easier for researchers to access and analyze large datasets.
This feature can be particularly useful for big data analysis and machine learning applications.
It is worth noting that PREGO's scores for other features, such as manual
curation and environment-taxa associations, are relatively low compared to
BacDive and Web of Microbes. However, its strengths in literature integration
and bulk data download may make it a useful platform for researchers with specific needs.


   \begin{table}[ht]

      \begin{adjustwidth}{-0.75cm}{}

      \begin{tabular}{@{}lllll@{}}
      
      \toprule
      Functionality & BacDive & Web of Microbes & NMDC & PREGO \\ \midrule
      manual curation & high & high & intermediate & low \\ 

      literature integration & limited & no & no & yes \\

      environment—taxa associations & yes & yes & yes & yes \\

      \begin{tabular}[c]{@{}l@{}}environment—process/\\ function associations\end{tabular} & no & no & no & yes \\

      process/function—taxa associations & yes & yes & yes & yes \\
      phenotypic data & yes & no & no & no \\

      data origin & \begin{tabular}[c]{@{}l@{}} original \\integration \end{tabular} & original & \begin{tabular}[c]{@{}l@{}} original \\integration \end{tabular} & integration \\

      spatial coordinates & yes & no & yes & no \\

      application programming interface & yes & no & yes & yes \\

      bulk download & limited & yes & yes & yes \\ \bottomrule

      \end{tabular}
      \end{adjustwidth}
      \caption[Feature comparison between PREGO and other similar platforms]{Feature comparison among platforms that facilitate knowledge discovery and integration of microbial data.}
      \label{table:prego4}
   
   \end{table}      


   \subsection{Soil microbiome}
   \label{subsec:prego-soil}



