% --------------------------------------------------
% 
% This chapter is for Crete system ecology
% 
% --------------------------------------------------


\chapter{Island Sampling Day, the case of Crete}
\label{cha:isd-crete-soil}

%\textbf{Citation:} \\ 

\section{Introduction}\label{intro}

Microbes influence global ecosystem functions \parencite{falkowski2008microbial}
and are ubiquitous \parencite{delgado2016microbial}. The terrestrial biomass of bacteria is
70 Gt C, second to plants with sixfold difference \parencite{bar2018biomass}. In subsurface systems, bacteria and
archaea are the most abundant, having 90\% of the biomass.

Microbial ecology based on amplicon 16s rRNA sequences has flourished since the
2010s. The endeavor to understand the microbial world faces multiple challenges
across the scientific workflow, from sampling to ecological analyses \parencite{Lee2012}.
Errors propagate starting with the sampling. There are contaminations from the
people in the field, in the lab for the DNA extraction \parencite{EISENHOFER2019105}. 
Errors from PCR amplification and errors from sequencing \parencite{Schloss2011, Schimer2015}.
Approximation and computation errors from algoritms that cluster, measure similarities between
sequences. Semantic errors because of reductionist approaches and/or oversimplification
of the microbial communities. Microbiome matrix data are compositional \parencite{Gloor2017}


Amplicon rRNA sequencing provides a collection of sequence reads per sample. 
The ecological interpretation of the reads requires their transformation to
taxonomic information. To do so there are two approaches currently in use, 
the clustering method and the denoising method. With clustering reads are 
grouped together and a best representing sequence is produced for each 
cluster, i.e. the Operetional Taxonomic Unit. This approach makes the OTUs 
from different runs, i.e executions of the algorithm and/or different studies
incoperable and irreproducible. Currently, many studies propose the use of Amplicon Sequence Variants \parencite{Callahan2017}. 
ASVs are real biological sequences and can be used for comparison. The influence of the different methods to subsequent ecological analyses has
little impact \parencite{Glassman2018}.




Soil ecosystems are the cornerstone of terrestrial habitats, biodiversity and henceforth human activities.
Soils are characterised by multiple properties; chemical, physical and biological that 
form complex interdependent interactions. Biodiversity of soils covers
all forms of life, fauna, flora, bacteria, archaea, fungi, viruses. 
Bacteria and archaea are considered major drivers for the functionality of soil.
They influence and are influenced by their environment and their community structure 
defies their macroscopic functionality \parencite{Bahram2018}.
Global soil microbiome studies have been employed to decipher soil microbiome
compositions \parencite{thompson2017a-communal, Delgado-Baquerizo-atlas, Labouyrie2023},
functions \parencite{Bahram2018} and biogeography \parencite{Martiny2006, guerra2020Blind}.
These results showed the remarkable diversity in soils yet there are blind spots \parencite{guerra2020Blind}
and these sampling are sparse when considering samples per area density. One of most resolute
study is by \parencite{Karimi2020} which exemplified the 
vast complexity of soil bacterial communities and the requirement of
dense samplings and isolated systems \parencite{Dini-Andreote2021}.

Islands are nature's labs \parencite{Whittaker2017} because of their isolation and smaller scale.
Borrowing this paradigm, soil microbes \parencite{Li2020} and mycorrhizal fungi \parencite{Delavaux2021} studies
in islands show the benefits of using islands as models. Ultimately islands can
set the ground to represent ecosystems with data and complex interactions \parencite{Davies2016}.
This was also the case for the Island Sampling Day (ISD) project \parencite{holm2024}
of the Genome Standards Consortium \parencite{Field2011}
during the 18th workshop in June 2026 in Crete island, Greece. The goal was to "put standards into action"
in a soil microbiome survey with a dense sampling, 0.017 samples per km\textsuperscript{2}.
Hence, ISD is a large scale study in the confined space of the island of Crete which 
is considered a miniature continent \parencite{Vogiatzakis2008_crete}.

In the ISD project, ten teams followed a pre-defined sampling route of sampling locations. Samples
were collected in a single day in order to control for variations in environmental
factors that may impact the abundance and diversity of microbes such as season,
temperature and humidity. Soil samples were collected by 26 participants in 10 teams.
For each sampling location, teams selected two specific sampling sites that were
at least 10 feet from the edge of the road and 2 feet (0.6 m) from the base of
the identified tree or plant. Flora located at the sampling site were
identified and photographed. Soil was collected from the two sub site locations (5-10 feet (1.5-3 m) apart).
At each sub site, soil was sampled (3 replicates, collected one inch apart). 

Two replicate sets of the 144 sub site samples were shipped, on dry ice, to the
University of Maryland Soil Lab (Dr. Stephanie Yarwood) for RNA and DNA extraction
and soil chemistry analysis (USDA PERMIT NUMBER: P330-16-00090).
Prior to DNA extraction, the moisture, total organic Carbon and Nitrogen (CN) weights
and CN analysis were determined. The CN analysis was conducted on 10/4/2016.
Dr. Yarwoodâ€™s group submitted 144 combustion capsules of soil for C and N analyses using LECO CN628 

DNA was extracted from 144 soil samples following the Earth Microbiome Project (EMP)
standard protocols, utilizing the MoBio RNA extraction kit and the  MoBio DNA elution kit to go with RNA kit (Qiagen).
Quantification was performed using Qbit (Thermofisher) to confirm extraction, Qbit dsDNA
quantification. Three of the samples, collected in sand, had no detectable DNA
following extractions. Additionally, 17 samples had a DNA amount too low for metagenomics: 1 from ISD1, ISD4, and ISD35, 6 from ISD7, 4 from ISD8, and 3 from ISD10. A second attempt to re-extract with MoBio powermax kit (2g soil). 

Amplicon libraries were prepared and sequenced (16S ribosomal (rRNA) amplicon
sequenced (16S rRNA V3, V4 region)) on an Illumina HiSeq2500 with processing of
the reads conducted to determine the relative abundances and taxonomic diversity of bacterial taxa in soil.
Sequencing analysis was conducted at the Institute for Genome Sciences, Genomics
Resource Center (Maryland Genomics) at the University of Maryland School of Medicine.
The V3-V4 region of 16S rRNA gene from each sample was sequenced, using a
PCR-based protocol that targets the V3-V4 region of the 16S rRNA gene (V3 V4 primers: Forward: 5'-ACTCCTACGGGAGGCAGCAG-3'; Reverse: 5'-GGACTACHVGGGTWTCTAAT-3').

In this study




\section{Materials and Methods}\label{methods}

   \begin{figure}[h]
      \centering
      \includegraphics[width=\textwidth,height=\textheight,keepaspectratio]{figures/isd_crete_flowchart.png}
      \caption[Reproducible workflow of ISD analysis]{Three steps of the ISD bioinformatics workflow, reproducible by design}
      \label{fig:isd_workflow_taxonomy}
   \end{figure}
   
\subsection{Island Sampling Day: Crete}\label{isd_data}

This study uses the ISD Crete data that have been deposited
in the European Nucleotide Archive (ENA) at EMBL-EBI under accession number PRJEB21776.
The raw sequences (fastq files) and metadata (xml files) were downloaded with custom scripts using the ENA API \parencite{Yuan2023}.
Specific information regarding the ISD sampling, DNA extractions, PCR and sequencing
protocols are discussed here \parencite{holm2024}. The bioinformatics workflow is 
reproducible as described in Appendix Figure \ref{fig:isd_workflow_taxonomy}.

Amplicon Sequence Variants were inferred using DADA2 \parencite{Callahan2016} for 
filtering, denoising and chimeric reads removal. Normalisation of the reads
across samples was implemented with the SRS R package \parencite{Beule2020}. Samples
with less than 10000 reads were removed. ASVs were assigned to taxonomy using 
DADA2 with Silva 138 \parencite{quast_silva_2013}.

\begin{figure}[t] 
    \centering\includegraphics[width=\columnwidth]{isd_map_fig1-small}
    \caption{Crete ISD sampling. A. The routes of the single day event sampling. B. Beta diversity differences are represented with color.}
    \label{fig:isd_crete_sampling}
\end{figure}



\subsection{Tools}\label{Coding environment}
PEMA for OTU inference \parencite{zafeiropoulos2020pema}
U-CIE R package for coloring 3 dimensional data \parencite{Koutrouli2022}

Visualisation was implemented with ggplot2 \parencite{wickham_ggplot2_2016} and pheatmap \parencite{Kolde2019}.
The environment we worked had Python 3.11.4, R version 4.3.2 \parencite{rcoreteam}
and Julia language version 1.9.3 \parencite{Julia-2017}in Julia language version 1.9.3 \parencite{Julia-2017}.
Finally, computations were performed on HPC infrastructure of HCMR \parencite{zafeiropoulos_0s_2021}.

\section{Results}\label{results}

\subsection{Soil microbiome}\label{soil_microbiome}

The Illumina HiSeq2500 libraries yielded 51 million sequenced reads (16-20 GB) with
an average of 355,326 reads/sample (range 2,437-525144). Sequences and the
metadata thereof were submitted to the ENA under the study accession number PRJEB21776.
Sequencing produced 51 million reads, with an average of 250-500 K reads per
sample (15-20 GB of data), with an average of 355,326 average reads per sample
(range: 2,437-525,144) and less than 15,000 reads for reads of either less than
445 bp or over 515 bp, Figure \ref{fig:isd_srs-curve_samples}.
   
   \begin{figure}[h]
      \centering
      \includegraphics[width=\textwidth,height=\textheight,keepaspectratio]{figures/isd_crete_srs_curve.jpeg}
      \caption[SRS curve]{SRS curve of the samples. }
      \label{fig:isd_srs-curve_samples}
   \end{figure}

\begin{figure}[t] 
    \centering\includegraphics[width=\columnwidth]{figures/isd_community_site_locations_dif.png}
\caption{Sample dissimilarity between samples in the same site (2 subsites) and the rest.}
    \label{fig:isd_site_locations}
\end{figure}

In summary, DADA2 resulted in 216,360 ASVs (2,704 unique
taxa; 1123 ASVs at species level and 1059 at genus level) and
PEMA (VSEARCH) in 13,285 OTUs.
The representative Phyla ( > 5\% presence in all samples) are:
Actinobacteriota, Proteobacteria, Chloroflexi, Acidobacteriota,
Bacterodota and Planctomycetota, Figure \ref{fig:isd_top_phyla_samples}.
There are 25 specialist (samples < 10 and mean relative
abundance > 0.003) and 146 generalist taxa (samples > 120), Figure \ref{fig:isd_fig2_taxonomy}.

\begin{figure}[t] 
    \centering\includegraphics[width=\columnwidth]{isd_fig2_taxonomy}
    \caption{Taxonomic prevalence and representative phyla of the soil bacteria of Crete. 
    A. Distribution of ASVs samples proportion. B. Distribution of taxa samples
proportion and categorisation in generalists and specialists. C. Phyla samples proportion.
D. Phyla relative abundance box plots, each dot represents one sample that the phylum occurs.}
    \label{fig:isd_fig2_taxonomy}
\end{figure}

Alpha-diversity (Shannon Index) is significantly differed by physical and chemical features.
Shannon diversity was negatively correlated with elevation (p=0.02), and
positively correlated with both soil moisture and organic carbon (p < 0.001).

\begin{table}[]
    \caption{Taxonomic depth of ASVs and the unique number of taxa of each level.\label{table:asv_taxonomy}}%
\begin{tabular}{@{}lllll@{}}
classification depth & Total ASV    & Total (ASV) taxa & Total OTUs & Total (OTU) taxa\\
Kingdom              & 1974         & 2                & 284        & 2               \\
Phylum               & 4034         & 33               & 121        & 15              \\
Order                & 38517        & 193              & 1224       & 135             \\
Class                & 24157        & 83               & 978        & 62              \\
Family               & 71355        & 287              & 2319       & 218             \\
Genus                & 90137        & 1166             & 1920       & 582             \\
Species              & 9120         & 1338             & 44         & 43              \\
Total                & $\sim$239000 & 3102             & 6890       & 1057            
\end{tabular}
\label{table:asv_taxonomy}
\end{table}

\begin{figure}[h]
      \centering
      \includegraphics[width=\textwidth,height=\textheight,keepaspectratio]{figures/isd_asv_taxonomy_ratios_top_phyla_samples.png}
      \caption[Top phyla of each samples]{The top phyla for all samples with their relative abundance}
      \label{fig:isd_top_phyla_samples}
\end{figure}
   

\subsection{Communities}\label{communities}
Microbial beta diversity differently associated with physical and chemical
features of ISD. PCoA 1 (describing 14\% of the variance) was largely driven by
elevation (F=23, p < 0.001). PCoA 2, which explained 12\% of the variance,
was significantly associated with soil moisture (F=90, p < 0.001).
PCoA 2 also positively correlated with both organic carbon and nitrogen.


\section{Discussion}\label{discussion}

Multiple works have emerged the past 5 years about enumerating all
biodiversity \parencite{Anthony2023} and deciphering the biogeochemical 
processes and interactions of fauna, flora and microbes in global
studies \parencite{Fry2019, Crowther2019,GRANDY201640,Delgado-Baquerizo2020} and
in mountain peaks soil microbiomes \parencite{Adamczyk2019}. One of our profound
results is that soil bacterial biodiversity is very complex, even sites a few meters apart can differ
significantly in their community composition, Figure \ref{fig:isd_site_locations}.
This is a fact that is sometimes neglected in worldwide studies and the island biogeography
paradigm can assist to remove clutter.

Apart from the profound diversity in soils, there is also high speciation and uniqueness. 
As shown in Figure \ref{fig:isd_fig2_taxonomy}, most ASVs occur in 1 or 2 or 3 samples.
This is different when compering with the ocean. When using the deepest taxonomic
level of ASVs is possible to identify the specialists and generalists \parencite{Barberan2012}. 
In addition, focusing on the phyla, we see a pattern looking like a phase transition, from 
rare phyla to phyla that dominate all samples, Figure \ref{fig:isd_fig2_taxonomy} C. Lastly, in Figure \ref{fig:isd_fig2_taxonomy} D,
and \ref{fig:isd_top_phyla_samples}, we found some distinct top phyla profiles of samples.

Elevation gradients of biodiversity are known since Humboldt's work \parencite{Rahbek2019} 
yet these patterns remain elusive regarding the soil microbiome \parencite{Looby2020, Siles2023}.
Mostly because it's difficult to isolate other co-founding effects \parencite{Nottingham2018}.
In our results elevation showed important distinction of taxa and of diversity. Yet more work is 
needed to explore the Asterousia mountain transect in isolation to avoid indirect influences of 
other variables.


\section{Conclusion}

Deciphering and validating the results presented here requires future work.
Even though amplicon studies in soil should be interpreted with caution \parencite{alteio2021} they 
can act as early warning signals towards public health concerns \parencite{banerjee2023Soil}.
In addition, the immediate release and availability of these data is crucial for 
taking action.
The pillar of data integration is the unrestricted open data across disciplines and 
the open source software.
"A holistic perspective on soil architecture is needed as a key to soil functions" \parencite{philippot2024the-interplay}, is 
an important statement for future soil projects.
Shotgun metagenomics and metatranscriptomics can unleash the functional potential of
topsoil along with other advancements like long reads sequencing. Higher resolution
samplings using grid system will enhance the resolution and also the resampling of
ISD sites in different time points will provide additional insights to the complex soil 
functions and expand the positive and negative associations in soils \parencite{Liu2024}.
Lastly, global hotspots \parencite{Guerra2022} and soil ecosystem conservation is needed as 
a whole and expanding current protection of specific species \parencite{guerra2021tracking}.
This along with policy \parencite{KONINGER2022} across countries \parencite{Putten2023}
without borders and the implementation of legislation in Greece \parencite{SCHISMENOS2022100035} is 
imperative.


\section{Data and Code}
The documentation and scripts developed for this study are available in
\href{https://github.com/savvas-paragkamian/crete_soil_microbiome/}{Crete soil microbiome github repository}.
This repository contains all the necessary scripts for the data retrieval,
filtering and ASV inference, taxonomy assignment, data integration of spatial data, 
functional annotation and the subsequent analyses and visualisation.
Code is structured to be reproducible and interoperable.

\section{Competing interests}
No competing interest is declared.

\section{Author contributions statement}
Conceptualization: LS, EP, GK, AM;
Data curation: JH, LS, EP, SY, SP;
Formal Analysis: SP, JH;
Funding acquisition: LS, AM, SP;
Investigation: LS, SY, SP, JH;
Methodology: SP, GK, EP, LS, SY, CC, CP, HZ, MS;
Project administration: LS, EP, GK;
Resources: EP, LS, SY, AM;
Software: SP, JH, HZ;
Supervision: LS, EP, GK, PS;
Validation: JH, CP, CC, HZ, PS, DT, MS, LS;
Visualization: SP, JH;
Writing â€“ original draft: SP;
Writing â€“ review and editing: All authors.

\section{Acknowledgments}
The authors thank the anonymous reviewers for their valuable suggestions.

\section{Funding}
SP was supported by the 3rd H.F.R.I. Scholarships for PHD Candidates (no. 5726) for his work.
