% --------------------------------------------------
% 
% This chapter is for PREGO
% 
% --------------------------------------------------


\chapter{Harmonising the literature and metagenomic resources to infer microbial, environmental and functional relations}
\label{cha:prego}


\section{Introduction}
\label{sec:prego-intro}

The rapid advancements in genomic technology have significantly impacted the
all fields of biology, creating a need for the establishment
of genomic standards to ensure the reliability and reproducibility of research
findings \parencite{Field2011}. As genomic datasets grow exponentially, the importance of these standards
in genomic research cannot be overstated. Standards are essential for enabling
the comparison of results across different studies, facilitating data sharing and
collaboration, and ultimately driving the progression of genomic medicine \parencite{vangay2021microbiome}.
However, the potential of genomic data is often limited by the lack of accompanying
metadata.

Data devoid of metadata are essentially a black box, with no contextual
information to interpret the results or to assess their relevance \parencite{michener_nongeospatial_1997}.
This issue is
prevalent in genomics, where the complexity and diversity of the data
necessitate comprehensive metadata for accurate analysis \parencite{vangay2021microbiome}.
Without proper metadata, it becomes challenging to assess data quality,
reproducibility, and to ensure that research conclusions are robust.
The emergence of metagenomic resources has added a new dimension to genomic
research, with a myriad of databases providing a wealth of data on microbial
communities from various environments.
Examples of public resources are MGnify \parencite{mitchell2020mgnify},
JGI/IMG \parencite{chen2021img} and MG-RAST \parencite{wilke2015restful}). Each one 
has it's one pipeline to analyse user submitted environmental sequences. MGnify
directly imports data from ENA sequence database.
These resources are invaluable for
understanding the genetic diversity and functional potential of microbial
ecosystems. Adding to the complexity, the sheer volume of data and the varying formats in which
they are presented can make comparing and combining these resources a daunting task. 

Metagenomic mining approaches are required to leverage vast databases and analyze
microorganisms from diverse environments \parencite{delmont2011metagenomic}.
Mining allows scientists to extract and interpret genetic material 
from multiple resources thus providing a comprehensive view of the microbiome.
Metadata associated with each sample in these databases, such as source information and collection conditions,
contribute significantly to the contextual understanding and statistical interpretation of the data.
They enable researchers to make meaningful comparisons across different studies.
Furthermore, the application of metagenomic mining to these rich databases can help
uncover novel genes, metabolic pathways, and potential therapeutic targets \parencite{ma2023a-genomic},
thus driving advancements in health, agriculture, and environmental sustainability.
Yet there isn't a cohesive and standard way of comparing metagenomic data and 
metadata because users often publish their data with different types of metadata.

To accomplish harmonisation and comparability of data, named entity recognition techniques have become essential for
efficiently harmonising relevant information from metadata.
Named entity recognition is a term identification process from text 
and is part of the Natural Language Processing field. It is based upon
vocabularies and ontologies that provided multiple versions of terms and 
hierarchy relationships. Environment Ontology (ENVO) \parencite{buttigieg2016environment} 
is an important ontology that has classified environments and metagenomic 
resources suggest it's usage in the metadata in fields like biome and 
environmental feature. Molecular functions and biological processes are two separate 
ontologies, part of Gene Ontology \parencite{ashburner2000gene, gene2021gene} that
are widely used in bioinformatics and are open source. 
Regarding functions and pathways, the KEGG database \parencite{kanehisa2000kegg} is the most used resource
in microbial ecology, yet it requires a licence to implement it in a new resource.
Taxa names are also extracted from free text based on different taxonomies. In microbiology,
NCBI taxonomy \parencite{schoch2020ncbi} is convenient to use and implement because every taxon 
is represented by sequences. Nevertheless, LPSN (List of Prokaryotic names with Standing in Nomenclature) \parencite{parte2020list}
is a curated nomenclature that has high accuracy. Based on some of the aforementioned 
systems it is possible to extract entity names from metadata fields and free text in general 
in order to homogenise metadata and data fields.

Apart from metagenomic data, scientific literature contains invaluable information
about microbes, their functions and habitats. This information is usually hidden 
or captured by experts on the fields. 
Text mining methods have been long assisted in extracting knowledge from 
the scientific literature \parencite{jensen2006Literature}. 
By leveraging text mining, researchers can more readily identify and use the rich information that
is often hidden within the text, thereby accelerating discoveries in this
rapidly evolving field \parencite{badal2019Challenges}. Text mining is a multiple step
process \parencite{10.5555/1199003}. It begins with establishing a corpus with 
documents specifically formatted for the tools used. PubMed \parencite{roberts2001pubmed}
holds more that 40 million abstracts through MEDLINE. These data are accessible 
and open access, therefore useful for text mining applications. A downside is the 
domain of publications because it doesn't cover most of ecological and environmental 
journals. Then these documents are analysed with Named Entity Recognition and a 
score of co-occurrences between entities is produced. This lies in the principle 
that the highest the co-mention of terms the higher the probability there is 
a biological meaning of this association \parencite{jensen2006Literature}.
Text mining approaches have been very successful in unearthing novel 
associations of proteins and deceases \parencite{pletscher2015diseases}.

PREGO, the resource developed and presented in \textcite{microorganisms10020293}
as part of this PhD (see \ref{fig:prego_paper}), is a hypothesis generation platform designed to integrate the 
available knowledge about microbes and their processes and environments. 
It is a global platform, that combines metagenomic data with literature
using an established text mining methodology. 
In this chapter, my contributions in the development of PREGO are presented along with the analysis of global 
soil microbiome knowledge.

\section{Methods}
\label{sec:prego-methods}

   \subsection{Pipeline and structure}
   \label{subsec:prego-pipeline}

PREGO's pipeline can be summarised in 6 steps as presented in Figure \ref{fig:prego-pipeline}.
In steps 1 and 2, the web resources are harvested from 3 different types of data,
literature, environmental samples and genome annotations. 
In step 3, the Named Entity Recognition system of the Jensen Lab Tagger is employed \parencite{jensen2016one}
to identify taxa, environments and processes. In addition, in genome annotations
the processes entities are mapped to Gene Ontology terms with custom scripts. 
During the step 4, the co-occurrences of terms is calculated and scored,
each type of resource has it's own scoring scheme. 
This creates a global association network with taxa, processes and environments as 
nodes, co-occurrences as edges and score values as edge weights. 
Lastly, this knowledge graph is uploaded and provided in a web interface 
and through Application Programming Interface.

   \begin{figure}[hbt!]
      \centering
      \includegraphics[width=0.85\columnwidth]{figures/prego_analysis.png}
      \caption[PREGO analysis methodology]{
         PREGO methodology: retrieval of 3 types of open access data resources, literature, environmental DNA samples, and genomic annotations. 
         Named Entity Recognition and co-occurrences bring together associations from organisms, environments and processes. 
         These associations has a score and are provided is a Web interface and an API. Figure modified version of \parencite{microorganisms10020293}.
      }
      \label{fig:prego-pipeline}
   \end{figure}

The knowledge graph in Figure \ref{fig:prego-pipeline} step 5, contains
nodes categorised in three entity types: \textit{Process}, \textit{Environment}, and \textit{Organism}. 
Organisms, i.e taxa, are the microbial NCBI Taxonomy Ids (Bacteria, Archaea, and unicellular eukaryotes).
All environments are mapped to terms of Environment Ontology. 
Each process corresponds to either a Biological process (GObp) or a Molecular function (GOmf) identifier of Gene Ontology. 

PREGO's knowledge base is structured into three channels, depended on the category of the data type.
\textit{Literature} channel is about the associations extracted by abstracts and full open access text of scientific articles.
The \textit{Annotated Genomes and Isolates} channel contains genome annotations and their harmonised metadata.
The \textit{Environmental Samples} channel integrates metagenomic analyses from amplicon and shotgun sequencing studies. 
Hence, in this channel, the taxonomic and functional profiles and their metadata are harmonised and the associations are extracted.

\subsection{Dictionary of entities}
\label{subsec:prego-dict}

PREGO's entities are types and each term corresponds to specific resource identifier. 
NCBI taxonomy identifiers for taxa, Environment Ontology for environments and
Gene Ontology for Biological Processes (GObp) and Molecular Functions (GOmfs).
Gene Ontology was selected because it has a Creative Commons Attribution 4.0 License
and because it has been mapped to many other resource identifiers.
These terms, their names and synonyms comprise the PREGO dictionary. In addition,
some terms have been removed because of their uniqueness and/or ubiquitousness which reduce the precision of text mining systems.
These terms comprise the blocklist, also mentioned as stopword list, a separate file in the dictionary called prego global.
The ORGANISMS \parencite{pafilis2013species} and ENVIRONMENTS \parencite{pafilis2015environments} are evaluated dictionaries
whereas Gene Ontology Biological Process and Molecular Function are in experimental stage.
The dictionary is 
available for download in the Jensen Lab text mining \href{https://jensenlab.org/resources/textmining/#dictionaries}{resources}. 
NCBI contains all taxonomy, therefore it was filtered for bacteria, archaea and
for the unicellular eukaryotes. Due to unicellular complex taxonomy a manually curated
list was populated with these taxa.

Some resources use KEGG orthology terms or Uniprot50 ids to indicate functions. 
Hence, a mapping KEGG orthology to GOmf and Uniprot50 to GOmf was carried out via UniProtKB mapping files (see \texttt{idmapping.dat} and \texttt{idmapping\_selected.tab} files). 

\subsection{Scoring associations}
\label{scoring}

In the vast knowledge base of PREGO, scoring is crucial to sort 
the results. Ranking the association based on their trustworthiness and
relevance to the user's query is what makes platforms useful to users.
Hence, scoring is an indispensable component of data platforms 
that respond to queries.

PREGO responses to the user's query with associations that have the highest score
through the three channels of information. 
Channels have a different scoring schemes based on the data structure to
rank the associations of terms.
The score of every association across entities and channels is normalised in the interval $(0,5]$.
This transformation unifies the ranking.
Furthermore, the values are rounded, with ceiling function, to be integers
and are represented with stars in the web interface as a visual aid to the user.

Scoring starts with the contingency table (Table~\ref{table:pregoA1}) of $X$ and $Y$.
For example, $X = 1$ is a NCBI id and $Y = 1$ represents an GObp term id. 
The $c_{1,1}$ represents the count of times that these two terms, $X = 1$ and $Y = 1$, 
are mentioned together, i.e. the co-mention frequency. 
The marginals, $c_{1,.}$ and $c{.,1}$ for $x$ and $y$ respectively,
are the frequencies for each entity. 
Different scoring indices are used depending on the channel and the origin of the data.
Each scoring scheme, must be evaluated and benchmarked because there is not a 
single perfect measure. 

\begin{table}[ht]
   \centering
   \begin{tabular}{c|llll}
    & \multicolumn{4}{l}{Y = y} \\ \cline{2-5} 
   \multirow{4}{*}{X = x} &  & Yes & No & Total \\ \cline{3-5} 
    & \multicolumn{1}{l|}{Yes} & $c_{x,y}$ & $c_{x,0}$ & $c_{x,.}$ \\
    & \multicolumn{1}{l|}{No} & $c_{0,y}$ & $c_{0,0}$ & $c_{0,.}$ \\
    & \multicolumn{1}{l|}{Total} & $c_{.,y}$ & $c_{.,0}$ & $c_{.,.}$
   \end{tabular}
   \caption[PREGO contingency table between two terms]{Contingency table of $X = x$ and $Y = y$.}
   \label{table:pregoA1}
\end{table}

The Literature channel scores associations of each pair of entities based on their co-occurrence. 
Literature scores associations depending on the level of co-occurrence, and distinguishes
the sentence level (higher score) from the paragraph level and finally the document level.
Environmental Samples channel scores associations based on abundance tables and metadata. Hence, the number of samples that entities co-occur. 
Thus, it has two levels, microorganisms with metadata, and the sample identifiers.
The Genome Annotation and Isolates channel has predetermined values of scores
based on the resource and the level of manual curation which contain the most trustworthy associations. 

   \subsection{Literature}
   \label{subsec:prego-tm}

   \subsubsection{Corpus}
The \textit{Literature} channel of PREGO contains the associations between 
entities as extracted by PubMeD®. Abstracts and full text articles were retrieved
from MEDLINE® and PubMed Central® Open Access Subset (PMC OA Subset) \parencite{sayers2021database}, respectively. 
NCBI, which hosts these data, has a powerful File Transfer Protocol (FTP) service 
that is used to periodically retrieve new published documents. The download time 
is around 5 hours.
The retrieved files are XML files, compressed. These raw files are the are converted
to tabular format while maintaining the following information:

\begin{itemize}
    \item PMID DOI
    \item Authors
    \item Journal.volume:pages
    \item year
    \item title
    \item Abstract
\end{itemize}

Each file contains ~30000 lines which, sum to 33 million abstracts. Some
abstracts are duplicated across files, so in order to avoid bias in tagging,
a script keeps only the latest occurrence of duplicated abstracts.
The PMC Open Access Subset has two directories: oa comm and oa noncomm.
They are separated because they contain different licences. For commercial usage,
articles in the oa comm directory are allowed because they have CC BY and CC0 licenses.
For non-commercial usage, oa noncomm can be used which has full text documents
licensed under all Creative Commons license types with the exclusion of CC BY and CC0)
and oa comm directories. 

   \subsubsection{Tagger}

Text mining is applied on the compiled corpus, mentioned before,
using the EXTRACT tagger \parencite{pafilis2016extract, jensen2016one}.
The tagger is implemented in C++ and first loads all entities of the dictionary
and then it recognises the co-occurrences in the corpus and returns them with a score. 
The tagger output files, the pubmed tsv files and dictionary files are used by
the 6 perl scripts, Figure \ref{fig:prego-textmining-structure}. More specifically, 
the create documents script reads all the tsv files of pubmed and stores all abstracts
in a single file while simultaneously selecting the latest abstract of the duplicated ones. 
The total duration is 211 minutes. The maximum amount of memory used is 6gb and the tagger
is running using 8 threads.
   
\begin{figure}[hbt!]
      \centering
      \includegraphics[width=0.85\columnwidth]{figures/prego_textmining-structure.png}
      \caption[PREGO analysis methodology]{
         Structure of the Literature channel with multiple scripts working together.
         The initial data are the Dictionary and Pubmed. Each one has preprocessing steps. 
         The Text mining steps are the last steps to create the database pairs file 
         with all associations and scores.
      }
      \label{fig:prego-textmining-structure}
\end{figure}

\subsubsection{Score}

Literature channel adopts the scoring scheme of STRING 9.1 \parencite{franceschini2012string}
and COMPARTMENTS \parencite{binder2014compartments}. This scoring has three steps. 
First, for each co-occurrence of a pair of entities, the following weighted count is
calculated based on the textual level of the co-occurrence:

\begin{equation}
   c_{x,y} = \sum_{k=1}^{n}{w_d \delta_{dk}(x,y) +w_p \delta_{p,k}(x,y) + w_s \delta_{sk}(x,y)}
   \label{eq:prego-score-1}
\end{equation}

Weights of each level of the document ($k$) are used.
$w_d = 1$ is the weight for a co-occurrence at the document
level. 
At he paragraph level, the $w_p = 2$ is used. And at the sentence level, the most
informative level, the weight is $w_s = 0.2$.
The delta functions equal to one (Equation~\ref{eq:prego-score-1})
If the co-occurrence is apparent, the delta functions equal to one (Equation~\ref{eq:prego-score-1}),
and zero otherwise.
Therefore, the weighted count increases as the entities are mentioned 
closer to each other in the text. 
After the counts, the score is calculated with the following equation:

\begin{equation}
   score_{x,y} = c_{x,y}^a (\frac{c_{x,y} c_{.,.}}{c_{x, .}c_{.,y}})^{1-a}
   \label{eq:prego-score-2}
\end{equation}

where $a = 0.6$ is a weight, which has been benchmarked in other 
text mining applications ~\parencite{franceschini2012string, binder2014compartments, pletscher2015diseases}. 
The $c_{x,.}$, $c_{.,.}$, $c_{.,y}$ are the weighted frequencies as shown in Table~\ref{table:pregoA1}.
They are calculated estimated using the same Equation~\ref{eq:prego-score-1}. 

The value of Equation~\ref{eq:prego-score-2} is not robust as
the volume of the corpus increases (MEDLINE PubMed—PMC OA).
To reduce this sensitivity the score is further transformed to $z$-score divided by two.
The maximum value of associations in the Literature channel is 4 out of 5, because 
text mining has many false positives which introduce uncertainties.

   \subsection{Environmental Samples}
   \label{subsec:prego-envsamples}

Environmental Samples channel contains associations extracted by environmental DNA studies, 
namely amplicon and shotgun metagenomic analyses. These data were downloaded with custom API clients
of MGnify \parencite{mitchell2020mgnify} and MG-RAST \parencite{wilke2015restful} repositories.
From these repositories the taxonomic and functional profiles were downloaded along with their 
studies metadata. Both amplicon and shotgun metagenomic have taxonomic profiles, while 
only the latter include also functional profiles.
KEGG Orthology terms and/or Uniref identifiers of functional profiles are mapped to GOmf terms 
prior to the analysis with the custom mapping scripts, see \ref{subsec:prego-dict}.

PREGO processes these data per sample, so per sample the profiles are associated with the metadata. 
To harmonise the data and metadata names, organisms, environments, processes, functions, the EXTRACT tagger and the dictionary is used, as described previously. 
This process creates a entity pairs co-occurrence file for all samples. 
These associations are subsequently scored. The score is based on the count of
samples the entity of interest co-occurs with specific sample metadata or profile (function or taxonomic). 

More specifically, for every term $x$, the number of samples it is present is calculated as the background 
and the count of samples of the co-occurrent term (metadata background) $c.,y$ (see Table~\ref{table:pregoA1}). 
Each association between terms $x$,$y$ is found in $c_{x,y}$ samples. 
These calculations are performed separately for each resource.
The equation of the score is:

\begin{equation}
   score_{x,y} = 2.0*{\frac{\sqrt{c_{x,y}}}{c_{.,y}^{0.1}}}
\end{equation}

Environmental Samples score is asymmetric meaning that the score of $score_{x,y}$ is different from $score_{y,x}$.
This is the case because the denominator is the marginal frequency of the associated entity. 
The score is reduced when $c_{.,y}$ is increasing, i.e., the total number of samples occurrences of $y$. 
So the score increases when the number of samples that the entities co-occur 
approaches the number of samples of the marginal of $y$.
The exponents moderate the increase of the score and they were 
manually set to $0.5$ for the numerator and 
to $0.1$ for the denominator.
Lastly, the value of the score is capped in the range $(0,4]$.

   \subsection{Annotated Genomes and Isolates}
   \label{subsec:prego-isolates}

Annotated genomes and isolates are the manually curated data of PREGO's knowledge base.
This fact makes this channel the most trustworthy because every single species/strain have manually curated metadata. 
JGI-IMG \parencite{chen2021img, mukherjee2021genomes} has millions of genes that 
correspond to isolated genomes, SAGs and MAGs. These annotations and their corresponding metadata
were collected using web scrapping because even though the data are open there isn't an API client for these data.
The metadata about environments were harmonised using the EXTRACT tagger leading to organisms—environments co-occurrences.
The KEGG ids from the annotations were mapped to GOmf terms and were used for the organisms—processes associations.
   
The Struo pipeline \parencite{de2020struo}, which is based on the Genome Taxonomy DataBase (GTDB) (v.03-RS86) \parencite{parks2020complete} was also used.
This pipeline contains organisms—processes associations. 
UniRef50 annotations were changed to GOmf terms with custom mapping. 
GTDB genome taxonomy was mapped to the corresponding NCBI taxa. 
Struo's associations were assigned a confidence level of four out of five. 
This score reflects the high-quality of these data and metadata.
   
Lastly, BioProject data were used in PREGO from the NCBI FTP/e-utils services \parencite{sayers2021database}. 
The BioProject ids that has been assigned a PubMed abstract, a unicellular taxon, and Genome sequencing as data type were filtered.
The text mining pipeline described above, was used to extract associations of the
assigned taxon with the other entities in the abstracts. 
The BioProject derived associations were assigned the score of three (out of five)
because of the implementation of the text mining pipeline.


\subsection{Soil use case}
\label{soil-prego-m}

PREGO's knowledge base was searched for it's contents about soil environments. 
The dictionary of PREGO was first searched for the ENVO term of soil, ENVO:00001998,
and then all the child terms were also retrieved. Using this list of 
ENVO terms about soil, all channels of information were filtered, summarised and 
analysed for patterns as a use case. ARENA3D Web \parencite{karatzas2021arena3dweb} and
igraph \parencite{Csardi2006} were used for network analysis.

\subsection{Behind the scenes}
\label{deamons}

The pipeline of PREGO is streamlined and integrated to be executed regularly,
spanning from once a month to six-month cycles, ensuring that changes of resources are
continuously integrated into PREGO. These cycles are called daemons in PREGO and 
each channel has a dedicated script that executes all the code while 
maintaining backups. Daemons provide a streamlined method of executing all steps 
in PREGO as in Figure \ref{fig:prego-pipeline}.

PREGO's pipeline and hosting is running on a 64 GB RAM DELL R540, 20 core, Debian server.
The server has 2 terabyte SSD storage for high responsiveness. 
All code is versioned with the git versioning tool to keep history and 
resolve bugs. The development cycles of PREGO are based on DevOps practices that
combine development and operations and enable the efficient deployment of software services.
The Continuous Development and Continuous Integration (CD/CI) method proved to be
important aspect during the developed of PREGO in terms of multiple testing,
resolving bugs and effective collaborative coding. 

%\begin{figure}[hbt!]
%   \centering
%   \includegraphics[width=95mm]{figures/figure_A1_PREGO_daemons.png}
%   \caption[PREGO DevOps]{Scripts called daemons are executing the PREGO methodology in cycles depending on the updates of the databases used. Figure adopted by \parencite{microorganisms10020293}.}
%   \label{fig:devops}
%\end{figure}


   \subsection{Code} 
   All code of PREGO is available in the following repositories as shown in Table \ref{tab:prego-repos}.
   Code produced for PREGO are available under BSD 2-Clause “Simplified” License.
Scripts, where third party libraries have been used, are subject to their individual licenses.
API calls were implemented in Python 3. Daemons and FTP downloads were written in Bash and 
mappings and statistics in GNU AWK.

\begin{table}[]
\centering
\caption{The repositories of PREGO. Each repository corresponds to a module of the platform.}
\label{tab:prego-repos}
\resizebox{\textwidth}{!}{%
\begin{tabular}{lll}
    \textbf{Task}           & \textbf{Repository} & \textbf{Licence}                   \\
gathering data & \href{https://github.com/lab42open-team/prego_gathering_data}{github.com/lab42open-team/prego\_gathering\_data}        & BSD 2-Clause “Simplified” \\
daemons        & \href{https://github.com/lab42open-team/prego_daemons}{github.com/lab42open-team/prego\_daemons}                       & BSD 2-Clause “Simplified” \\
mappings       & \href{https://github.com/lab42open-team/prego_mappings}{github.com/lab42open-team/prego\_mappings}                     & BSD 2-Clause “Simplified” \\
use cases      & \href{https://github.com/lab42open-team/prego_use_case}{github.com/lab42open-team/prego\_use\_case}                    & BSD 2-Clause “Simplified” \\
statistics     & \href{https://github.com/lab42open-team/prego_statistics}{github.com/lab42open-team/prego\_statistics}                 & BSD 2-Clause “Simplified” \\
tagger         & \href{https://github.com/larsjuhljensen/tagger}{https://github.com/larsjuhljensen/tagger}                              & BSD 2-Clause “Simplified” \\
mamba          & \href{https://github.com/larsjuhljensen/mamba}{https://github.com/larsjuhljensen/mamba}                                & BSD 2-Clause “Simplified” \\
dictionary     & \href{https://download.jensenlab.org/prego_dictionary.tar.gz}{https://download.jensenlab.org/prego\_dictionary.tar.gz} & CC-BY 4.0 license        
\end{tabular}%
}
\end{table}  


\section{Results}
\label{sec:prego-results}

   \subsection{PREGO Contents}
   \label{subsec:prego-contents}

The data sources of PREGO can be categorized into six types: abstracts, full articles,
isolates, annotated genomes, markergene samples, and metagenomic samples, Figure \ref{fig:prego-summary}.
In terms of metadata availability, JGI IMG, Struo, BioProject, and MG-RAST provide
metadata for their respective data types.
BioProject provides metadata for annotated genomes with abstracts, which could be useful for researchers seeking comprehensive information.
The number of items varies significantly across sources, with MEDLINE and PubMed
having the largest collection of items (33 million) and PubMed Central OA Subset
having a significant collection of full articles (2.7 million).
JGI IMG, Struo, BioProject, and MG-RAST have a smaller but still substantial number of isolates, annotated genomes, and markergene samples.
In terms of licenses, the majority of the sources have open licenses,
such as CC0, CC BY-SA 4.0, CC-BY, and NLM Copyright, which allows for the free
use and sharing of the data. This is an essential aspect of data sharing and
collaboration in the scientific community.

   \begin{figure}[hbt!]
      \centering
      \includegraphics[width=\columnwidth]{figures/PREGO_summary_channels.png}
      \caption[PREGO contents summary]{
         Resources, entities and associations summary in each channel of information of PREGO.}
      \label{fig:prego-summary}
   \end{figure}

Regarding the entity types, PREGO's knowledge base contains 364,000 taxa
(NCBI Taxonomy has 620,000 Bacteria, Archaea, and microbial eukaryotes). 
About 258,000 taxa are at the species level. 
All environment Ontology terms are found at about 1000 terms. Regarding Gene ontology, 15,000 biological process terms and 7.900 molecular function term are present.
PREGO's knowledge base of entities and associations are a multipartite network
with entities as nodes and co-occurrences as links with score as weight.

\subsection{Bulk download}
\label{bulk-download}

All the knowledge base of PREGO is available for download programmatically.
One file is provided per channel, along with md5sum files for sanity checks.
Data are compressed to tar.gz format so expect an order of magnitude higher file size when decompressed.
Literature channel data are available \href{https://prego.hcmr.gr/download/literature.tar.gz}{here}
at 5.4 gb size, \href{https://prego.hcmr.gr/download/literature.tar.gz.md5}{literature.tar.gz.md5} is the md5sum file.
Environmental samples are 0.69 gb compressed and available \href{https://prego.hcmr.gr/download/environmental\_samples.tar.gz}{here},
with \href{https://prego.hcmr.gr/download/environmental\_samples.tar.gz.md5}{environmental\_samples.tar.gz.md5} as well.
Annotated genomes tar.gz is 0.26 gb in size, download \href{https://prego.hcmr.gr/download/annotated\_genomes\_isolates.tar.gz}{here}, and 
the md5sum file \href{https://prego.hcmr.gr/download/annotated\_genomes\_isolates.tar.gz.md5}{here}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% soil %%%%%%%%%%%%%%%%%%%%%%%%%%%%
   \subsection{PREGO about soil}
   \label{subsec:prego-soil}

Searching for an environment in PREGO's knoweldge base is possible through
it's web interface, it's API and bulk data. This analysis used PREGO's bulk data. 
In total, the dictionary contains 115 ENVO terms related to soil. These terms were
used to filter associations of environments with organisms, processes and functions 
from all channels. This search resulted in 207939 PREGO associations. 
To keep the most trustful associations and to be comparable with the Chapter \ref{cha:isd-crete-soil}
filters were applied. In Literature channel, only associations with score 4.8 or higher were kept. 
In Environmental Samples and Annotated Genomes associations with score of 3 or higher were kept. 
In addition, only taxa from species and strain taxonomic level were selected.
This filtering led to 14,379 associations, 599 from Literature, 7003 from Environmental
Samples and 6777 from Annotated Genomes.

In total, 49 ENVO terms were associated with 6276 taxa, 169 biological processes
and 3017 molecular functions. The network was further reduced to keep the 
biggest connected component. 
The majority of associations, 10,929, were between environments and organisms.
Environments with molecular functions have 3139 associations, followed by 
environments with biological processes with 274 associations, Figure \ref{fig:prego-soil-network} and \ref{fig:prego-soil-network_k}.

The soil term (ENVO:00001998) is the parent of the rest ontology terms and has 11694 associations.
Loam (ENVO:00002258) has 1222 associations followed be
rhizosphere (ENVO:00005801) with 393 associations.
Forest soil (ENVO:00002261) and meadow soil (ENVO:00005761) are important terms
that have 300 associations and 226 associations, respectively.

   \begin{figure}[hbt!]
      \centering
      \includegraphics[width=\columnwidth]{figures/prego_soil_arena.png}
      \caption[PREGO soil network of literature and environmental samples]{
         The four layers of associations, -27 is environments, -2 is organisms, -21 is biological processes, -23 is molecular functions.
     The nodes that are green and labeled are the with the most associations,
     soil (ENVO:00001998), loam (ENVO:00002258), rhizosphere (ENVO:00005801), forest soil (ENVO:00002261), meadow soil (ENVO:00005761). 
 With gray are displayed the Literature derived associations while with orange the Environmental samples. }
      \label{fig:prego-soil-network}
   \end{figure}
   

   \begin{figure}[hbt!]
      \centering
      \includegraphics[width=\columnwidth]{figures/prego_soil_arena_knowledge.png}
      \caption[PREGO soil network of associations of annotated genomes]{
         Associations of Annotated Genomes in PREGO, -27 is environments, -2 is organisms. Notable nodes are soil (ENVO:00001998), contaminated soil (ENVO:00002116), loam (ENVO:00002258), rhizosphere (ENVO:00005801), forest soil (ENVO:00002261), meadow soil (ENVO:00005761).}
      \label{fig:prego-soil-network_k}
   \end{figure}

Regarding taxonomy, archaea have 122 taxa and 7 phyla, bacteria have 5985 taxa
in 119 phyla and eukaryotes have 224 in 5 phyla, Figure \ref{fig:prego-soil-phyla}.

   \begin{figure}[hbt!]
      \centering
      \includegraphics[width=\columnwidth]{figures/prego_soil_phyla_summary.png}
      \caption[PREGO soil taxonomy summary]{
         Phyla associated with soil in the PREGO knowledge base.}
      \label{fig:prego-soil-phyla}
   \end{figure}


\section{Discussion}
\label{sec:prego-discussion}

   \subsection{One stop shop}
   \label{subsec:prego-contents-disc}

PREGO approach is based upon four principles. First, open access data and
metadata integration along with literature are the pillar of the current
available knowledge to microbiologists. Second, the associations of microorganisms,
environments and processes are key for the deciphering of ecosystem function.
Third, the modular structure and regular cycles of updates are fundamental to
keep up with the rapid advances of microbial ecology. And last but not least,
PREGO is implemented to be easy to use and openly accessible to all with it’s graphical and programming user interface.

The harmonisation of information from multiple types of resources and of different
entities is a major issue in science today. PREGO approached this problem from 
the Natural Language Processing, data integration and statistics perspective offering a global integration 
of microbial knowledge. To accomplish this some strategic choices were made, 
namely using the same dictionary of terms for all resources and used 
the EXTRACT tagger to identify the terms in free text and metadata.
PREGO manages a great proportion of microbial taxa, even unicellular eukaryotes, in it's knowledge base.
Because of the different channels, PREGO extracts associations in the species and higher taxonomic levels.
Higher level taxonomic associations with environments and processes are present
which are useful for classification processes and versatility of taxa hypotheses.

PREGO provides all this knowledge base in multiple 
ways and is implemented to be easy to use and openly accessible to all with it’s graphical and programming user interface.
A dedicated web interface is useful to users for single queries
and exploration of literature and/or samples. The API is useful and 
robust for multiple calls and, not for the faint of heart, PREGO is 
available for bulk download, section \ref{bulk-download}. 


   \subsection{Platform comparison}
   \label{subsec:prego-similar-platforms}


Recently, data integration platforms are being developed.
Some notable examples are BacDive \parencite{reimer2021bacdive},
Web of Microbes \parencite{kosina2018web}, SPIRE \parencite{schmidt2023spire} and NMDC \parencite{eloe-fadrosh2022the-national}.
Because the field is still growing there multiple design and functionality 
differences of these platforms. Yet some important criteria are the following:
manual curation, literature integration, environment—taxa associations, environment—process associations,
data originality, process/function—taxa associations,
phenotypic data, spatial coordinates, application programming interface, bulk download.

BacDive seems to be the most versatile
platform yet, offering a wide range of features. It has a high level
of manual curation, with environment-taxa associations, process/function-taxa
associations, and phenotypic data available. Additionally, BacDive provides
spatial coordinates, an application programming interface but limited bulk download of data.
It has a high level of manual curation, indicating that it is well-maintained and regularly updated with accurate information.
As for the presence of various types of data, BacDive appears to be the platform
with the most comprehensive dataset, offering original and integrated data from
various sources. Its high scores for environment-taxa associations,
process/function-taxa associations, and phenotypic data suggest that it contains a broad range of microbial data.

Web of Microbes is a manually curated web database that focuses on original data, mostly from metabolomics.
The environment is defined based on the chemistry and metabolites. It contains taxa associations 
with functions. Users interact through the 
web interface and bulk download of the data. NMDC is an integrative platform 
that also hosts original data. It is designed to make data FAIR from 
experiments and integrating bioinformatics workflows as well. It doesn't 
host phenotypic data nor environments to processes associations. 
Also it has spatial coordinates and interactive maps to show samples. 
It is a major collaborative effort that has the state of the art user interface
and quality control systems, therefore providing the users a great platform 
to upload their -omics data for hosting and bioinformatic pipelines.
SPIRE is the newest such platform \parencite{schmidt2023spire}. 
It integrates environmental metagenomic samples after manual curation 
of their metadata. It contains environment-taxa associations and process/function-taxa associations.
The data are available through API and bulk download. 

PREGO appears to have some notable benefits. Specifically, it has a high score
for literature integration, which suggests that it is able to integrate microbial
data from various sources and publications. Additionally, PREGO is capable of
processing and storing data on environment-process/function associations, which
could be useful for researchers studying microbial processes and functions.
Another benefit of PREGO is its ability to facilitate bulk downloads of data,
making it easier for researchers to access and analyze large datasets.
This feature can be particularly useful for big data analysis and machine learning applications.
It is worth noting that PREGO's scores for other features, such as manual
curation and environment-taxa associations, are relatively low compared to
BacDive and Web of Microbes. However, its strengths in literature integration
and bulk data download may make it a useful platform for researchers with specific needs.

There are some caveats with PREGO's approach. For example, the named entity recognition 
is prone to identifying terms in cases where they have different meaning, i.e false positives.
To reduce this effect of the ambiguity of some words a blocklist is constantly 
updated in the dictionary. These errors are also propagated to the associations 
but through the scoring scheme and their lower abundance they usually have low ranks.
The reverse is also a major issue in most text mining application, i.e how many 
actual terms and associations are not retrieved? These are the false negatives.
These are important limitations because the context of the text is ignored.
Large Language Models are promising tools to add context and minimise these issues \parencite{doi2024biodiversity}.

Additionally, PREGO is bound to the dictionaries it uses.
This is an important asset because it provides an ecosystem of tools
and ontologies that is compatible. On the other hand, NCBI taxonomy 
for example, is one of the taxonomy nomenclatures used by microbiologists \parencite{sanford2021microbial}.
Regarding processes, KEGG orthology terms are the norm because of the tight 
integration to pathways, whereas PREGO uses Gene Ontology terms.

   \subsection{Soil microbiome}
   \label{subsec:prego-soil}

Soil is the most prominent environment in terrestrial ecosystems. Yet 
biological information about the microbes and their processes remains 
scattered. PREGO's knowledge base and interface provides a unique 
approach towards the harmonisation of this information. As shown 
in Figure \ref{fig:prego-soil-phyla} the information is scattered in multiple ways.
The source of information, i.e the channels of PREGO, is a major barrier of unification and
secondly a important limitation is the taxonomic barrier, i.e
different super-kingdoms of microbes, because different kingdoms have different experts.
Euryarchaeota and Crenarchaeota are phyla from
Archaea and 
small eukaryotes are mostly represented by the fungi phylum Ascomycota. Proteobacteria
and Actinobacteriae are dominating soil environments (Figure \ref{fig:prego-soil-phyla}),
global environments in general as shown in \textcite{microorganisms10020293} 
and Crete island soils (Figure \ref{fig:isd_fig2_taxonomy} of Chapter \ref{cha:isd-crete-soil}).
The abundance of taxa of Ascomycota indicates the importance of their interactions 
with bacteria and archaea which only recently has been explored \parencite{Labouyrie2023} in scale. 
There is an absence of Environmental samples about Ascomycota, but the Literature and Annotated genomes 
channel is evidence of their importance.
In addition, the are many \textit{Candidatus} phyla because these microbes are not 
cultured yet. These microbes that are in \textit{Candidatus} status are difficult to 
be retrieved by text mining and score lower, Figure \ref{fig:prego-soil-phyla}.
The soil case study of PREGO shows the importance of data integration techniques
and metadata in order to identify gaps in knowledge and discover what is known.


