% --------------------------------------------------
% 
% This chapter is for PREGO
% 
% --------------------------------------------------


\chapter{Harmonising the literature and metagenomic resources to infer microbial, environmental and functional relations}
\label{cha:prego}


\textbf{Citation:} \\ 
Zafeiropoulos H, Paragkamian S, Ninidakis S, Pavlopoulos GA, Jensen LJ, Pafilis E. PREGO: A Literature and Data-Mining Resource to Associate Microorganisms, Biological Processes, and Environment Types. Microorganisms. 2022; 10(2):293. \\ 
Shared co-first authorship.

\textbf{Journal:} Microorganisms

This article belongs to the Special Issue Selected Papers from the 9th Conference of the Hellenic Scientific Society MIKROBIOKOSMOS \textit{Beneficial Microbes at the Heart of Mikrobiokosmos}

DOI: \href{https://doi.org/10.3390/microorganisms10020293}{10.3390/microorganisms10020293}\footnote{
   For author contributions and supplementary material please refer to the relevant sections. 
   This is a modified version of the published version,
   in terms of relevance, coherence and formatting.}



% PREGO ABSTRACT
%\section{Abstract}
%
%   To elucidate ecosystem functioning, it is fundamental to recognize \textit{what} processes occur in which environments (\textit{where}) and which microorganisms carry them out (\textit{who}). 
%   Here, we present PREGO, a one-stop-shop knowledge base providing such associations. 
%   PREGO combines text mining and data integration techniques to mine such what-where-who associations from data and metadata scattered in the scientific literature and in public omics repositories. 
%   Microorganisms, biological processes, and environment types are identified and mapped to ontology terms from established community resources. 
%   Analyses of comentions in text and co-occurrences in metagenomics data/metadata are performed to extract associations and a level of confidence is assigned to each of them thanks to a scoring scheme. 
%   The PREGO knowledge base contains associations for 364,508 microbial taxa, 1090 environmental types, 15,091 biological processes, and 7,971 molecular functions with a total of almost 58 million associations. 
%   These associations are available through a web interface (\href{https://prego.hcmr.gr}{https://prego.hcmr.gr}), an Application Programming Interface (API), and bulk download. 
%   By exploring environments and/or processes associated with each other or with microbes, PREGO aims to assist researchers in design and interpretation of experiments and their results. 
%   To demonstrate PREGO's capabilities, a thorough presentation of its web interface is given along with a meta-analysis of experimental results from a lagoon-sediment study of sulfur-cycle related microbes.
%
% RPEGO INTRODUCTION
\section{Introduction}
\label{sec:prego-intro}

The rapid advancements in genomic technology have significantly impacted the field of biology and medicine, creating an imperative need for the establishment of genomic standards to ensure the reliability and reproducibility of research findings. As genomic datasets grow exponentially, the importance of these standards in genomic research cannot be overstated. Standards are essential for enabling the comparison of results across different studies, facilitating data sharing and collaboration, and ultimately driving the progression of genomic medicine (PMID:21713030).

However, the potential of genomic data is often limited by the lack of accompanying metadata. Data devoid of metadata are essentially a black box, with no contextual information to interpret the results or to assess their relevance. This issue is especially prevalent in genomics, where the complexity and diversity of the data necessitate comprehensive metadata for accurate analysis (PMID:21552244). Without proper metadata, it becomes challenging to assess data quality, reproducibility, and to ensure that research conclusions are robust.

The emergence of metagenomic resources has added a new dimension to genomic research, with a myriad of databases providing a wealth of data on microbial communities from various environments. These resources are invaluable for understanding the genetic diversity and functional potential of microbial ecosystems. However, the sheer volume of data and the varying formats in which they are presented can make mining these resources a daunting task. To overcome this, text mining techniques have become essential for efficiently extracting relevant information from the vast literature on metagenomics. By leveraging text mining, researchers can more readily identify and use the rich information that is often hidden within the text, thereby accelerating discoveries in this rapidly evolving field (PMID:16418747).


Old intro

Microbes are omnipresent and impact global ecosystem functions \parencite{falkowski2008microbial} through their abundance \parencite{bar2018biomass}, versatility \parencite{delgado2016microbial}, and interactions \parencite{rottjers2018hairballs}. 
   These facts have inspired microbiologists from diverse scientific fields to study their genotype and phenotype \parencite{morris2020linking}, their metabolism \parencite{biggs2015metabolic}, and their interactions with the environment \parencite{hall2018understanding}. 
   All this work has resulted in a wealth of knowledge available in the forms of literature and experimental data. Literature contains vast amounts of information in the free text form that overwhelms researchers. 
   Advanced text mining methods \parencite{jensen2006Literature} have been developed to assist this issue. 
   Experimental data and their metadata require mining \parencite{delmont2011metagenomic} as well for their integration, mostly through metagenomic mining from online repositories. 
   Hence, the combination of this knowledge about microbial life (who), their metabolic functions (what), and the environment they influence (where) is an important step to study ecosystem function \parencite{raes2008Molecular}.

   High Throughput Sequencing (HTS) has turned the page on microbial ecology studies \parencite{nilsson2019mycobiome}. 
   Over the past 20 years, both the taxonomic and the functional profiles of microbial communities from both local and large-scale regions (e.g., Tara Oceans \parencite{pesant2015open}, Earth Microbiome \parencite{gilbert2014earth}) are being accumulated at a higher and higher rate. 
   Extreme environments, i.e., areas with high salinity, low pH, etc., are being studied, providing us with unprecedented insight \parencite{shu2021microbial}. 
   Both amplicon and shotgun metagenomics studies have played a crucial part in this development. Latest technological breakthroughs, such as Metagenome-Assembled Genomes (MAGs) and Single Amplified Genomes (SAGs), are enhancing the assessment of the taxonomic and functional repertoire of microbiomes even further. 
   However, the mass use of these technologies and their consequent data have led to a number of needs and challenges, with metadata curation being among the most crucial ones.

   Standards-promoting communities, like \href{https://gensc.org/}{Genomic Standards Consortium (GSC)}\footnote{
      \href{https://gensc.org/}{https://gensc.org/}
   }, their efforts, like Minimum Information about any (x) Sequence (MIxS) \parencite{yilmaz2011minimum}, and projects endorsing those, like National Microbiome Data Collaborative (NMDC) \parencite{wood2020national, vangay2021microbiome}, offer guidelines and best-practices to assist the annotation of microbial ecology samples. 
   Controlled vocabularies and ontologies contribute to these efforts as they describe each subject area with formal terms \parencite{walls2014semantics}. 
   Environment types, for example, are described by the Environment Ontology (ENVO) \parencite{buttigieg2016environment}. 
   Other key biological aspects that have been captured include molecular functions (Gene Ontology Molecular Function (GOmf) \parencite{ashburner2000gene, gene2021gene}, Enzyme Commission nomenclature \parencite{noauthor_1999}, etc.), and the pathways carrying out different biological processes (GO Biological Process (GObp), MetaCyc \parencite{caspi2020metacyc}, etc.). 
   These knowledge structures, along with taxonomic centralized resources like the National Center for Biotechnology Information (NCBI) Taxonomy \parencite{schoch2020ncbi}
   and LPSN (List of Prokaryotic names with Standing in Nomenclature) \parencite{parte2020list}, 
   provide the means for a standardized representation of, for example, environments, process-oriented terms, and microbial taxa, respectively. Global-scale public resources (like MGnify \parencite{mitchell2020mgnify}, 
   JGI/IMG \parencite{chen2021img}, 
   MG-RAST \parencite{wilke2015restful}) 
   combine some of the aforementioned resources to support the collection, analysis, and distribution of multiple types of HTS data (e.g., amplicon, metagenomics, metatranscriptomics, etc.).

   Besides the data and the analyses \textit{per se}, the related scientific literature stores valuable information in billions of text lines. 
   PubMed \parencite{schoch2020ncbi} and PubMed Central (PMC) \parencite{roberts2001pubmed} are gateways to relationships among microbes (\textit{who}), the environments they live in (\textit{where}) and their associated processes and functions (\textit{what}) hidden in text \parencite{harmston2010papers}. 
   Text mining (on both literature and metadata) can serve the extraction of these relationships. Named Entity Recognition (NER) can, for example, locate organism names \parencite{pafilis2013species}, ENVO and GO terms \parencite{pafilis2016extract} mentioned in text and map them to their corresponding identifiers. 
   Association statistics, like co-mention analysis, can subsequently suggest ranked association among such entities \parencite{von2005string, franceschini2012string}. 
   The new era of omics has been interwoven with data integration \parencite{gomez2014data} by bringing together scattered and fragmented pieces of information.

   The time is ripe for tools that integrate all this knowledge and henceforth assist researchers to tackle major challenges like climate change \parencite{cavicchioli2019scientists}, 
   sustainability \parencite{d2021microbiome}, 
   and synthetic ecology \parencite{conde2020synthetic}.
   Many resources have emerged in this realm \parencite{baltoumas2021biomolecule}, 
   each one serving a specific purpose, such as BacDive \parencite{reimer2019bac}. 
   BacDive is a large-scale curated database with prokaryotic information about phenotypic, morphological, and metabolic information. 
   Other resources like Microbe Directory \parencite{shaaban2018microbe}, 
   Web of Microbes (WoM) \parencite{kosina2018web}, 
   and Microbial Interaction Network Database (\href{http://www.microbialnet.org/mind_home.html}{MIND}\footnote{
      \href{http://www.microbialnet.org/mind_home.html}{http://www.microbialnet.org/mind\_home.html}
   }) focus on microbial environmental conditions, metabolite interactions with microbes and microbe-microbe interactions, respectively. 
   In addition, taking advantage of aforementioned resources, novel pipelines, e.g., \parencite{tang2020tripartite}, are emerging with the aim to explore the network associations of who (microbial taxa) is performing what (microbial processes) and where (environments) directly using graph theory \parencite{koutrouli2020guide}. 
   These analyses and resources are important because microbiologists can enrich their data to explore hypotheses but also to identify potential gaps in knowledge regarding these associations \parencite{li2021microbial}.

   Here, we present PREGO, a hypothesis generation web resource that is designed to be useful to microbiologists—in particular microbial ecologists and environmental microbiologists. 
   Its specific aims include: 
   (a) the gathering of source data, metadata, and literature followed by the extraction of microorganism, process, environment associations contained therein, 
   (b) making such a mined knowledge base available to life sciences researchers via an easy to use and explore web portal. 
   As such, PREGO can be useful also to system microbiologists and large-scale data analysts through bulk download and programming access. 
   We document the principles, analysis methodology, and contents behind PREGO. 
   Last but not least, we demonstrate PREGO's capabilities for researcher-support related to the above through a case study involving sulfate-reducing microorganisms.



% RPEGO METHODS
\section{Methods \& Implementation}
\label{sec:prego-methods}

   PREGO is a resource designed to assist molecular ecologists in acquiring a single point overview of what-where-who process–environment–organism associations. The system is comprised of two main parts: (a) a server that periodically harvests data and extracts process-environment-organism associations from the scientific literature, environmental samples, and genome annotation sequences (Figure~\ref{fig:prego-pipeline}, step 1 to 5) and (b) a web-based interface as well as an Application Programming Interface (API) that provides users and programmers with a friendly way to extract and navigate across the process–environment–organism associations (Figure~\ref{fig:prego-pipeline}, step 6).

   \begin{figure}[h]
      \centering
      \includegraphics[width=0.98\columnwidth]{figures/prego_analysis_horizontal.png}
      \caption[PREGO analysis methodology]{
         PREGO analysis methodology: PREGO periodically retrieves three distinct types of data from open access resources. 
         Scientific text, environmental sample data, and genomic annotations are handled with respective methodologies in order to standardize their entities. 
         Named Entity Recognition and Comention/Co-occurrence analysis is the common framework in order to build a weighted association network with nodes being the entity identifiers. Lastly, all these associations are available through a Web interface and an API. 
         All these steps have been implemented in an autonomous way with regular cycles of updates (see Appendix~\ref{app:B}). 
         Icons used from the Noun Project released under CC BY: Books by Shakeel Ch., Bacteria by Maxim Kulikov, ftp by DinosoftLab, Mountain by Diane, Ship on Sea by farra nugraha, River by Chanut is Industries.
      }
      \label{fig:prego-pipeline}
   \end{figure}



   \subsection{Entity Types, Channels, and Associations}
   \label{subsec:prego-terms}

   PREGO supports three entity types: \textit{Process}, \textit{Environment}, and \textit{Organism}. 
   For interoperability and consistency, an ontology or taxonomy is adopted for each type of entity.
   Processes are represented as Gene Ontology (GO) terms and are grouped either as Biological processes (GObp) or as Molecular functions (GOmf). 
   In addition, Environments are represented by terms from the Environmental Ontology. 
   Organisms are represented by the microbial NCBI Taxonomy Ids (Bacteria, Archaea, and unicellular eukaryotes). For the unicellular eukaryotes, a custom list was populated with the unicellular eukaryotic taxa using a curated list.
   PREGO's contents are mainly divided into three distinct channels of information based on data origin and format (Figure~\ref{fig:prego-pipeline}, step 1). 
   The \textit{Literature} channel exploits scientific publications, i.e., abstracts and full text open access scientific publications (Table~\ref{table:prego1} and Section~\ref{subsec:prego-tm}). 
   Through the \textit{Annotated Genomes and Isolates} channel, PREGO retrieves genome annotations and their accompanying metadata (Table~\ref{table:prego1} and Section~\ref{subsec:prego-isolates}). 
   Finally, the \textit{Environmental Samples} channel supports the integration of metagenomic analyses from both amplicon and shotgun studies. 
   These include taxonomic and functional profiles along with their corresponding metadata (Table~\ref{table:prego1}, more details in Section~\ref{subsec:prego-envsamples}).


   % SOURCE DATABASES TABLE table:prego1
   \begin{table}[ht]
      \begin{tabular}{@{}ccccc@{}}
      \toprule
      \textbf{Source} & \textbf{\# items} & \textbf{Data type} & \textbf{Metadata} & \textbf{License} \\ \midrule
      \begin{tabular}[c]{@{}c@{}}MEDLINE and \\ PubMed\end{tabular} & 33 million & abstracts (text) & no & NLM Copyright \\
      \begin{tabular}[c]{@{}c@{}}PubMed Central \\ OA Subset\end{tabular} & 2.7 million & full article (text) & no & \begin{tabular}[c]{@{}c@{}}CC for Commercial, \\ non-commercial\end{tabular} \\
      JGI IMG & 9,644 & \begin{tabular}[c]{@{}c@{}}Isolates Annotated \\ genomes\end{tabular} & yes & JGI Data Policy \\
      Struo & 21,276 & Annotated genomes & no & MIT, CC BY-SA 4.0 \\
      BioProject & 18,752 & \begin{tabular}[c]{@{}c@{}}Annotated genomes \\ with abstracts (text)\end{tabular} & yes & INSDC policy \\
      \multirow{2}{*}{MG-RAST} & 16,096 & markergene samples & yes & CC0 \\
      & 7,965 & metagenomic samples & yes & CC0 \\
      MGnify & 10,500 & markergene samples & yes & CC-BY, CC0 \\ \bottomrule
      \end{tabular}
      \caption[Source databases integrated in PREGO and the number of items retrieved]{
         Source databases that are integrated in PREGO and the number of items retrieved. 
         The Open Access subset of PubMed Central has a Creative Commons license available for commercial and noncommercial use. 
         JGI has its own license, the same applies for BioProject, MEDLINE®, and PubMed® as well.
         }
         \label{table:prego1}
   \end{table}


   In cases in which the retrieved data and metadata are in text form, they are standardized to the aforementioned identifiers and taxonomies using Named Entity Recognition (NER) tools, namely the EXTRACT tagger \parencite{pafilis2016extract, jensen2016one}. 
   In cases where data contain KEGG Orthology terms and/or Uniref identifiers, they are mapped to the respective GOmf using the mapping files available from the UniProt). 
   Associations are extracted after the mapping and standardization of the entities from each resource (Figure~\ref{fig:prego-pipeline}, step 3).
   The association extraction pipeline is distinct for each channel and resource because of differences in the data type origin (see \texttt{prego\_gathering\_data} in the Availability of Supporting Source Codes section). 
   By the means of navigation, the large number of associations returned to the user require a type of sorting; 
   ideally, one that ranks the most trustworthy associations at the top. 
   For those reasons, each channel of PREGO has a dedicated scoring scheme bounded within the (0,5] space for consistency. 
   In Appendix, the scoring scheme of each channel is elaborated.




   \subsection{Text Mining of Scientific Literature}
   \label{subsec:prego-tm}

   PREGO implements a text mining methodology to extract associations of the aforementioned entities from literature. 
   The origin of text mining is a corpus that comprises scientific abstracts and full text articles from MEDLINE® and PubMed® and PubMed Central® Open Access Subset (PMC OA Subset) \parencite{sayers2021database}, respectively. 
   The building and periodic update of the corpus is possible through the NCBI File Transfer Protocol (FTP) services. 
   PREGO also has a dedicated text-mining dictionary (see Availability of Supporting Source Codes section) that contains the entities ids, names, synonyms, and neglected words (stop words). 
   PREGO dictionary incorporates the ORGANISMS \parencite{pafilis2013species} and ENVIRONMENTS \parencite{pafilis2015environments} evaluated dictionaries as well as the experimental dictionaries of Gene Ontology Biological Process and Molecular Function.
   Text mining is subsequently performed on the corpus using the dictionary through the EXTRACT tagger \parencite{pafilis2016extract, jensen2016one}. 
   The tagger recognizes the entities of the dictionary in each abstract and full text article and assigns their co-mentions with a score. 
   The score is sensitive to the text structural level of co-mention; higher to lower scoring when co-mention appears in the same sentence, then, in the same paragraph, and lastly in the same article. 
   All these are integrated and normalized to a single score for each association, as implemented in STRING 9.1 \parencite{franceschini2012string} (see Appendix ~\ref{app:C} for more details). 
   In addition, the tagger extracts each mention in every article to provide the origin of each association it extracts.


   \subsection{Annotated Genomes and Isolates}
   \label{subsec:prego-isolates}

   Annotated genomes and isolates comprise the most trustworthy data in PREGO's knowledge base because they refer to a single species/strain and also have manually curated metadata. 
   Among other data types, JGI-IMG \parencite{chen2021img, mukherjee2021genomes} includes millions of genes from isolated genomes (isolates), SAGs and MAGs. Such annotations, along with their corresponding metadata, were collected using web-parsing technologies. Their metadata, describing their related environment/ecosystem, were tagged using the EXTRACT tagger to infer organisms—environments associations. The annotated KEGG terms were mapped to GOmf terms (see Appendix A). The GOmf terms were then used to extract organisms—processes associations.
   
   The Struo pipeline \parencite{de2020struo} and its outcome when using the Genome Taxonomy DataBase (GTDB) (v.03-RS86) \parencite{parks2020complete} was exploited to enrich organisms—processes associations. 
   A set of 21,276 representative genomes, accompanied by UniRef50 annotations, was retrieved using the provided FTP server. The annotations were then mapped to GOmf terms. 
   Related GTDB genomes were mapped to their corresponding NCBI taxa. 
   All associations extracted from these resources were assigned arbitrarily a confidence level of four out of five. 
   This score choice reflects the high-quality of these data and metadata.
   
   In addition, BioProject data were integrated to PREGO using the NCBI FTP/e-utils services \parencite{sayers2021database}. 
   The BioProject ids that were integrated are the ones that have been assigned a PubMed abstract, a unicellular taxon, and Genome sequencing as data type. Then, using the text mining pipeline, associations were extracted connecting the assigned taxon with the rest of the entities that appear in the abstracts. This method resulted in associations that were assigned a confidence level of three (out of five) because of the combined method of curated data with text mining.


   \subsection{Environmental Samples}
   \label{subsec:prego-envsamples}

   MGnify \parencite{mitchell2020mgnify} and MG-RAST \parencite{wilke2015restful} repositories provide a great number of public metagenomic records. 
   In the PREGO framework, both amplicon and shotgun metagenomic analyses are retrieved periodically along with their corresponding metadata. 
   Data retrieval from these resources is possible from their Application Programming Interfaces (APIs). Marker gene analyses are retrieved and by measuring
   the co-occurrence of taxa present in the various environmental types (e.g., biomes, materials, features, etc.) organisms—environments associations are extracted. 
   These associations emerge when a taxon is reported together with a certain environmental type, being mentioned in the metadata of a sample (metadata based co-occurrence). 
   Similarly, analyses of metagenomic samples along with their corresponding metadata and annotations are also retrieved and organisms—environments, organisms—processes and processes—environments are extracted. 
   The processes—environments associations are possible through co-occurrence of the functional annotations of metagenomes with the environmental metadata of the samples.
   
   In all cases, the EXTRACT tagger is used on the microorganism names and the corresponding metadata of each sample to identify their identifiers (NCBI ids, ENVO terms, GOmf, GObp). 
   All associations in this channel are scored based on the number of samples the entity of interest co-occurs with specific sample metadata (e.g., environmental type) or annotations (functional annotations or taxonomic annotations). 
   The same scoring scheme was implemented across the channel resources (see Appendix~\ref{app:C} for more details), which ranks these associations with a value in the (0,5] space.


   \subsection{Sequence Search}
   \label{subsec:prego-seq-search}

   In the case of organisms, PREGO enables sequence-based queries, meaning a sequence (amplicon) can be used as an entry point like it was a taxon name. 
   To this end, a custom database was built using a set of reference custom databases for four commonly used marker genes. For 16S and 18S rRNA, the SILVA database (v.138) \parencite{quast_silva_2013} and the PR2 database (version\_4.14.0) \parencite{guillou2012protist, del2018eukref} were used. 
   Cytochrome c oxidase I (COI) \parencite{suter2021capturing} is another commonly used marker gene; 
   for this reason, Midori 2 (version GB243) \parencite{leray2018midori} was integrated in PREGO's custom database. 
   Finally, for the Internal transcribed spacer (ITS), common in studies focusing on Fungi, the Unite (version 8.3, accessed 10.05.2021) \parencite{nilsson2019unite} database was added.



   \subsection{Back-End Server and Front-End Implementation}
   \label{subsec:prego-back-end}

   PREGO is a multi-tier web-based application. 
   It is hosted on a 64 GB RAM DELL R540, 20 core, Debian server. 
   Custom API clients (written in Python) are responsible for retrieving the data and metadata from each source (Figure~\ref{fig:prego-pipeline}, step 2). 
   These clients as well as the subsequent methodology (Figure~\ref{fig:prego-pipeline}, step 3 to 6) are updated in regular cycles using custom daemons (see Appendix~\ref{app:B}, Figure~\ref{fig:devops}). 
   The \textit{mamba/blackmamba} web framework underlies communication to the Postgres association-holding database and the client-side communication. 
   HTML 5, Ajax, JQuery, and custom Javascript enhance the user web experience. 
   PREGO supports widely used browsers (e.g., Chrome, Firefox, Safari, Edge) in various operating systems, such as Windows 10, Linux (Ubuntu 18), and MacOS (10.12, 11).


% RPEGO RESULTS
\section{Results \& Validation}
\label{sec:prego-results}

   \subsection{The PREGO Web Resource}
   \label{subsec:prego-web-resource}

   Users can access the PREGO contents through its web User Interface (UI) (Figures~\ref{fig:prego_ui} and \ref{fig:prego_ui_resources}), its Application Programming Interface (API) (Figure~\ref{fig:prego_api_schema}), or bulk download of all associations (Appendix D). 
   The User Interface comes with two search fields: a plain text search and a sequence search (Figure~\ref{fig:prego_ui}a). 
   The latter is used when the user wants to search for a taxon sequence (see Section~\ref{subsec:prego-seq-search} for supported sequence databases). 
   The plain text search supports three types of entry points; the user can search for a taxon name, e.g., \textit{Methanosarcina mazei}, an environmental type, e.g., lagoon, or a biological process e.g., methanogenesis. 
   In all entry points, PREGO returns an overview page consisting of tabs with associations of the entity of interest with the entities of the two other types (Figure~\ref{fig:prego_ui}b–d) as well as Documents and Downloads tabs (Figure~\ref{fig:prego_ui}e,f).


   \begin{figure}[h]
      \centering
      \includegraphics[width=0.98\columnwidth]{figures/prego_ui.png}
      \caption[PREGO web user interface]{
         PREGO web user interface. 
         (a) There are two search fields, plain text and taxa sequences. 
         (b-d) three associations tabs each one presenting associations of the querred entity with the respective entities, Environments (b), Biological Process (c ) and Molecular Function (d). 
         Three channels of information are distinguishing the associations based on the original data. 
         (e) Documents tab presents the scientific articles that mention the queried entity highlighted with color. 
         (f) Downloads tab provides the associations of each channel (when available) to be downloaded in JSON and TSV format.
      }
      \label{fig:prego_ui}
   \end{figure}


   Regarding the association tabs, when a taxon is used as a query, PREGO returns an overview page consisting of tabs for environments, biological processes, and molecular functions. When an environmental type is used as input, PREGO returns the organisms that have been found to be related to it, as well as the Biological Processes observed in the given environment. 
   Lastly, if a biological process is under study, PREGO returns a tab with the organisms along with another tab with the Environments related to the process. 
   Notably, only the associations with scores higher than $0.5$ are presented in the web platform and are sorted in descending order based on their score. 
   The score is visualized with a five-star system (see Appendix~\ref{app:C} for the scoring scheme).
   Every association tab contains three tables with associations derived from the PREGO channels (see Section~\ref{sec:prego-methods}) along with their supported evidence. 
   The user can both search and scroll through these tables, which makes knowledge extraction easier in cases where, for example, Isolate data contain hundreds of associations. 
   In the \textit{Literature} channel, each association is supported by the scientific articles with text-mining identified co-mentions. 
   When a user clicks on an association, a popup window appears. This window displays abstracts or excerpts of full text with the associated entities highlighted (Figure~\ref{fig:prego_ui_resources}a). 
   Additionally, the Environmental Samples and Genome annotations and Isolates channels support evidence for each association by providing links to more detailed information. 
   In the former channel, when the users click on an association, they are redirected to pertinent sample pages of MGnify (Figure~\ref{fig:prego_ui_resources}b). 
   Similarly, the latter redirects users to JGI and NCBI genomes when the associations originated from JGI—IMG and Struo, respectively (Figure~\ref{fig:prego_ui_resources}c).


   \begin{figure}[h]
      \centering
      \includegraphics[width=0.85\textwidth]{figures/prego_ui_resources.png}
      \caption[PREGO in action - examples]{ 
         Each association is supported by original data. 
         (a) Literature channel has a pop-up functionality that displays the scientific articles that each specific association occurs with highlighted color. 
         (b) Environmental Samples channel redirects to the samples that support the specific association (currently only is supported MGnify). 
         (c) Annotated Genomes channel similarly redirects to the isolates ids that each association is based on (both Struo and JGI IMG are supported).
      }
      \label{fig:prego_ui_resources}
   \end{figure}


   The \textit{Documents} tab includes a list of scientific publications where the queried entity is mentioned. 
   Through the \textit{Downloads} tab, users are able to get all of the PREGO associations found for their query, per entity type (e.g., all the environments found related to an organism) and per channel (e.g., all the Environments found related to an organism through the \textit{Literature} channel). 
   This data retrieval functionality is also available via the PREGO API (syntax described in Figure~\ref{fig:prego_api_schema}). 
   Finally, all PREGO associations are available for bulk download from each channel (see Table~\ref{table:pregoA1}).



   \begin{figure}[h]
      \centering
      \includegraphics[width=135mm]{figures/figure_4_PREGO API.png}
      \caption[The PREGO API schema]{The PREGO API schema.}
      \label{fig:prego_api_schema}
   \end{figure}


   \subsection{PREGO in Action}
   \label{subsec:prego-action}

   To demonstrate PREGO's potential, we present four different ways that PREGO can assist molecular ecologists. 
   The demo focuses on the sulfate-reducing microorganisms (SRMs) as well as the processes and environments that relate to sulfate reduction. 
   Through this demo, we highlight how the different channels may provide complementary insights regarding different taxonomic levels and different association types.

   \subsection*{\textit{Which Environments Are Related to a Taxon?}}
   \label{subsec:envo-taxa}

   Based on Pavloudi et al. (2017) \parencite{pavloudi2017diversity}, several bacterial and archaeal SRM were found in lagoonal sediments, after amplifying and sequencing the dissimilatory sulfite reductase $\beta$-subunit (dsrB). 
   Using PREGO for the case of Desulfobacteraceae, the family in which the majority of the observed OTUs of the study belonged to, several environmental types similar to lagoons were retrieved from both the \textit{Literature} and the \textit{Environmental samples} channels (Figure~\ref{fig:prego_ui_resources}a,b). Moreover, most of them had a high $z$-score, such as \textit{"sediment"}, \textit{"sludge"}, and \textit{"activated sludge"}. 
   Several dissimilar environmental types were associated with Desulfobacteraceae, e.g., \textit{"oil reservoir"} indicating them as potential environments where sulfate reduction takes place. 
   However, the presence of taxa within that family in different environments, from \textit{"sea water"} to \textit{"forest"} and \textit{"Wastewater treatment plant"}, may suggest that this family has ubiquitous representatives in diverse conditions.

   Searching for \textit{Desulfatiglans anilini} (\href{https://prego.hcmr.gr/example1}{example1}\footnote{\href{https://prego.hcmr.gr/example1}{https://prego.hcmr.gr/example1}}, accessed on 24 December 2021) at the species level, 
   the most abundant species in Pavloudi et al. (2017) and, 
   for \textit{Desulfatiglans anilini} DSM 4660 strain ({\href{https://prego.hcmr.gr/example2}{example 2}\footnote{\href{https://prego.hcmr.gr/example2}{https://prego.hcmr.gr/example2}}, accessed on 24 December 2021), 
   PREGO provides associations with the \textit{"Anaerobic sediment"}, \textit{"Marine oxygen minimum zone"}, and \textit{"Anaerobic digester sludge"} terms. 
   These associations further corroborate the relationship between the species and sulfate reduction. 
   More specifically, the \textit{"sulfur spring"} ENVO term was retrieved from the Environmental samples channel as well.


   \subsection*{\textit{Which Biological Processes and Molecular Functions Are Related to a Taxon?}}
   \label{subsec:proc-taxa}

   According to Pavloudi et al. (2017), \textit{Desulfatiglans anilini} plays an important role in sulfate reduction. 
   The Biological Processes provided by PREGO's Literature channel are the GO terms \textit{"Sulfate reduction"}, \textit{"Sulfide oxidation"}, and \textit{"Sulfide ion homeostasis"}, which support this claim. 
   In addition, the \textit{"Denitrification pathway"} term was also retrieved. 
   This is rather interesting as it is in line with what Pavloudi et al. (2017) discussed about the SRMs and their ability to use various electron acceptors, e.g., nitrate and nitrite.

   Furthermore, PREGO's Molecular Function tab provides more insight on this example. Several GO terms related to sulfate reduction (e.g., terms related to \textit{"sulfite reductase"}) were associated with DSM 4660 strain and Desulfatiglans anilini species in multiple channels. Interestingly, in the case of the strain query, the Annotated Genomes channel returned many GO terms related to the nitrogen fixation, e.g., \textit{"nitric oxide dioxygenase activity"}.


   \subsection*{\textit{Which Taxa Are Related to a Biological Process?}}
   \label{subsec:taxa-proces}

   PREGO can be also used to report organisms that relate to a certain biological process. Searching for \textit{"dissimilatory sulfate reduction"} associations with taxa (\href{https://prego.hcmr.gr/example3}{example 3}\footnote{\href{https://prego.hcmr.gr/example3}{https://prego.hcmr.gr/example3}}, accessed on 24 December 2021) resulted in several taxa that were mentioned in the Pavloudi et al. (2017) study. 
   For example, taxa such as Thermodesulfobacteria and Thermodesulfovibrio were found among the entries with the highest score (e.g.,) based on the Literature channel. 
   The other two channels did not contain any associations. 
   Using the \textit{"Sulfate assimilation"} (\href{https://prego.hcmr.gr/example4}{example 4}\footnote{\href{https://prego.hcmr.gr/example4}{https://prego.hcmr.gr/example4}}, accessed on 24 December 2021) as the biological process input, PREGO results showed several genera that were missing from PREGO results concerning the \textit{"dissimilatory sulfate reduction"}. 
   Hence, manual search of GObp terms that describe the actual biological process of interest is more insightful.


   \subsection*{\textit{Are There Any Associations between Environments and Biological Processes?}}
   \label{subsec:envo-proc}

   Are there other environmental types, except the lagoonal sediments, in which sulfate assimilation occurs? In that question, and in \textit{"dissimilatory sulfate reduction"} (\href{https://prego.hcmr.gr/example3}{example 3}) in particular, PREGO assigns the highest score to “sediment” while, among others, \textit{"anoxic water"}, \textit{"oil reservoir"}, \textit{"mud volcano"}, and \textit{"basalt"} are potentially associated with environments related to sulfate reduction.
   
   Inversely, PREGO is insightful about occurring processes in a specific environmental type. 
   For example, searching for the biological processes that take place in \textit{"basalt"} (\href{https://prego.hcmr.gr/example5}{example 5}\footnote{\href{https://prego.hcmr.gr/example5}{https://prego.hcmr.gr/example5}}, accessed on 24 December 2021), processes like \textit{"Nitrogen fixation"} and \textit{"Reactive nitrogen species metabolic process"} stand out. 
   However, sulfate reduction is not among the associations. 
   However, when asking for \textit{"Mafic lava"} (\href{https://prego.hcmr.gr/example6}{example 6}\footnote{\href{https://prego.hcmr.gr/example6}{https://prego.hcmr.gr/example6}}, accessed on 24 December 2021), both the \textit{"nitrogen fixation"} and \textit{"Sulfur compound metabolic process"} terms are returned. 
   This highlights the suggestions of Pavloudi et al. (2017), regarding the potential use of various electron acceptors from the different strains present in different environmental types.


   \subsection{PREGO Contents}
   \label{subsec:prego-contents}

   PREGO contains the literature, environmental samples, and genome annotations of the resources shown in Table~\ref{table:prego1}. 
   The extracted contents of these resources have resulted to a knowledge base with ~364 K distinct taxonomic groups (out of a pool of ~$620$K Bacteria, Archaea, and microbial eukaryotes, based on NCBI Taxonomy) from which ~$258$K are at the species level (Table~\ref{table:prego2}). 
   These taxa are associated with ~1 K Environment Ontology terms, ~15 K GObp terms, and with ~7.9 K GOmf terms. 
   Combining the above, PREGO maintains a knowledge base of entities and associations between them that form a multipartite network with entities as nodes and scored associations between them as weighted links.


   % PREGO ENTITIES AFTER NER AND MAPPING - table:prego2
   \begin{table}[ht]
      \begin{tabular}{@{}cccrccc@{}}
      \toprule
      \textbf{Channel} & \textbf{Source} & \multicolumn{2}{c}{\textbf{Taxonomy}} & \textbf{\begin{tabular}[c]{@{}c@{}}Environ- \\ ments\end{tabular}} & \textbf{\begin{tabular}[c]{@{}c@{}}Biological \\ Processes\end{tabular}} & \textbf{\begin{tabular}[c]{@{}c@{}}Molecular \\ Functions\end{tabular}} \\ \midrule
      \multirow{3}{*}{Literature} & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MEDLINE \\ PubMed - \\ PMC OA\end{tabular}} & Strains & 8,929 & \multirow{3}{*}{1,077} & \multirow{3}{*}{15,079} & \multirow{3}{*}{7,318} \\
      &  & Species & 240,377 &  &  &  \\
      &  & Total & 342,506 &  &  &  \\
      \multirow{9}{*}{\begin{tabular}[c]{@{}c@{}}Environ- \\ mental \\ samples\end{tabular}} & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MG-RAST \\ amplicon\end{tabular}} & Strains & 1,392 & \multirow{3}{*}{162} & \multirow{3}{*}{-} & \multirow{3}{*}{-} \\
      &  & Species & 4,324 &  &  &  \\
      &  & Total & 5,859 &  &  &  \\
      & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MG-RAST \\ metagenome\end{tabular}} & Strains & 2,522 & \multirow{3}{*}{258} & \multirow{3}{*}{-} & \multirow{3}{*}{3,839} \\
      &  & Species & 4,406 &  &  &  \\
      &  & Total & 7,157 &  &  &  \\
      & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MGnify \\ amplicon\end{tabular}} & Strains & 2 & \multirow{3}{*}{216} & \multirow{3}{*}{11} & \multicolumn{1}{l}{\multirow{3}{*}{-}} \\
      &  & Species & 1,471 &  &  & \multicolumn{1}{l}{} \\
      &  & Total & 2,955 &  &  & \multicolumn{1}{l}{} \\
      \multirow{9}{*}{\begin{tabular}[c]{@{}c@{}}Annotated\\Genomes \&\\Isolates\end{tabular}} & \multirow{3}{*}{JGI IMGisolates} & Strains & 2,398 & \multirow{3}{*}{241} & \multirow{3}{*}{-} & \multirow{3}{*}{3,670} \\
      &  & Species & 11,203 &  &  &  \\
      &  & Total & 13,849 &  &  &  \\
      & \multirow{3}{*}{STRUO} & Strains & 6 & \multirow{3}{*}{-} & \multirow{3}{*}{-} & \multirow{3}{*}{2,789} \\
      &  & Species & 19,289 &  &  &  \\
      &  & Total & 19,325 &  &  &  \\
      & \multirow{3}{*}{BioProject} & Strains & 5,754 & \multirow{3}{*}{309} & \multirow{3}{*}{626} & \multirow{3}{*}{-} \\
      &  & Species & 3,373 &  &  &  \\
      &  & Total & 9,393 &  &  &  \\
      \multirow{3}{*}{Total} & \multirow{3}{*}{All} & Strains & 12,840 & \multirow{3}{*}{1,090} & \multirow{3}{*}{15,091} & \multirow{3}{*}{7,971} \\
      &  & \multicolumn{1}{l}{} & \multicolumn{1}{l}{} &  &  &  \\
      &  & \multicolumn{1}{l}{} & \multicolumn{1}{l}{} &  &  &  \\ \bottomrule
      \end{tabular}

      \caption[The entities of PREGO after the NER and mapping of every source]{
         The entities of PREGO after the NER and mapping of every source. 
         Counts of distinct entities of Taxa, Environments (ENVO terms), Biological Processes (Gene Ontology Biological process) and Molecular Function (Gene Ontology Molecular Function).
      }
      \label{table:prego2}

   \end{table}


   As shown in Figure~\ref{fig:prego-entities}, in its current version (December 2021), PREGO knowledge base covers $157$ bacterial phyla ($107$ are Candidatus), $23$ phyla from archaea ($18$ are Candidatus), and $22$ unicellular eukaryotic phyla described in the NCBI Taxonomy database. 
   The number of bacterial taxa present among the associations of each phylum ranges from the order of $10$s, as in the case of \textit{Candidatus Coatesbacteria}, to hundreds of thousands, e.g., Actinobacteriae. 
   The number of environmental types, found among the PREGO associations for each phylum, ranges from just a few to up to $1000$. 
   Similarly, the number of biological processes that have been related to the various phyla may range from less than a dozen, e.g., Yanofskybacteria to up to several thousands, e.g., Bacteroidetes. On the contrary, the number of molecular functions found to be related to taxa of each phylum is rather constant in all three domains.

   \begin{figure}[h]
      \centering
      \includegraphics[width=0.85\textwidth]{figures/figure_5_all_channels_ranks.png}  
      \caption[Summary of the unique entities per phylum for each of the four entity types on PREGO]{Summary of all the unique entities per phylum for each of the four entity types (in log10 scale) that appear in PREGO. Phyla are grouped based on their superkingdom (in log10 scale). Only phyla for which associations are available in the PREGO platform are mentioned.}
      \label{fig:prego-entities}
   \end{figure}


% RPEGO DISCUSSION
\section{Discussion}
\label{sec:prego-discussion}

   \subsection{PREGO Contents}
   \label{subsec:prego-contents-disc}

   On its current version and according to the NCBI Taxonomy that it is based on, PREGO manages to cover a great range of microbial taxa, as most (if not all phyla) are present in the knowledge base (Figure~\ref{fig:prego-entities}). 
   The different number of organisms' entities per phylum highlights the diverse number of the members of the various phyla. On the contrary, the similar number of molecular functions in all cases indicates the robustness of the main metabolic processes required for life. 
   With respect to biological processes, their number per phylum varies to some extent, especially for the case of Bacteria and Archaea. 
   That could be observed as, in many cases, phyla that have been recently described using molecular techniques have not been studied extensively yet, e.g., Candidatus Delongbacteria. 
   As expected, the number of environmental types that have been associated with members of each phylum varies, as a phylum may be universally present, while others could be strongly niche-specific (e.g., Hydrothermarchaeota).

   Because of its three different channels, PREGO manages to extract associations both in the species and higher taxonomic levels. The Isolates channel supports explicit associations at the species level (Table~\ref{table:prego3} and Figure S3). 
   Interestingly, the number of such genomes seems to have reached a plateau for now, as PREGO-like platforms include the same order of magnitude. 
   The \textit{Literature} channel, on the other hand, promotes the extraction of associations at higher taxonomic levels (Table~\ref{table:prego3} and Figure S1). 
   This also applies to environment—organisms associations derived from the Environmental Samples channel (Table~\ref{table:prego3} and Figure S2). Associations regarding biological processes, though, are strongly enhanced by the Literature channel and the massive increase of literature.


   % ASSOCIATIOS TABLE - table:prego3
   \begin{sidewaystable}
      \begin{tabular}{@{}cccccrcr@{}}
      \toprule
      \textbf{Channel} & \textbf{Source} & \textbf{\begin{tabular}[c]{@{}c@{}}Environments \\ - \\ Processes\end{tabular}} & \textbf{\begin{tabular}[c]{@{}c@{}}Environments \\ - \\ Functions\end{tabular}} & \textbf{Taxonomy} & \multicolumn{1}{c}{\textbf{\begin{tabular}[c]{@{}c@{}}Taxa \\ -\\  Environments\end{tabular}}} & \textbf{\begin{tabular}[c]{@{}c@{}}Taxa \\ - \\ Processes\end{tabular}} & \multicolumn{1}{c}{\textbf{\begin{tabular}[c]{@{}c@{}}Taxa \\ -\\ Function\end{tabular}}} \\ \midrule
      \multirow{3}{*}{Literature} & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MEDLINE \\ PubMed - \\ PMC OA\end{tabular}} & \multirow{3}{*}{883,997} & \multirow{3}{*}{422,579} & Strains & 69,968 & \multicolumn{1}{r}{590,630} & 384,079 \\
      &  &  &  & Species & 778,877 & \multicolumn{1}{r}{3,501,635} & 1,961,920 \\
      &  &  &  & Total & 1,669,608 & \multicolumn{1}{r}{7,969,310} & 4,613,827 \\
      \multirow{9}{*}{\begin{tabular}[c]{@{}c@{}}Environmental \\ samples\end{tabular}} & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MG-RAST \\ amplicon\end{tabular}} & \multirow{3}{*}{-} & \multirow{3}{*}{-} & Strains & 13,645 & \multirow{3}{*}{-} & \multicolumn{1}{c}{\multirow{3}{*}{-}} \\
      &  &  &  & Species & 39,007 &  & \multicolumn{1}{c}{} \\
      &  &  &  & Total & 53,439 &  & \multicolumn{1}{c}{} \\
      & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MG-RAST \\ metagenome\end{tabular}} & \multirow{3}{*}{-} & \multirow{3}{*}{620,846} & Strains & 262,106 & \multirow{3}{*}{-} & 8,626,328 \\
      &  &  &  & Species & 103,913 &  & 10,715,548 \\
      &  &  &  & Total & 372,301 &  & 19,950,096 \\
      & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}MGnify \\ amplicon\end{tabular}} & \multirow{3}{*}{-} & \multirow{3}{*}{-} & Strains & 18 & - & \multicolumn{1}{l}{} \\
      &  &  &  & Species & 30,122 & \multicolumn{1}{r}{351} & \multicolumn{1}{c}{-} \\
      &  &  &  & Total & 111,976 & \multicolumn{1}{r}{2,097} & \multicolumn{1}{l}{} \\
      \multirow{9}{*}{\begin{tabular}[c]{@{}c@{}}Annotated Genomes \\ and Isolates\end{tabular}} & \multirow{3}{*}{\begin{tabular}[c]{@{}c@{}}JGI IMG\\ isolates\end{tabular}} & \multirow{3}{*}{-} & \multirow{3}{*}{-} & Strains & 8,229 & \multirow{3}{*}{-} & 3,461,693 \\
      &  &  &  & Species & 42,141 &  & 13,216,559 \\
      &  &  &  & Total & 50,888 &  & 16,821,850 \\
      & \multirow{3}{*}{STRUO} & \multirow{3}{*}{-} & \multirow{3}{*}{-} & Strains & \multicolumn{1}{c}{\multirow{3}{*}{-}} & \multirow{3}{*}{-} & 1,803 \\
      &  &  &  & Species & \multicolumn{1}{c}{} &  & 4,070,195 \\
      &  &  &  & Total & \multicolumn{1}{c}{} &  & 4,079,312 \\
      & \multirow{3}{*}{BioProject} & \multirow{3}{*}{-} & \multirow{3}{*}{-} & Strains & 3,263 & \multicolumn{1}{r}{7,473} & \multicolumn{1}{l}{} \\
      &  &  &  & Species & 4,187 & \multicolumn{1}{r}{4,294} & \multicolumn{1}{l}{} \\
      &  &  &  & Total & 7,641 & \multicolumn{1}{r}{12,169} & \multicolumn{1}{l}{} \\
      \multirow{3}{*}{Total} & \multirow{3}{*}{All} & \multirow{3}{*}{883,997} & \multirow{3}{*}{1,043,425} & Strains & 357,229 & \multicolumn{1}{r}{598,103} & 12,473,903 \\
      &  &  &  & Species & 998,247 & \multicolumn{1}{r}{3,506,280} & 29,964,222 \\
      &  &  &  & Total & 2,265,853 & \multicolumn{1}{r}{7,983,576} & 45,465,085 \\ \cmidrule(l){5-8} 
      \end{tabular}
      \caption[Associations among the PREGO entities]{
         The associations between entities of PREGO after co-occurrence analysis: The supported entity types of associations are Environments—Biological Processes, Environments—Molecular Functions, Taxa—Environments, Taxa—Biological Processes, Taxa—Molecular Functions.
      }
      \label{table:prego3}
   \end{sidewaystable}

   Additionally, the text mining methodology of the Literature channel has retrieved most of the entities present in PREGO knowledge base (Table~\ref{table:prego2}). 
   A significant contribution to the taxa with associations is due to the PMC OA processing by the text mining pipeline of the Literature channel. 
   This is in-line with reports in other applications of text mining when using full text articles \parencite{westergaard2018comprehensive}. 
   However, the resulting associations are suggestive because of the text mining nature, and therefore subject for further review by the users.

   \subsection{Related Tools' Functionality and Content}
   \label{subsec:prego-similar-platforms}

   There is an emerging niche for tools similar to PREGO to bring forward microbe associations and metadata. 
   Table~\ref{table:prego4} summarizes the common and different features of BacDive, WoM, NMDC data portal, and PREGO. 
   All of them commonly share the environmental associations and biological/metabolic processes with the microbes.

   BacDive is a well-established platform with a focus on phenotype and cultivation information for about 100,000 prokaryotes, bacteria, and archaea. 
   It has a high level of curation for most of its input types, like literature, internal databases, and personal collections. 
   The NMDC data portal has published the scheme, the user interface, and a demonstrative collection of samples that will be populated later on. Standout features are the spatial visualization with coordinates and the detailed information of the samples, e.g., sequencing instruments and methodology. 
   An alternative approach is facilitated by WoM, which aims to bind chemistry to microbes. An environment, in particular, is defined as the starting metabolite pool that is transformed by an organism.
   Another tool is The Microbe Directory that contains fully curated metadata for about 8000 microbes from all superkingdoms. This tool focuses on conditions of growth and on host taxa.

   Complementary to these tools, PREGO contains associations of bacteria, archaea, and eukaryotes. Distinctive features are the associations of environments with processes/functions and the large-scale literature integration with text mining. Most importantly, most of the tools are complementary to each other with minimum overlap, an indication of the opportunities for further innovative synergies.


   % PREGO COMPARISONS - table:prego4
   \begin{table}[ht]

      \begin{adjustwidth}{-0.75cm}{}

      \begin{tabular}{@{}lllll@{}}
      
      \toprule
      Functionality & BacDive & Web of Microbes & NMDC & PREGO \\ \midrule
      manual curation & high & high & intermediate & low \\ 

      literature integration & limited & no & no & yes \\

      environment—taxa associations & yes & yes & yes & yes \\

      \begin{tabular}[c]{@{}l@{}}environment—process/\\ function associations\end{tabular} & no & no & no & yes \\

      process/function—taxa associations & yes & yes & yes & yes \\
      phenotypic data & yes & no & no & no \\

      data origin & \begin{tabular}[c]{@{}l@{}} original \\integration \end{tabular} & original & \begin{tabular}[c]{@{}l@{}} original \\integration \end{tabular} & integration \\

      spatial coordinates & yes & no & yes & no \\

      application programming interface & yes & no & yes & yes \\

      bulk download & limited & yes & yes & yes \\ \bottomrule

      \end{tabular}
      \end{adjustwidth}
      \caption[Feature comparison between PREGO and other similar platforms]{Feature comparison among platforms that facilitate knowledge discovery and integration of microbial data.}
      \label{table:prego4}
   
   \end{table}      

   \subsection{PREGO Next Steps}
   \label{subsec:prego-next-steps}

   PREGO is a user-friendly association mining and sharing platform. 
   Its modular web-architecture grants it the flexibility for further improvements in the aforementioned aspects, namely: 
   source datasets, user interface, entity, and association scope expansion. Regarding datasets, additional data, such as transcriptomes from MGnify and other records annotated with metadata from studies in \href{https://ebi-metagenomics.github.io/blog/2021/11/17/Publication-Annotations/}{EuroPMC}, accessed on 24 December 2021) \parencite{ferguson2021europe}, could be incorporated. 
   Similarly, the \href{https://data.microbiomedata.org/}{NMDC data platform standards-compliant annotated records}\footnote{\href{https://data.microbiomedata.org/}{https://data.microbiomedata.org/}} (accessed on 24 December 2021) could serve as an additional resource with its high-quality metadata \parencite{wood2020national, vangay2021microbiome}. 
   Reciprocally, if requested, pertinent literature and association summaries could be programmatically offered to interested third parties.

   Furthermore, the entity types supported by the PREGO system could be expanded. For example, GOmf terms could be upgraded as a search-entry point to the system. 
   In addition, disease and tissue describing terms, already supported by the PREGO-underlying EXTRACT system \parencite{pafilis2016extract}, could enter the PREGO ecosystem of associated entities. 
   From a statistics perspective, the calculation of a combined association score, when an association is reported by more than one channel of information, could be another feature to add.

   The user interface can be enhanced to support multiple entity and/or sequence queries, instead of single ones. 
   Sequences can be processed by taxonomy assignment pipelines (e.g., PEMA \parencite{zafeiropoulos2020pema}) and be converted into searching PREGO for associations. 
   In addition, network visualization tools, like Arena3Dweb \parencite{karatzas2021arena3dweb}, could allow interactive browsing of associations through multi-layered graphs. 
   Enrichment analyses, like those performed by OnTheFly2.0 \parencite{baltoumas2021onthefly2} or Flame \parencite{thanati2021flame}, 
   can be incorporated. 
   Omics data analysis pipelines, like MiBiOmics \parencite{zoppi2021mibiomics}, environment associations with sequences using SeqEnv \parencite{sinclair2016seqenv} and biogeochemical associations with metagenomic data with DiTing~\parencite{xue_diting_2021} could be enabled by comparing the associations pertinent to different groups of entities. 
   The computationally intensive tasks of multiple queries, taxonomy assignments to sequences and enrichment analysis could be offered by our in-house High Performance Computing facility (https://hpc.hcmr.gr/, accessed on 24 December 2021) \parencite{zafeiropoulos_0s_2021} in synergy with pertinent Research Infrastructures like \href{https://elixir-europe.org}{ELIXIR}\footnote{\href{https://elixir-europe.org}{https://elixir-europe.org}} (accessed on 24 December 2021) and \href{https://www.lifewatch.eu/}{LifeWatch ERIC}\footnote{\href{https://www.lifewatch.eu/}{https://www.lifewatch.eu/}} (accessed on 24 December 2021).



   \subsection*{Availability of Supporting Source Codes:} 
   The PREGO software modules are available under BSD 2-Clause “Simplified” License. Scripts, where additional libraries have been used, are subject to their individual licenses. More information on each module can be found as listed below:
   
   \begin{itemize}
      \item prego\_gathering\_data 
      \href{https://github.com/lab42open-team/prego_gathering_data}{github.com/lab42open-team/prego\_gathering\_data}
      \item prego\_daemons \href{https://github.com/lab42open-team/prego_daemons}{github.com/lab42open-team/prego\_daemons}
      \item prego\_mappings \href{https://github.com/lab42open-team/prego_mappings}{github.com/lab42open-team/prego\_mappings} 
      \item prego\_statistics \href{https://github.com/lab42open-team/prego_statistics}{github.com/lab42open-team/prego\_statistics}
   \end{itemize}

   Additional software and curated lists along with their individual license are:
   \begin{itemize}
      \item tagger:	\href{https://github.com/larsjuhljensen/tagger}{https://github.com/larsjuhljensen/tagger}, BSD 2-Clause "Simplified" License
      \item mamba: \href{https://github.com/larsjuhljensen/mamba}{https://github.com/larsjuhljensen/mamba}, BSD 2-Clause "Simplified" License 
      \item tagger dictionary:  \href{https://download.jensenlab.org/}{https://download.jensenlab.org/} and there in: \\
      \href{https://download.jensenlab.org/prego_dictionary.tar.gz}{https://download.jensenlab.org/prego\_dictionary.tar.gz}, CC-BY 4.0 license
   \end{itemize}


PREGO produces entity identifiers either by Named Entity Recognition (NER) with the EXTRACT tagger or by mapping retrieved identifiers to the selected ones. 
PREGO adopted NCBI taxonomy identifiers for taxa, Environmental Ontology for environments and Gene Ontology as a structure knowledge scheme for Processes (GObp) and Molecular Functions (GOmfs). 
The latter was for reasons that are two-fold, first Gene Ontology has a Creative Commons Attribution 4.0 License and second there are many resources that have mapped their identifiers to Gene Ontology.
MG-RAST metagenomes and JGI/IMG isolates annotations come with KEGG orthology (KO) terms; 
Struo-oriented genome annotations, on the other hand, have Uniprot50 ids. 
The mapping from KO to GOmf and Uniprot50 to GOmf is implemented via UniProtKB mapping files of their FTP server (see \texttt{idmapping.dat} and \texttt{idmapping\_selected.tab} files). 
By using the 3-column mapping file, the initial annotations were mapped to GOmf. As a complement, a list of metabolism-oriented KEGG ORTHOLOGY (KO) terms has been built (see \textit{prego\_mappings} in the Availability of Supporting Source Codes section).
Finally, as STRUO annotations refer to GTDB genomes, \href{http://ftp.tue.mpg.de/ebio/projects/struo/GTDB_release89/metadata/}{publicly available mappings} (accessed on 24 December 2021) were used to link the genomes used with their corresponding NCBI Taxonomy entries.



\section{Daemons}
\label{app:B}

An important component PREGO approach (Figure A1) is the regular updates which keep PREGO in line with the literature and microbiology data advances. 
The updates are implemented with custom scripts called daemons that are executed regularly spanning from once a month up to six-month cycles. 
This variation occurs because of the API requirements of each web resource as well as the computational intensity of the association extraction from the retrieved data.

\begin{figure}[ht]
   \centering
   \includegraphics[width=95mm]{figures/figure_A1_PREGO_daemons.png}
   \caption[PREGO DevOps]{Software daemons perform all steps of the PREGO methodology in a continuous manner similar to the Continuous Development and Continuous Integration method.}
   \label{fig:devops}
\end{figure}


Each Daemon is attached to a resource because its data retrieval methods (API, FTP) and following steps, shown in Figure A1, require special handling and multiple scripts (see \textit{prego\_daemons} in the Availability of Supporting Source Codes section).

\section{Scoring}
\label{app:C}


Scoring in PREGO is used to answer the questions:
\begin{itemize}
   \item Which associations are more thrustworthy?
   \item Which associations are more relevant to the user's query?
\end{itemize}

Relevant, informative, and probable associations are presented to the user through the three channels that were discussed previously. 
Each channel has its own scoring scheme for the associations it contains and all of them are fit in the interval $(0,5]$ to maintain consistency. 
The values of the score are visually shown as stars. 
The Genome Annotation and Isolates channel has fixed values of scores depending on the resource because Genome Annotation is straightforward, and the microbe id is known a priori. 
On the other hand, Environmental Samples channel data are based on samples, which contain metagenomes and OTU tables. 
Thus, it has two levels of organization, microbes with metadata, and sample identifiers. Each association of two entities is scored based on the number of samples they co-occur. 
A Literature channel scoring scheme is based on the co-mention of a pair of entities in each document, paragraph, and sentence. The differences in the nature of data require different scoring schemes in these channels.
The contingency table (Table~\ref{table:pregoA1}) of two random variables, $X$ and $Y$ are the starting point for the calculation of scores. The term $X = 1$ might be a specific NCBI id and $Y = 1$ a ENVO term. 
The $c_{1,1}$ is the number of instances that two terms of $X = 1$ and $Y = 1$ are co-occurring, i.e., the joint frequency. 
The marginals are the $c_{1,.}$ and $c{.,1}$ for $x$ and $y$, respectively, which are the backgrounds for each entity type. 
Different handling of these frequencies leads to different measures. 
There is not a perfect scoring scheme, just the one that works best on a particular instance. 
Consequently, scoring attributes require testing different measures and their parameters.



\begin{table}[ht]
   \centering
   \begin{tabular}{c|llll}
    & \multicolumn{4}{l}{Y = y} \\ \cline{2-5} 
   \multirow{4}{*}{X = x} &  & Yes & No & Total \\ \cline{3-5} 
    & \multicolumn{1}{l|}{Yes} & $c_{x,y}$ & $c_{x,0}$ & $c_{x,.}$ \\
    & \multicolumn{1}{l|}{No} & $c_{0,y}$ & $c_{0,0}$ & $c_{0,.}$ \\
    & \multicolumn{1}{l|}{Total} & $c_{.,y}$ & $c_{.,0}$ & $c_{.,.}$
   \end{tabular}
   \caption[PREGO contingency table between two terms]{Contingency table of co-occurrences between entities $X = x$ and $Y = y$. 
   This is the basic structure for all scoring schemes. $c_{x,y}$ is the count of the co-occurrence of these entities. $c_{x,.}$ is the count of the $x$ with all the entities of $Y$ type (e.g., Molecular function). Conversely, $c_{.,y}$ is the count of $y$ with all the entities of $X$ type (e.g., taxonomy}
   \label{table:pregoA1}
\end{table}


\section*{Literature Channel}

Scoring in the Literature channel is implemented as in STRING 9.1 \parencite{franceschini2012string} and COMPARTMENTS \parencite{binder2014compartments}, where the text mining method uses a three-step scoring scheme. 
First, for each co-mention/co-occurrence between entities (e.g., Methanosarcina mazei with Sulfur carrier activity), a weighted count is calculated because of the complexity of the text.  


\begin{equation}
   c_{x,y} = \sum_{k=1}^{n}{w_d \delta_{dk}(x,y) +w_p \delta_{p,k}(x,y) + w_s \delta_{sk}(x,y)}
   \label{eq:prego-score-1}
\end{equation}



Different weights are used for each part of the document ($k$) for which both entities have been co-mentioned, $w_d = 1$ for the weight for the whole document level, $w_p = 2$ for the weight of the paragraph level, and $w_s = 0.2$ for the same sentence weight. 
Additionally, the delta functions are one (Equation~\ref{eq:prego-score-1}) in cases the co-mention exists, zero otherwise. Thus, the weighted count becomes higher as the entities are mentioned in the same paragraph and even higher when in the same sentence.
Subsequently, the co-occurrence score is calculated as follows:

\begin{equation}
   score_{x,y} = c_{x,y}^a (\frac{c_{x,y} c_{.,.}}{c_{x, .}c_{.,y}})^{1-a}
   \label{eq:prego-score-2}
\end{equation}
   


where $a = 0.6$ is a weighting factor, and the $c_{x,.}$, $c_{.,.}$, 
$c_{.,y}$ are the weighted counts as shown in Table~\ref{table:pregoA1} estimated using the same Equation~\ref{eq:prego-score-2}. 
This value of the weighting factor has been chosen because it has been optimized and benchmarked in various 
applications of text mining~\parencite{franceschini2012string, binder2014compartments, pletscher2015diseases}. 
The value of Equation~\ref{eq:prego-score-2} is sensitive to the increasing size of the number of documents (MEDLINE PubMed—PMC OA).
Therefore, to obtain a more robust measure, the value of the score is transformed to $z$-score. 
This transformation is elaborated in detail in the COMPARTMENTS resource \parencite{binder2014compartments}. 
Finally, the confidence score is the $z$-score divided by two. Cases in which the scores exceed the (0,4] interval are capped to a maximum of 4 to reflect the uncertainty of the text mining pipeline.

\section*{Environmental Samples Channel}

Data from environmental samples are OTU tables and metagenomes. 
Thus, for each entity $x$, the number of samples is calculated as the background 
and a number of samples of the associated entity (metadata background) $c.,y$ (see Table~\ref{table:pregoA1}). 
Each association between entities $x$,$y$ has a number of samples, $c_{x,y}$ that they co-occur. 
Note that each resource is independent and the scoring scheme is applied to its entities. 
This means that the same association can appear in multiple resources with different scores. 
The score is calculated with the following formula:

\begin{equation}
   score_{x,y} = 2.0*{\frac{\sqrt{c_{x,y}}}{c_{.,y}^{0.1}}}
\end{equation}


This score is asymmetric because the denominator is the marginal of the associated entity. 
Thus, the score decreases as the marginal of $y$ is increasing, i.e., the number of samples that $y$ is found. 
On the other hand, it promotes associations in which the number of samples of 
the association are similar to the marginal of $y$. 
The exponents on the numerator and denominator equal to $0.5$ and 
to $0.1$, respectively, in order to reduce the rapid increase of score.
Lastly, the value of the score is capped in the range $(0,4]$.


\section{Bulk download}
\label{app:D}

   Users can also download programmatically all associations per channel through the links that are shown in Table~\ref{table:prego-appD-1}. 
   The data are compressed to reduce the download size and md5sum files are provided as well for a sanity check of each download.

   % PREGO BULK DOWNLOAD TABLE 
   \begin{table}[ht]
      
      \begin{adjustwidth}{-2cm}{}

      \begin{tabular}{llll}
      \toprule
      Channel & Link & md5sum & Size (in GB) \\ \midrule

      Literature & \href{https://prego.hcmr.gr/download/literature.tar.gz}{literature.tar.gz} & \href{https://prego.hcmr.gr/download/literature.tar.gz.md5}{literature.tar.gz.md5} & 5.4 \\

      \begin{tabular}[c]{@{}l@{}}Environmental \\ Samples\end{tabular} &
      \href{https://prego.hcmr.gr/download/environmental\_samples.tar.gz}{environmental\_samples.tar.gz} & 
      \href{https://prego.hcmr.gr/download/environmental\_samples.tar.gz.md5}{environmental\_samples.tar.gz.md5}
      & 0.69 \\

      \begin{tabular}[c]{@{}l@{}}Annotated \\ genomes and \\ isolates\end{tabular} & 
      \href{https://prego.hcmr.gr/download/annotated\_genomes\_isolates.tar.gz}{annotated\_genomes\_isolates.tar.gz} &
      \href{https://prego.hcmr.gr/download/annotated\_genomes\_isolates.tar.gz.md5}{annotated\_genomes\_isolates.tar.gz.md5} & 0.26 \\ \bottomrule
      \end{tabular}
      \end{adjustwidth}
      \caption[PREGO Bulk download links and md5sum files.]{Bulk download links and md5sum files.}
      \label{table:prego-appD-1}
   \end{table}
