
\chapter{Crete Microbiome Appendix} % Main appendix title

\label{AppendixD} 

\section{Amplicon 16s rRNA is soil}

\section{Errors in Amplicon Microbial Ecology}

Microbial ecology based on amplicon 16s rRNA sequences has flourished since the
2010s. The endeavor to understand the microbial world faces multiple challenges
across the scientific workflow, from sampling to ecological analyses \parencite{Lee2012}.

Errors propagate starting with the sampling. There are contaminations from the
people in the field, in the lab for the DNA extraction \parencite{EISENHOFER2019105}. 

Errors from PCR amplification and errors from sequencing \parencite{Schloss2011, Schimer2015}.
Approximation and computation errors from algoritms that cluster, measure similarities between
sequences.

Semantic errors because of reductionist approaches and/or oversimplification
of the microbial communities. Microbiome matrix data are compositional \parencite{Gloor2017}

\section{Taxonomy, OTU and ASV}

Amplicon rRNA sequencing provides a collection of sequence reads per sample. 
The ecological interpretation of the reads requires their transformation to
taxonomic information. To do so there are two approaches currently in use, 
the clustering method and the denoising method. With clustering reads are 
grouped together and a best representing sequence is produced for each 
cluster, i.e. the Operetional Taxonomic Unit. This approach makes the OTUs 
from different runs, i.e executions of the algorithm and/or different studies
incoperable and irreproducible.

Currently, many studies propose the use of Amplicon Sequence Variants \parencite{Callahan2017}. 
ASVs are real biological sequences and can be used for comparison.

The influence of the different methods to subsequent ecological analyses has
little impact \parencite{Glassman2018}.

   \begin{figure}[h]
      \centering
      \includegraphics[width=\textwidth,height=\textheight,keepaspectratio]{figures/isd_crete_flowchart.png}
      \caption[Reproducible workflow of ISD analysis]{Three steps of the ISD bioinformatics workflow, reproducible by design}
      \label{fig:isd_workflow_taxonomy}
   \end{figure}
   
   \begin{figure}[h]
      \centering
      \includegraphics[width=\textwidth,height=\textheight,keepaspectratio]{figures/isd_asv_taxonomy_ratios_top_phyla_samples.png}
      \caption[Top phyla of each samples]{The top phyla for all samples with their relative abundance}
      \label{fig:isd_top_phyla_samples}
   \end{figure}
   
   \begin{figure}[h]
      \centering
      \includegraphics[width=\textwidth,height=\textheight,keepaspectratio]{figures/isd_crete_srs_curve.jpeg}
      \caption[SRS curve]{SRS curve of the samples. }
      \label{fig:isd_srs-curve_samples}
   \end{figure}

\begin{figure}[t] 
    \centering\includegraphics[width=\columnwidth]{figures/isd_community_site_locations_dif.png}
\caption{Sample dissimilarity between samples in the same site (2 subsites) and the rest.}
    \label{fig:isd_site_locations}
\end{figure}

